{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "I wanted to make changes to ../../documentation/mvp_v1.md, whilst being able to see the changes rendered. https://github.com/jupyter/notebook/issues/2485 suggests copying the markdown to a new cell in a notebook, then copying back once complete, so this is what I do here."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# File tracking system minimal viable product scope\n",
    "V1.1 by Magnus Manske and Richard Pearson, 2018-07-18\n",
    "\n",
    "## Introduction\n",
    "This document describes the scope of the minimal viable product (MVP) for a new file tracking system (preliminarily designated FITS). The problem that FITS is intended to solve is described in the next section. The scope of the MVP is described in the following section. Following this, some key use cases for FITS are described, including an indication of whether these use cases will be catered for by the MVP. This document concludes with pointers to other relevant documents, relationship to the separate sample information management system (SIMS) project, and future plans.\n",
    "\n",
    "## Problem\n",
    "For our work, we rely on data files generated by machines (e.g. DNA sequencing, genotyping), supplied by third parties, or generated from such files through computation. We need to be able to know where all these files are, what they contain, how they relate to samples, and to each other.\n",
    "\n",
    "Previously, the information described above has either been sourced from the Solaris database, from Sanger sequencing systems (Multi-LIMS warehouse (mlwh), iRODs baton) or from a combination of Solaris and Sanger sequencing systems. This approach is considered non-viable going forwards because:\n",
    "* The Solaris database is being decommissioned and is no longer been populated with new data\n",
    "* There are known errors in file-sample mappings in the Multi-LIMS warehouse\n",
    "* We will need to store data that are not in Sanger systems, e.g. files of samples sequenced at locations other than the Sanger which we want to include in builds and releases\n",
    "\n",
    "## Scope\n",
    "The MVP _will_\n",
    "* Store the location of files in various storage systems\n",
    "* Store metadata of these files (e.g. file type, file size, MD5 sum)\n",
    "* Store metadata about the data contained these files (e.g. insert size of paired end sequencing data)\n",
    "* Store the relations of files to each other (e.g. a BAM and a CRAM file containing the same read data, which files(s) another file is based on)\n",
    "* Store sample identifiers and their relation to the files\n",
    "* Store metadata about the sample IDs (e.g. Oxford/ROMA codes)\n",
    "* Be updated via automated processes (e.g. a cronjob reading Sanger sequencing updates)\n",
    "* Be updated manually (e.g. to correct false legacy data)\n",
    "* Be queried to get information about samples\n",
    "* Be queried to get information about files\n",
    "* Store mappings from files to samples for all parasite and vector MalariaGEN samples that have been whole-genome sequenced using Illumina technology at the Sanger\n",
    "\n",
    "The MVP will _not necessarily_ store mappings from files to samples for \n",
    "* amplicon sequencing data\n",
    "* other genotyping technologies such as Sequenom/Agena\n",
    "* human samples\n",
    "* samples not sequenced at the Sanger\n",
    "\n",
    "## Use cases\n",
    "The section describes a set of use cases that must be satisfied for the MVP to be considered complete, together with a brief description of their status in the current version of the file tracking system at the time of writing. The appendix gives details of queries related to some of these use cases that can currently be performed in FITS at the time of writing.\n",
    "\n",
    "In the following, a build manifest is considered to be a file with one line per file, and to contain at the minimum:\n",
    "* iRODs path\n",
    "* Sample ID (Oxford code or ROMA ID)\n",
    "* Alfresco study code\n",
    "* NCBI taxon ID\n",
    "* Manual QC status\n",
    "* Date manual QC complete\n",
    "* ENA run accession\n",
    "* ENA sample accession\n",
    "\n",
    "Note that in the future, it is expected that information on Sample IDs, Alfresco study codes and taxon IDs will come from SIMS, but until SIMS is operational, this information will have to come from FITS, otherwise it will not be possible to use FITS for creating build manifests. The definitive source for file-to-sample, sample-to-study and sample-to-taxon mappings should be considered to be Solaris for samples contained in Solaris, or mlwh/iRODS for samples that are not. Note that some older files are included in Solaris but not in mlwh.\n",
    "\n",
    "For an example of a recently created build manifest, see https://github.com/wtsi-team112/Pv4/blob/master/notebooks/rp7/20180525_Pv4_manifest.ipynb\n",
    "\n",
    "### Create a build manifest containing all samples from a species\n",
    "See (how to build a manifest)[https://github.com/wtsi-team112/fits/blob/master/documentation//How_to_build_a_manifest.md].\n",
    "\n",
    "### Create a build manifest given a set of sequencescape IDs, oxford code/ROMA IDs or Alfresco study codes\n",
    "See (how to build a manifest)[https://github.com/wtsi-team112/fits/blob/master/documentation//How_to_build_a_manifest.md].\n",
    "\n",
    "### Determine which samples from a given study have been sequenced\n",
    "This is currently considered satisfied for the case of sequencescape study IDs. For more recent samples there should be a 1-to-1 mapping from sequencescape to Alfresco study codes, and therefore this use case can be considered satisfied going forwards.\n",
    "\n",
    "### Determine how many samples have been sequenced, broken down by species\n",
    "This is currently considered unsatisfied because taxon mappings from Solaris are not included\n",
    "\n",
    "\n",
    "### Populate FITS with mappings from Solaris\n",
    "Done.\n",
    "\n",
    "### Manually alter FITS mappings\n",
    "The key mappings that need to be captured are:\n",
    "- File-to-sample\n",
    "- Sample-to-taxon\n",
    "- Sample-to-alfresco_study_code\n",
    "The changes will need to be done in such a way that data can be overwritten or removed by later updates from mlwh/iRODS or elsewhere.\n",
    "\n",
    "### Update the file tracking database with the latest from Multi-LIMS warehouse/baton (iRODs)\n",
    "Could the way this is currently done with ./fits update_sanger?overwrite data from Solaris or manual edits? \n",
    ": No.\n",
    "\n",
    "## Other documents\n",
    "* Database description. Note that at the time of writing this requires the changes suggested by Richard in an email sent 28/06/2018 18:13\n",
    "* Command line utility description. These still needs to be written.\n",
    "* Process description for automatic and manual updating. This still needs to be written.\n",
    "\n",
    "## Relationship to SIMS\n",
    "For the moment, FITS and SIMS are developed separately. FITS will eventually pull information from SIMS via API, but there is no perceived need for information to flow in the other direction. FITS can match samples and files to SIMS UUIDs via high-level (e.g. ROMA/Oxford) or low-level (e.g. Sequenscape) IDs.\n",
    "\n",
    "## Future plans\n",
    "The following is a list of things that are considered to be out of scope for the MVP.\n",
    "\n",
    "### Sequencing not done at WSI\n",
    "FITS can store any file, given a unique path and a storage “engine”. Currently, the only engine supported is Sanger Sequencing iRODs, but more can be added without changing the database schema. Processes to import and update such data will have to be developed.\n",
    "\n",
    "The first use case here is likely to be storing ENA run accessions.\n",
    "\n",
    "### Expanding the scope to include amplicon sequencing data\n",
    "FITS is agnostic to file types. Storage engines such as “Team 112 iRODs” or “Sanger NFS” can be added. Processes to import and update such data will have to be developed.\n",
    "\n",
    "### Expanding the scope to include human data\n",
    "FITS is agnostic to species, however, storing of data about human samples might influence the storage location (US cloud?).\n",
    "\n",
    "### Expanding the scope to include pipeline outputs, e.g. release files\n",
    "FITS is agnostic to file types.  Processes to import and update such data will have to be developed.\n",
    "\n",
    "### Command line tool\n",
    "A command line tool exists. It can be used to update FITS from the  Multi-LIMS warehouse database. A query engine to ease generation of manifest files without compromising generic queries is under development.\n",
    "\n",
    "### Web front end\n",
    "A web frontend is not planned at this time, however, one could be developed rather easily.\n",
    "\n",
    "### Moving the system to a public cloud\n",
    "The FITS database currently resides at Sanger. For better interoperability with Oxford and/or cloud locations, a move to a cloud-based database system is planned once the MVP has stabilized. A database-as-a-service, rather than a generic server running a MySQL client, would be preferred, for ease of maintenance, backups, availability etc.\n",
    "Storage of human sample data might complicate finding an adequate cloud location.\n",
    "\n",
    "## Appendix - current status of use cases\n",
    "### Create a build manifest given a set of sequencescape IDs\n",
    "Possible with current version:\n",
    "```\n",
    "SELECT vw_sample_tag.value,full_path FROM vw_sample_tag,vw_sample_file WHERE tag_id=3585 AND vw_sample_tag.sample_id=vw_sample_file.sample_id AND `value` IN (list_of_sequenscape_IDs);\n",
    "```\n",
    "See (how to build a manifest)[https://github.com/wtsi-team112/fits/blob/master/documentation//How_to_build_a_manifest.md].\n",
    "\n",
    "### Create a build manifest given a set of Oxford codes and/or ROMA IDs\n",
    "Possible with current version, similar to above.\n",
    "\n",
    "### Create a build manifest given a set of Alfresco study codes\n",
    "The current version does not track Alfresco study codes. These can be imported, though a sample tracking system might be a more appropriate place for this information.\n",
    "\n",
    "### Create a build manifest containing all samples from a species\n",
    "Possible with current version; for “sample” meaning “Multi-LIMS warehouse sample ID”, and species defined as Sequenscape taxon ID for “Plasmodium falciparum”:\n",
    "```\n",
    "SELECT DISTINCT sample_id,(SELECT group_concat(DISTINCT `value`) FROM vw_sample_tag st2 WHERE tag_id IN (3589,3586) AND st1.sample_id=st2.sample_id) AS sample_name FROM vw_sample_tag st1 WHERE tag_id=3600 AND `value`='5833';\n",
    "```\n",
    "\n",
    "### Determine which samples from a given study have been sequenced\n",
    "Possible with current version; for “sample” meaning “Multi-LIMS warehouse sample ID”:\n",
    "```\n",
    "SELECT `value` AS sequenscape_study_id,count(*) AS cnt FROM vw_sample_tag WHERE tag_id=3593 GROUP BY `value` ORDER BY cnt DESC;\n",
    "```\n",
    "\n",
    "### Determine how many samples have been sequenced, broken down by species\n",
    "Possible with current version; for “sample” meaning “Multi-LIMS warehouse sample ID”:\n",
    "```\n",
    "SELECT `value`,count(*) AS cnt,(SELECT taxon_name FROM sequenscape_taxa WHERE taxon_id=`value`) AS name FROM vw_sample_tag WHERE tag_id=3600 GROUP BY `value` ORDER BY cnt DESC;\n",
    "```\n",
    "\n",
    "### Create a detailed BAM/CRAM manifest file with metadata, based on species common names\n",
    "Possible with current version; example for Plasmodium vivax V4 (this included a lot of filtering of unwanted files/samples):\n",
    "```\n",
    "SELECT file.id,file.full_path,\n",
    "(SELECT group_concat(value) FROM vw_sample_tag,vw_sample_file WHERE file.id=file_id AND vw_sample_file.sample_id=vw_sample_tag.sample_id AND tag_id=3561) AS ox_code,\n",
    "(SELECT group_concat(value) FROM vw_file_tag WHERE file.id=file_id AND tag_id=3591) AS common_name,\n",
    "(SELECT group_concat(value) FROM vw_sample_tag,vw_sample_file WHERE file.id=file_id AND vw_sample_file.sample_id=vw_sample_tag.sample_id AND tag_id=3600) AS taxon_code\n",
    "FROM file\n",
    "WHERE storage=1\n",
    "AND file.id IN (SELECT file_id FROM vw_file_tag WHERE tag_id=3591 AND value in ('Plasmodium vivax','vivax','P.vivax','Plasmodium Vvax','P. Vivax')) /*SPECIES NAMES*/\n",
    "AND full_path NOT LIKE \"%_human%\" AND full_path NOT LIKE \"%_phix%\" /*NOT HUMAN OR PHIX*/\n",
    "AND file.id NOT IN (SELECT parent FROM file_relation WHERE relation=3595) /*NOT IDENTICAL TO OTHER FILE*/\n",
    "AND EXISTS (SELECT * FROM vw_file_tag WHERE file.id=vw_file_tag.file_id AND tag_id=3576 AND value IN ('bam','cram')) /*FILE TYPE*/\n",
    "AND EXISTS (SELECT * FROM vw_file_tag WHERE file.id=vw_file_tag.file_id AND tag_id=3581 AND value=1) /*MANUAL QC*/\n",
    "AND NOT EXISTS (SELECT * FROM vw_file_tag WHERE file.id=vw_file_tag.file_id AND tag_id=3582 AND value=1) /*NO R&D*/\n",
    "```\n",
    "\n",
    "### Update the file tracking database with the latest from Multi-LIMS warehouse/baton (iRODs)\n",
    "A command-line interface software exists. The command to perform the above operation on Sanger systems is:\n",
    "./fits update_sanger\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
