{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "See https://github.com/wtsi-team112/fits/issues/4\n",
    "\n",
    "Magnus sent an email 17/09/2018 14:57 with 13 \"unexplained\" files that are in fits with sample ID of a sample in Pf6.2 manifest, but the file itself is not in Pf6.2 manifest. I suspect that these files might have been sequenced in \"invalid\" studies. I have considered a file to be an invalid study if, after mapping from sequencescape to alfresco studies using ```/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/sequencescape_alfresco_study_mappings.txt```, the Alfresco study code does not start with four integers. An example of a sequencescape study that fall in this category is study 1935 (\"AT-rich amplification\"). In this notebook, I determine which files have been excluded due to having an \"invalid\" study."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       ".container {\n",
       "    width: 100%;\n",
       "}\n",
       "# div#notebook {\n",
       "#     padding-top: 1em;\n",
       "# }\n",
       "# #header-container {\n",
       "#     display: none;\n",
       "# }\n",
       "# #header-bar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #maintoolbar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #menubar-container {\n",
       "#     position: fixed;\n",
       "#     margin-top: 0;\n",
       "# }\n",
       "site {\n",
       "    height: auto !important;\n",
       "}\n",
       "</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) \n",
      "[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]\n",
      "numpy 1.13.3\n",
      "scipy 0.19.1\n",
      "pandas 0.22.0\n",
      "numexpr 2.6.4\n",
      "pysam 0.8.4\n",
      "pysamstats 0.24.3\n",
      "petlx 1.0.3\n",
      "vcf 0.6.8\n",
      "h5py 2.7.0\n",
      "tables 3.4.2\n",
      "zarr 2.2.0\n",
      "scikit-allel 1.1.10\n"
     ]
    }
   ],
   "source": [
    "%run ../setup.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Inputs\n",
    "mlwh_missing_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_missing_exceptions.txt'\n",
    "mlwh_study_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_study_exceptions.txt'\n",
    "mlwh_sample_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_sample_exceptions.txt'\n",
    "mlwh_taxon_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_taxon_exceptions.txt'\n",
    "sequencescape_alfresco_study_mappings_fn='/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/sequencescape_alfresco_study_mappings.txt'\n",
    "\n",
    "# Outputs\n",
    "# output_fn = 'Pf_62_Pv4_like_manifest_20180917.txt'\n",
    "# manifest_symlink = 'pf_62_manifest.txt'\n",
    "input_dir = '/lustre/scratch118/malaria/team112/pipelines/setups/pf_62/input'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create assay data table for MalariaGEN samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def set_irods_name(row):\n",
    "    # First calculate prefix using nonhuman dependent on id_run and species (taxon)\n",
    "    # Then calculate tag string (empty string if no tag)\n",
    "    # Then calculate file extension\n",
    "    # Then concat these three strings\n",
    "    if (\n",
    "        (row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 6750)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    elif (\n",
    "        ~(row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 7100)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    else:\n",
    "        prefix = \"%d_%d\" % (row['id_run'], row['position'])\n",
    "        \n",
    "    if np.isnan(row['tag_index']):\n",
    "        tag_string = ''\n",
    "    else:\n",
    "        tag_string = '#%d' % row['tag_index']\n",
    "    \n",
    "    if row['id_run'] <= 10300:\n",
    "        file_extension = '.bam'\n",
    "    else:\n",
    "        file_extension = '.cram'\n",
    "    \n",
    "    irods_filename = prefix + tag_string + file_extension\n",
    "    return(irods_filename)\n",
    "\n",
    "def set_sample_id(row):\n",
    "    if row['sample'] is None:\n",
    "        return('DS_%s' % row['name'])\n",
    "    else:\n",
    "        return('DS_%s' % row['sample'])\n",
    "\n",
    "def is_valid_alfresco_code(s):\n",
    "    try: \n",
    "        int(s[0:4])\n",
    "        return True\n",
    "    except ValueError:\n",
    "        return False\n",
    "\n",
    "def create_build_manifest(\n",
    "#     taxon_ids=[5833, 5855, 5858, 5821, 7165, 7173, 30066], # Pf, Pv, Pm, Pb, Ag, A. arabiensis, A. merus\n",
    "    taxon_ids=[5833, 36329, 5847, 57267, 137071], # Pf, 3D7, V1, Dd2, HB3\n",
    "    # 1089 is excluded as this is R&D samples. 1204 and 1176 are excluded as these are CP1 samples. 1175 is excluded as some Pf R&D samples were incorrectly tagged with this study\n",
    "    studies_to_exclude = ['1089-R&amp;D-GB-ALCOCK', '1204-PF-GM-CP1', '1176-PF-KE-CP1', '1175-VO-KH-STLAURENT'],\n",
    "    miseq_runs_to_allow = [13809, 13810], # Two Miseq runs within 24 samples on each from study 1095-PF-TZ-ISHENGOMA that were included in Pf 6.0\n",
    "    input_dir = input_dir,\n",
    "    mlwh_missing_exceptions_fn = mlwh_missing_exceptions_fn,\n",
    "    mlwh_study_exceptions_fn = mlwh_study_exceptions_fn,\n",
    "    mlwh_sample_exceptions_fn = mlwh_sample_exceptions_fn,\n",
    "    mlwh_taxon_exceptions_fn = mlwh_taxon_exceptions_fn,\n",
    "    sequencescape_alfresco_study_mappings_fn = sequencescape_alfresco_study_mappings_fn,\n",
    "    output_columns = [\n",
    "        'path',\n",
    "        'study',\n",
    "        'sample',\n",
    "        'lane',\n",
    "        'reads',\n",
    "        'paired',\n",
    "        'irods_path',\n",
    "        'sanger_sample_id',\n",
    "        'taxon_id',\n",
    "        'study_lims',\n",
    "        'study_name',\n",
    "        'id_run',\n",
    "        'position',\n",
    "        'tag_index',\n",
    "        'qc_complete',\n",
    "        'manual_qc',\n",
    "        'description',\n",
    "        'instrument_name',\n",
    "        'instrument_model',\n",
    "        'forward_read_length',\n",
    "        'requested_insert_size_from',\n",
    "        'requested_insert_size_to',\n",
    "        'human_percent_mapped',\n",
    "        'subtrack_filename',\n",
    "        'subtrack_files_bytes',\n",
    "        'ebi_run_acc'\n",
    "    ]\n",
    "):\n",
    "    \"\"\"Create a DataFrame to be used as a build manifest.\n",
    "    \n",
    "    A build manifest here is a list of all the \"lanelets\" that need to be\n",
    "    included in a build. The output will typically be written to a\n",
    "    tab-delmited file, either to use as input to vr-pipe, or perhaps as\n",
    "    input to a vr-track DB.\n",
    "\n",
    "    Args:\n",
    "        taxon_ids (list of int): The taxons of the build (P. falciparum is '5833',\n",
    "            P. vivax is '5855').\n",
    "        sequencescape_alfresco_study_mappings_fn (str): filename of\n",
    "            mappings from sequencscape to alfresco study mappings\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The build manifest.\n",
    "\n",
    "    \"\"\"\n",
    "    \n",
    "    # Read in taxon exceptions file\n",
    "    df_mlwh_taxon_exceptions = pd.read_csv(mlwh_taxon_exceptions_fn, sep='\\t', dtype={'tag_index': str, 'taxon_id': int}, index_col='irods_filename')\n",
    "    df_mlwh_taxon_exceptions['tag_index'].fillna('', inplace=True)\n",
    "    # Identify which samples in exceptions file match the taxon of current build, and which don't\n",
    "    df_mlwh_exceptions_this_taxon = df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)]\n",
    "    df_mlwh_exceptions_other_taxa = df_mlwh_taxon_exceptions.loc[~(df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids))]\n",
    "\n",
    "    # Read in sample exceptions file\n",
    "    df_mlwh_sample_exceptions = pd.read_csv(mlwh_sample_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in study exceptions file\n",
    "    df_mlwh_study_exceptions = pd.read_csv(mlwh_study_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in missing exceptions file\n",
    "    df_mlwh_missing_exceptions = pd.read_csv(mlwh_missing_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "    df_mlwh_missing_exceptions['derivative_sample_id'] = 'DS_' + df_mlwh_missing_exceptions['sample_id']\n",
    "    df_mlwh_missing_exceptions = df_mlwh_missing_exceptions.loc[\n",
    "        df_mlwh_missing_exceptions['taxon_id'].isin(taxon_ids)\n",
    "    ]\n",
    "\n",
    "    # Read in SequenceScape-Alfresco study mappings\n",
    "    df_sequencescape_alfresco_study_mappings = pd.read_csv(sequencescape_alfresco_study_mappings_fn, sep='\\t', index_col='seqscape_study_id')\n",
    "    sequencescape_alfresco_study_mappings_dict = df_sequencescape_alfresco_study_mappings.loc[\n",
    "        :,\n",
    "#         df_sequencescape_alfresco_study_mappings['build_flag']==1,\n",
    "        'alfresco_study_code'\n",
    "    ].to_dict()\n",
    "    \n",
    "    # Read in data from mlwh matching this taxon, plus samples from exceptions file matching this taxon\n",
    "    conn = pymysql.connect(\n",
    "        host='mlwh-db',\n",
    "        user='mlwh_malaria',\n",
    "        password='Solaris&2015',\n",
    "        db='mlwarehouse',\n",
    "        port=3435\n",
    "    )\n",
    "    \n",
    "    sql_query = 'SELECT \\\n",
    "        study.name as study_name, \\\n",
    "        study.id_study_lims as study_lims, \\\n",
    "        sample.supplier_name as sample, \\\n",
    "        sample.name, \\\n",
    "        sample.sanger_sample_id, \\\n",
    "        sample.taxon_id, \\\n",
    "        iseq_product_metrics.id_run, \\\n",
    "        iseq_product_metrics.position, \\\n",
    "        iseq_product_metrics.tag_index, \\\n",
    "        iseq_product_metrics.num_reads, \\\n",
    "        iseq_product_metrics.human_percent_mapped, \\\n",
    "        iseq_run_lane_metrics.instrument_name, \\\n",
    "        iseq_run_lane_metrics.instrument_model, \\\n",
    "        iseq_run_lane_metrics.paired_read, \\\n",
    "        iseq_run_lane_metrics.qc_complete, \\\n",
    "        iseq_flowcell.manual_qc, \\\n",
    "        iseq_flowcell.requested_insert_size_from, \\\n",
    "        iseq_flowcell.requested_insert_size_to, \\\n",
    "        iseq_flowcell.forward_read_length, \\\n",
    "        iseq_run_status_dict.description \\\n",
    "    FROM \\\n",
    "        study, \\\n",
    "        iseq_flowcell, \\\n",
    "        sample, \\\n",
    "        iseq_product_metrics, \\\n",
    "        iseq_run_status, \\\n",
    "        iseq_run_lane_metrics, \\\n",
    "        iseq_run_status_dict \\\n",
    "    WHERE \\\n",
    "        study.id_study_tmp = iseq_flowcell.id_study_tmp and \\\n",
    "        iseq_flowcell.id_sample_tmp = sample.id_sample_tmp and \\\n",
    "        iseq_flowcell.manual_qc = 1 and \\\n",
    "        iseq_product_metrics.id_iseq_flowcell_tmp = iseq_flowcell.id_iseq_flowcell_tmp and \\\n",
    "        iseq_run_status.id_run = iseq_product_metrics.id_run and \\\n",
    "        iseq_product_metrics.id_run = iseq_run_lane_metrics.id_run and \\\n",
    "        iseq_product_metrics.position = iseq_run_lane_metrics.position and \\\n",
    "        iseq_run_status.iscurrent = 1 and \\\n",
    "        ( ( iseq_run_lane_metrics.instrument_model != \"MiSeq\" ) or (iseq_product_metrics.id_run in (%s)) ) and \\\n",
    "        iseq_run_status.id_run_status_dict = iseq_run_status_dict.id_run_status_dict and \\\n",
    "        study.faculty_sponsor = \"Dominic Kwiatkowski\" and \\\n",
    "        ( ( sample.taxon_id in (%s) ) or ' % (\n",
    "            ', '.join([str(x) for x in miseq_runs_to_allow]),\n",
    "            ', '.join([str(x) for x in taxon_ids])\n",
    "        )\n",
    "    sql_query = sql_query + ' or '.join(\n",
    "        df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)].apply(\n",
    "            lambda x: '(iseq_product_metrics.id_run=\"%s\" and iseq_product_metrics.position=\"%s\" and iseq_product_metrics.tag_index=\"%s\")' % (x['id_run'], x['position'], x['tag_index']),\n",
    "            1\n",
    "        )\n",
    "    )\n",
    "    sql_query = sql_query + ')'\n",
    "#         (iseq_flowcell.manual_qc = 1 or iseq_flowcell.manual_qc is null) and \\\n",
    "    \n",
    "    df_return = pd.read_sql(sql_query, conn)\n",
    "\n",
    "    # Replace missing taxon_id with -1 (can't have missing int values in pandas Series)\n",
    "    df_return['taxon_id'] = df_return['taxon_id'].fillna(-1).astype('int32')\n",
    "\n",
    "    # Determine file name in iRods\n",
    "    df_return['irods_filename'] = df_return.apply(set_irods_name, 1)\n",
    "    df_return.set_index('irods_filename', inplace=True)\n",
    "\n",
    "    # Determine alfresco study and change any incorrect studies\n",
    "    df_return['alfresco_study_code'] = df_return['study_lims'].apply(\n",
    "        lambda x: sequencescape_alfresco_study_mappings_dict[int(x)] if int(x) in sequencescape_alfresco_study_mappings_dict else ''\n",
    "    )\n",
    "    empty_alfresco_study_code = (df_return['alfresco_study_code'] == '')\n",
    "    df_return.loc[empty_alfresco_study_code, 'alfresco_study_code'] = df_return.loc[empty_alfresco_study_code, 'study_name']\n",
    "    \n",
    "    study_exception_indexes = df_mlwh_study_exceptions.index[\n",
    "        df_mlwh_study_exceptions.index.isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[study_exception_indexes, 'alfresco_study_code'] = df_mlwh_study_exceptions.loc[study_exception_indexes, 'alfresco_study_code']\n",
    "\n",
    "    # Change any incorrect taxon IDs\n",
    "    taxon_exception_indexes = df_mlwh_exceptions_this_taxon.index[\n",
    "        df_mlwh_exceptions_this_taxon.index.to_series().isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[taxon_exception_indexes, 'taxon_id'] = df_mlwh_exceptions_this_taxon.loc[taxon_exception_indexes, 'taxon_id']\n",
    "\n",
    "    # Determine sample ID and change any incorrect sample IDs\n",
    "    df_return['derivative_sample_id'] = df_return.apply(set_sample_id, 1)\n",
    "    sample_exception_indexes = df_mlwh_sample_exceptions.index[\n",
    "        df_mlwh_sample_exceptions.index.isin(df_return.index)\n",
    "    ]    \n",
    "    df_return.loc[sample_exception_indexes, 'derivative_sample_id'] = 'DS_' + df_mlwh_sample_exceptions.loc[sample_exception_indexes, 'sample_id']\n",
    "    df_return.drop(['sample', 'name'], axis=1, inplace=True)\n",
    "    \n",
    "    # Remove any samples that are not in this taxon\n",
    "    df_return = df_return.loc[~df_return.index.isin(df_mlwh_exceptions_other_taxa.index)]\n",
    "    \n",
    "    # Merge in missing exceptions\n",
    "    df_return = df_return.append(df_mlwh_missing_exceptions.drop('study_group', 1))\n",
    "    \n",
    "    # Remove any samples that don't have an alfresco study\n",
    "    invalid_codes = ~df_return['alfresco_study_code'].apply(is_valid_alfresco_code)\n",
    "    if(np.count_nonzero(invalid_codes) > 0):\n",
    "        print('Determined %d files with invalid alfresco_study_code:' % np.count_nonzero(invalid_codes))\n",
    "        print(df_return['alfresco_study_code'].loc[invalid_codes].value_counts())\n",
    "        df_return = df_return.loc[invalid_codes]\n",
    "    else:\n",
    "        print(\"No files with invalid alfresco_study_code\")\n",
    "\n",
    "    # Determine filename in subtrack (for very old samples, there is only a .srf file in subtrack)\n",
    "    # Note we determined the id_run cutoff (5750) for bam/srf after looking at previous runs. There seems\n",
    "    # to be a grey area between runs 5500 and 5750 where some files are srf and some are bam\n",
    "    df_return['subtrack_filename'] = df_return.index\n",
    "    df_return.loc[df_return['id_run'] < 5750, 'subtrack_filename'] = df_return.loc[df_return['id_run'] < 5750, 'subtrack_filename'].apply(lambda x: x.replace('.bam', '.srf'))\n",
    "    \n",
    "    # Read in data from substrack, most importantly to get run accessions\n",
    "    sql_query = \"\\\n",
    "    SELECT \\\n",
    "        files.file_name as irods_filename, \\\n",
    "        files.bytes as subtrack_files_bytes, \\\n",
    "        files.timestamp as subtrack_files_timestamp, \\\n",
    "        files.public_date as subtrack_files_public_date, \\\n",
    "        submission.id as subtrack_submission_id, \\\n",
    "        submission.created as subtrack_submission_created, \\\n",
    "        submission.release_date as subtrack_submission_release_date, \\\n",
    "        submission.timestamp as subtrack_submission_timestamp, \\\n",
    "        submission.ext_db as subtrack_submission_ext_db, \\\n",
    "        submission.ebi_sample_acc, \\\n",
    "        submission.ebi_exp_acc, \\\n",
    "        submission.ebi_study_acc, \\\n",
    "        submission.ebi_sub_acc, \\\n",
    "        submission.ebi_run_acc \\\n",
    "    FROM \\\n",
    "        submission, \\\n",
    "        files \\\n",
    "    WHERE \\\n",
    "        files.sub_id = submission.id AND \\\n",
    "        files.file_name in (%s)\\\n",
    "    \" % (\"'\" + \"', '\".join(df_return['subtrack_filename']) + \"'\")\n",
    "\n",
    "    conn= pymysql.connect(\n",
    "        host='shap',\n",
    "        user='ega_dataset',\n",
    "        password='ega_dataset',\n",
    "        db='subtrack',\n",
    "        port=3303\n",
    "    )\n",
    "\n",
    "    df_run_accessions = pd.read_sql(sql_query, conn).set_index('irods_filename')\n",
    "    \n",
    "    # Merge in subtrack data\n",
    "    df_return = df_return.join(df_run_accessions, on='subtrack_filename', how='left')\n",
    "\n",
    "    # Remove unwanted studies\n",
    "    df_return = df_return.loc[~df_return['alfresco_study_code'].isin(studies_to_exclude)]\n",
    "\n",
    "    # Create other columns\n",
    "    df_return.rename(columns={'alfresco_study_code': 'study', 'paired_read': 'paired'}, inplace=True)\n",
    "    df_return['lane'] = df_return.index.to_series().apply(lambda x: x.split('.')[0])\n",
    "    df_return['path'] = df_return.apply(lambda x: \"%s/%s\" % (input_dir, x.name), 1)\n",
    "    df_return['reads'] = df_return['num_reads'].fillna(-1).astype(int)\n",
    "    df_return['paired'] = 1\n",
    "    df_return['irods_path'] = df_return.apply(lambda x: \"/seq/%d/%s\" % (x['id_run'], x.name), 1)\n",
    "    df_return['sample'] = df_return['derivative_sample_id'].str.replace('DS_', '')\n",
    "\n",
    "    # Remove lanes with zero reads\n",
    "    df_return = df_return.loc[df_return['reads'] != 0]\n",
    "    \n",
    "    # Remove control samples (see email from Sonia 03/05/2018 12:20)\n",
    "    df_return = df_return.loc[df_return['sample'].str.slice(0, 1) != 'C']\n",
    "    \n",
    "    # Restrict to certain columns\n",
    "    if output_columns is not None:\n",
    "        df_return = (\n",
    "            df_return[output_columns]\n",
    "            .sort_values(['study', 'sample', 'id_run', 'position', 'tag_index'])\n",
    "        )\n",
    "    \n",
    "    print(df_return.shape)\n",
    "\n",
    "    return(df_return) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Determined 63 files with invalid alfresco_study_code:\n",
      "AT-rich amplification                                 53\n",
      "Test sequencing of high level human contamination.     6\n",
      "Plasmodium falciparum genome variation 1               4\n",
      "Name: alfresco_study_code, dtype: int64\n",
      "(63, 26)\n"
     ]
    }
   ],
   "source": [
    "df_lanelets_with_invalid_codes = create_build_manifest()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(63, 26)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>irods_path</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8002_4#7.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>3D7-MspJ-dig</td>\n",
       "      <td>8002_4#7</td>\n",
       "      <td>5719248</td>\n",
       "      <td>1</td>\n",
       "      <td>/seq/8002/8002_4#7.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>36329</td>\n",
       "      <td>1935</td>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>8002</td>\n",
       "      <td>4</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HS28</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>600.0</td>\n",
       "      <td>72.02</td>\n",
       "      <td>8002_4#7.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#4.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>3D7-no-dig</td>\n",
       "      <td>8002_4#4</td>\n",
       "      <td>92952</td>\n",
       "      <td>1</td>\n",
       "      <td>/seq/8002/8002_4#4.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>36329</td>\n",
       "      <td>1935</td>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>8002</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HS28</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>600.0</td>\n",
       "      <td>82.98</td>\n",
       "      <td>8002_4#4.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408_5_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>3D7_150</td>\n",
       "      <td>5408_5_nonhuman</td>\n",
       "      <td>25650602</td>\n",
       "      <td>1</td>\n",
       "      <td>/seq/5408/5408_5_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>36329</td>\n",
       "      <td>1935</td>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>5408</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2010-11-12 22:37:15</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL19</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>56.19</td>\n",
       "      <td>5408_5_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                  path  \\\n",
       "irods_filename                                                           \n",
       "8002_4#7.bam         /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "8002_4#4.bam         /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "5408_5_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "\n",
       "                                     study        sample             lane  \\\n",
       "irods_filename                                                              \n",
       "8002_4#7.bam         AT-rich amplification  3D7-MspJ-dig         8002_4#7   \n",
       "8002_4#4.bam         AT-rich amplification    3D7-no-dig         8002_4#4   \n",
       "5408_5_nonhuman.bam  AT-rich amplification       3D7_150  5408_5_nonhuman   \n",
       "\n",
       "                        reads  paired                     irods_path  \\\n",
       "irods_filename                                                         \n",
       "8002_4#7.bam          5719248       1         /seq/8002/8002_4#7.bam   \n",
       "8002_4#4.bam            92952       1         /seq/8002/8002_4#4.bam   \n",
       "5408_5_nonhuman.bam  25650602       1  /seq/5408/5408_5_nonhuman.bam   \n",
       "\n",
       "                    sanger_sample_id  taxon_id study_lims  \\\n",
       "irods_filename                                              \n",
       "8002_4#7.bam                    None     36329       1935   \n",
       "8002_4#4.bam                    None     36329       1935   \n",
       "5408_5_nonhuman.bam             None     36329       1935   \n",
       "\n",
       "                                study_name  id_run  position  tag_index  \\\n",
       "irods_filename                                                            \n",
       "8002_4#7.bam         AT-rich amplification    8002         4        7.0   \n",
       "8002_4#4.bam         AT-rich amplification    8002         4        4.0   \n",
       "5408_5_nonhuman.bam  AT-rich amplification    5408         5        NaN   \n",
       "\n",
       "                             qc_complete  manual_qc  description  \\\n",
       "irods_filename                                                     \n",
       "8002_4#7.bam         2012-06-11 13:52:53        1.0  qc complete   \n",
       "8002_4#4.bam         2012-06-11 13:52:53        1.0  qc complete   \n",
       "5408_5_nonhuman.bam  2010-11-12 22:37:15        1.0  qc complete   \n",
       "\n",
       "                    instrument_name instrument_model  forward_read_length  \\\n",
       "irods_filename                                                              \n",
       "8002_4#7.bam                   HS28            HiSeq                   75   \n",
       "8002_4#4.bam                   HS28            HiSeq                   75   \n",
       "5408_5_nonhuman.bam            IL19               HK                   76   \n",
       "\n",
       "                     requested_insert_size_from  requested_insert_size_to  \\\n",
       "irods_filename                                                              \n",
       "8002_4#7.bam                              150.0                     600.0   \n",
       "8002_4#4.bam                              150.0                     600.0   \n",
       "5408_5_nonhuman.bam                         0.0                       0.0   \n",
       "\n",
       "                     human_percent_mapped    subtrack_filename  \\\n",
       "irods_filename                                                   \n",
       "8002_4#7.bam                        72.02         8002_4#7.bam   \n",
       "8002_4#4.bam                        82.98         8002_4#4.bam   \n",
       "5408_5_nonhuman.bam                 56.19  5408_5_nonhuman.srf   \n",
       "\n",
       "                     subtrack_files_bytes ebi_run_acc  \n",
       "irods_filename                                         \n",
       "8002_4#7.bam                          NaN        None  \n",
       "8002_4#4.bam                          NaN        None  \n",
       "5408_5_nonhuman.bam                   NaN         NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_columns = 50\n",
    "print(df_lanelets_with_invalid_codes.shape)\n",
    "df_lanelets_with_invalid_codes[0:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AT-rich amplification                                 53\n",
       "Test sequencing of high level human contamination.     6\n",
       "Plasmodium falciparum genome variation 1               4\n",
       "Name: study, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_lanelets_with_invalid_codes['study'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df_output = df_lanelets_with_invalid_codes[\n",
    "    ['study', 'study_lims', 'sample', 'qc_complete', 'description', 'instrument_model', 'instrument_model', 'forward_read_length',\n",
    "     'requested_insert_size_from', 'ebi_run_acc', 'lane']\n",
    "].sort_values('lane')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>sample</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>lane</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12135_6#66.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PD0338-C</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR490344</td>\n",
       "      <td>12135_6#66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#67.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0196-C</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR490345</td>\n",
       "      <td>12135_7#67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#68.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0008-C</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR490346</td>\n",
       "      <td>12135_7#68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#69.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0145-C</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR490347</td>\n",
       "      <td>12135_7#69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#70.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0022-C</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR490348</td>\n",
       "      <td>12135_7#70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#71.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0212-C</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR490349</td>\n",
       "      <td>12135_7#71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3530_8_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK76-C_200</td>\n",
       "      <td>2009-08-26 00:56:41</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>150.0</td>\n",
       "      <td>ERR012749</td>\n",
       "      <td>3530_8_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408_2_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK0076</td>\n",
       "      <td>2010-11-12 22:37:15</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5408_2_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5408_5_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3D7_150</td>\n",
       "      <td>2010-11-12 22:37:15</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5408_5_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5535_3_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3d7-plt</td>\n",
       "      <td>2010-12-09 10:57:01</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5535_3_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5562_2_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3d7-ph</td>\n",
       "      <td>2010-12-21 18:55:00</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5562_2_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5598_6_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3d7_Dx_5</td>\n",
       "      <td>2010-12-22 07:07:36</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5598_6_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5598_7_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3d7_Dx_10</td>\n",
       "      <td>2010-12-22 07:07:36</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5598_7_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5701_8_nonhuman#1.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK76-Plt-1</td>\n",
       "      <td>2011-02-02 21:17:28</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5701_8_nonhuman#1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5701_8_nonhuman#2.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK76-Kapa-2</td>\n",
       "      <td>2011-02-02 21:17:28</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5701_8_nonhuman#2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5701_8_nonhuman#3.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK76-Phus-3</td>\n",
       "      <td>2011-02-02 21:17:28</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5701_8_nonhuman#3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5701_8_nonhuman#4.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK76-Dx-4</td>\n",
       "      <td>2011-02-02 21:17:28</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5701_8_nonhuman#4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5701_8_nonhuman#5.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK76-plt-notmac-5</td>\n",
       "      <td>2011-02-02 21:17:28</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5701_8_nonhuman#5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5701_8_nonhuman#6.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Pk76-kap-notmac-6</td>\n",
       "      <td>2011-02-02 21:17:28</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5701_8_nonhuman#6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5704_1_nonhuman.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Plt-no-tmac</td>\n",
       "      <td>2011-02-12 10:35:37</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5704_1_nonhuman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5756_8_nonhuman#11.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3d7-FZ-11</td>\n",
       "      <td>2011-02-16 22:40:22</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR043381</td>\n",
       "      <td>5756_8_nonhuman#11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5756_8_nonhuman#12.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3d7-NF-12</td>\n",
       "      <td>2011-02-16 22:40:22</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR043382</td>\n",
       "      <td>5756_8_nonhuman#12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5928_8_nonhuman#1.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Plt-3d7-no-tmac</td>\n",
       "      <td>2011-03-19 21:39:31</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>None</td>\n",
       "      <td>5928_8_nonhuman#1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5928_8_nonhuman#2.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Accu-3d7-no-tmac</td>\n",
       "      <td>2011-03-19 21:39:31</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047177</td>\n",
       "      <td>5928_8_nonhuman#2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5928_8_nonhuman#3.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapR-3d7-no-tmac</td>\n",
       "      <td>2011-03-19 21:39:31</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047178</td>\n",
       "      <td>5928_8_nonhuman#3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5928_8_nonhuman#4.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapHifi-3d7-no-tmac</td>\n",
       "      <td>2011-03-19 21:39:31</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047179</td>\n",
       "      <td>5928_8_nonhuman#4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5941_1_nonhuman#1.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Plt-PK76-no-tmc</td>\n",
       "      <td>2011-03-28 09:32:22</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>None</td>\n",
       "      <td>5941_1_nonhuman#1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5941_1_nonhuman#2.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Accu-PK76-no-tmc</td>\n",
       "      <td>2011-03-28 09:32:22</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>None</td>\n",
       "      <td>5941_1_nonhuman#2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5941_1_nonhuman#3.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapR-PK76-no-tmc</td>\n",
       "      <td>2011-03-28 09:32:22</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>None</td>\n",
       "      <td>5941_1_nonhuman#3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5941_1_nonhuman#4.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapHifi-PK76-no-tmc</td>\n",
       "      <td>2011-03-28 09:32:22</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>None</td>\n",
       "      <td>5941_1_nonhuman#4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_3_nonhuman#1.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Phus_3d7</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047184</td>\n",
       "      <td>6346_3_nonhuman#1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_3_nonhuman#2.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Accu_3d7</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047185</td>\n",
       "      <td>6346_3_nonhuman#2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_3_nonhuman#3.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapR_3d7</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047186</td>\n",
       "      <td>6346_3_nonhuman#3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_3_nonhuman#4.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapHifi_3d7</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047187</td>\n",
       "      <td>6346_3_nonhuman#4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_5_nonhuman#1.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Phus_PK76</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047188</td>\n",
       "      <td>6346_5_nonhuman#1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_5_nonhuman#2.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>Accu_PK76</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047189</td>\n",
       "      <td>6346_5_nonhuman#2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_5_nonhuman#3.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapR_PK76</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047190</td>\n",
       "      <td>6346_5_nonhuman#3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6346_5_nonhuman#4.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>KapHifi_PK76</td>\n",
       "      <td>2011-06-15 12:57:52</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HK</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>100.0</td>\n",
       "      <td>ERR047191</td>\n",
       "      <td>6346_5_nonhuman#4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#10.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>Tr5E_1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR205946</td>\n",
       "      <td>6901_8_nonhuman#10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#11.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>Tr3E</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR205947</td>\n",
       "      <td>6901_8_nonhuman#11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#2.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>Tr5E</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR205938</td>\n",
       "      <td>6901_8_nonhuman#2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#3.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>Tr3E_1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>200.0</td>\n",
       "      <td>ERR205939</td>\n",
       "      <td>6901_8_nonhuman#3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#1.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>90Hs-3D7-plus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#10.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PM56-minus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#11.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PM65-minus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#12.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PM116-minus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#4.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PM56-plus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#5.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PM65-plus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#6.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PM116-plus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7560_8#7.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>90Hs-3D7-minus</td>\n",
       "      <td>2012-03-01 17:33:07</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>7560_8#7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#1.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>90Hs-3D7-Spri-1</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#2.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>90Hs-3D7-Spri-2</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#21.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PF312-MspJI</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#22.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PH579-MspJI</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#23.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PH603-MspJI</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#3.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>90Hs-3D7-Gel-no-Act</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#4.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3D7-no-dig</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#5.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>90Hs-3D7-gel-plus-Act</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#6.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>PK76-GE-WGA</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8002_4#7.bam</th>\n",
       "      <td>AT-rich amplification</td>\n",
       "      <td>1935</td>\n",
       "      <td>3D7-MspJ-dig</td>\n",
       "      <td>2012-06-11 13:52:53</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>150.0</td>\n",
       "      <td>None</td>\n",
       "      <td>8002_4#7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>63 rows  11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                    study  \\\n",
       "irods_filename                                                              \n",
       "12135_6#66.cram         Test sequencing of high level human contaminat...   \n",
       "12135_7#67.cram         Test sequencing of high level human contaminat...   \n",
       "12135_7#68.cram         Test sequencing of high level human contaminat...   \n",
       "12135_7#69.cram         Test sequencing of high level human contaminat...   \n",
       "12135_7#70.cram         Test sequencing of high level human contaminat...   \n",
       "12135_7#71.cram         Test sequencing of high level human contaminat...   \n",
       "3530_8_nonhuman.bam                                 AT-rich amplification   \n",
       "5408_2_nonhuman.bam                                 AT-rich amplification   \n",
       "5408_5_nonhuman.bam                                 AT-rich amplification   \n",
       "5535_3_nonhuman.bam                                 AT-rich amplification   \n",
       "5562_2_nonhuman.bam                                 AT-rich amplification   \n",
       "5598_6_nonhuman.bam                                 AT-rich amplification   \n",
       "5598_7_nonhuman.bam                                 AT-rich amplification   \n",
       "5701_8_nonhuman#1.bam                               AT-rich amplification   \n",
       "5701_8_nonhuman#2.bam                               AT-rich amplification   \n",
       "5701_8_nonhuman#3.bam                               AT-rich amplification   \n",
       "5701_8_nonhuman#4.bam                               AT-rich amplification   \n",
       "5701_8_nonhuman#5.bam                               AT-rich amplification   \n",
       "5701_8_nonhuman#6.bam                               AT-rich amplification   \n",
       "5704_1_nonhuman.bam                                 AT-rich amplification   \n",
       "5756_8_nonhuman#11.bam                              AT-rich amplification   \n",
       "5756_8_nonhuman#12.bam                              AT-rich amplification   \n",
       "5928_8_nonhuman#1.bam                               AT-rich amplification   \n",
       "5928_8_nonhuman#2.bam                               AT-rich amplification   \n",
       "5928_8_nonhuman#3.bam                               AT-rich amplification   \n",
       "5928_8_nonhuman#4.bam                               AT-rich amplification   \n",
       "5941_1_nonhuman#1.bam                               AT-rich amplification   \n",
       "5941_1_nonhuman#2.bam                               AT-rich amplification   \n",
       "5941_1_nonhuman#3.bam                               AT-rich amplification   \n",
       "5941_1_nonhuman#4.bam                               AT-rich amplification   \n",
       "...                                                                   ...   \n",
       "6346_3_nonhuman#1.bam                               AT-rich amplification   \n",
       "6346_3_nonhuman#2.bam                               AT-rich amplification   \n",
       "6346_3_nonhuman#3.bam                               AT-rich amplification   \n",
       "6346_3_nonhuman#4.bam                               AT-rich amplification   \n",
       "6346_5_nonhuman#1.bam                               AT-rich amplification   \n",
       "6346_5_nonhuman#2.bam                               AT-rich amplification   \n",
       "6346_5_nonhuman#3.bam                               AT-rich amplification   \n",
       "6346_5_nonhuman#4.bam                               AT-rich amplification   \n",
       "6901_8_nonhuman#10.bam           Plasmodium falciparum genome variation 1   \n",
       "6901_8_nonhuman#11.bam           Plasmodium falciparum genome variation 1   \n",
       "6901_8_nonhuman#2.bam            Plasmodium falciparum genome variation 1   \n",
       "6901_8_nonhuman#3.bam            Plasmodium falciparum genome variation 1   \n",
       "7560_8#1.bam                                        AT-rich amplification   \n",
       "7560_8#10.bam                                       AT-rich amplification   \n",
       "7560_8#11.bam                                       AT-rich amplification   \n",
       "7560_8#12.bam                                       AT-rich amplification   \n",
       "7560_8#4.bam                                        AT-rich amplification   \n",
       "7560_8#5.bam                                        AT-rich amplification   \n",
       "7560_8#6.bam                                        AT-rich amplification   \n",
       "7560_8#7.bam                                        AT-rich amplification   \n",
       "8002_4#1.bam                                        AT-rich amplification   \n",
       "8002_4#2.bam                                        AT-rich amplification   \n",
       "8002_4#21.bam                                       AT-rich amplification   \n",
       "8002_4#22.bam                                       AT-rich amplification   \n",
       "8002_4#23.bam                                       AT-rich amplification   \n",
       "8002_4#3.bam                                        AT-rich amplification   \n",
       "8002_4#4.bam                                        AT-rich amplification   \n",
       "8002_4#5.bam                                        AT-rich amplification   \n",
       "8002_4#6.bam                                        AT-rich amplification   \n",
       "8002_4#7.bam                                        AT-rich amplification   \n",
       "\n",
       "                       study_lims                 sample          qc_complete  \\\n",
       "irods_filename                                                                  \n",
       "12135_6#66.cram              2876               PD0338-C  2014-02-28 12:53:57   \n",
       "12135_7#67.cram              2876               QE0196-C  2014-02-28 12:53:57   \n",
       "12135_7#68.cram              2876               QE0008-C  2014-02-28 12:53:57   \n",
       "12135_7#69.cram              2876               QE0145-C  2014-02-28 12:53:57   \n",
       "12135_7#70.cram              2876               QE0022-C  2014-02-28 12:53:57   \n",
       "12135_7#71.cram              2876               QE0212-C  2014-02-28 12:53:57   \n",
       "3530_8_nonhuman.bam          1935             PK76-C_200  2009-08-26 00:56:41   \n",
       "5408_2_nonhuman.bam          1935                 PK0076  2010-11-12 22:37:15   \n",
       "5408_5_nonhuman.bam          1935                3D7_150  2010-11-12 22:37:15   \n",
       "5535_3_nonhuman.bam          1935                3d7-plt  2010-12-09 10:57:01   \n",
       "5562_2_nonhuman.bam          1935                 3d7-ph  2010-12-21 18:55:00   \n",
       "5598_6_nonhuman.bam          1935               3d7_Dx_5  2010-12-22 07:07:36   \n",
       "5598_7_nonhuman.bam          1935              3d7_Dx_10  2010-12-22 07:07:36   \n",
       "5701_8_nonhuman#1.bam        1935             PK76-Plt-1  2011-02-02 21:17:28   \n",
       "5701_8_nonhuman#2.bam        1935            PK76-Kapa-2  2011-02-02 21:17:28   \n",
       "5701_8_nonhuman#3.bam        1935            PK76-Phus-3  2011-02-02 21:17:28   \n",
       "5701_8_nonhuman#4.bam        1935              PK76-Dx-4  2011-02-02 21:17:28   \n",
       "5701_8_nonhuman#5.bam        1935      PK76-plt-notmac-5  2011-02-02 21:17:28   \n",
       "5701_8_nonhuman#6.bam        1935      Pk76-kap-notmac-6  2011-02-02 21:17:28   \n",
       "5704_1_nonhuman.bam          1935            Plt-no-tmac  2011-02-12 10:35:37   \n",
       "5756_8_nonhuman#11.bam       1935              3d7-FZ-11  2011-02-16 22:40:22   \n",
       "5756_8_nonhuman#12.bam       1935              3d7-NF-12  2011-02-16 22:40:22   \n",
       "5928_8_nonhuman#1.bam        1935        Plt-3d7-no-tmac  2011-03-19 21:39:31   \n",
       "5928_8_nonhuman#2.bam        1935       Accu-3d7-no-tmac  2011-03-19 21:39:31   \n",
       "5928_8_nonhuman#3.bam        1935       KapR-3d7-no-tmac  2011-03-19 21:39:31   \n",
       "5928_8_nonhuman#4.bam        1935    KapHifi-3d7-no-tmac  2011-03-19 21:39:31   \n",
       "5941_1_nonhuman#1.bam        1935        Plt-PK76-no-tmc  2011-03-28 09:32:22   \n",
       "5941_1_nonhuman#2.bam        1935       Accu-PK76-no-tmc  2011-03-28 09:32:22   \n",
       "5941_1_nonhuman#3.bam        1935       KapR-PK76-no-tmc  2011-03-28 09:32:22   \n",
       "5941_1_nonhuman#4.bam        1935    KapHifi-PK76-no-tmc  2011-03-28 09:32:22   \n",
       "...                           ...                    ...                  ...   \n",
       "6346_3_nonhuman#1.bam        1935               Phus_3d7  2011-06-15 12:57:52   \n",
       "6346_3_nonhuman#2.bam        1935               Accu_3d7  2011-06-15 12:57:52   \n",
       "6346_3_nonhuman#3.bam        1935               KapR_3d7  2011-06-15 12:57:52   \n",
       "6346_3_nonhuman#4.bam        1935            KapHifi_3d7  2011-06-15 12:57:52   \n",
       "6346_5_nonhuman#1.bam        1935              Phus_PK76  2011-06-15 12:57:52   \n",
       "6346_5_nonhuman#2.bam        1935              Accu_PK76  2011-06-15 12:57:52   \n",
       "6346_5_nonhuman#3.bam        1935              KapR_PK76  2011-06-15 12:57:52   \n",
       "6346_5_nonhuman#4.bam        1935           KapHifi_PK76  2011-06-15 12:57:52   \n",
       "6901_8_nonhuman#10.bam        578                 Tr5E_1  2011-11-01 12:52:25   \n",
       "6901_8_nonhuman#11.bam        578                   Tr3E  2011-11-01 12:52:25   \n",
       "6901_8_nonhuman#2.bam         578                   Tr5E  2011-11-01 12:52:25   \n",
       "6901_8_nonhuman#3.bam         578                 Tr3E_1  2011-11-01 12:52:25   \n",
       "7560_8#1.bam                 1935          90Hs-3D7-plus  2012-03-01 17:33:07   \n",
       "7560_8#10.bam                1935             PM56-minus  2012-03-01 17:33:07   \n",
       "7560_8#11.bam                1935             PM65-minus  2012-03-01 17:33:07   \n",
       "7560_8#12.bam                1935            PM116-minus  2012-03-01 17:33:07   \n",
       "7560_8#4.bam                 1935              PM56-plus  2012-03-01 17:33:07   \n",
       "7560_8#5.bam                 1935              PM65-plus  2012-03-01 17:33:07   \n",
       "7560_8#6.bam                 1935             PM116-plus  2012-03-01 17:33:07   \n",
       "7560_8#7.bam                 1935         90Hs-3D7-minus  2012-03-01 17:33:07   \n",
       "8002_4#1.bam                 1935        90Hs-3D7-Spri-1  2012-06-11 13:52:53   \n",
       "8002_4#2.bam                 1935        90Hs-3D7-Spri-2  2012-06-11 13:52:53   \n",
       "8002_4#21.bam                1935            PF312-MspJI  2012-06-11 13:52:53   \n",
       "8002_4#22.bam                1935            PH579-MspJI  2012-06-11 13:52:53   \n",
       "8002_4#23.bam                1935            PH603-MspJI  2012-06-11 13:52:53   \n",
       "8002_4#3.bam                 1935    90Hs-3D7-Gel-no-Act  2012-06-11 13:52:53   \n",
       "8002_4#4.bam                 1935             3D7-no-dig  2012-06-11 13:52:53   \n",
       "8002_4#5.bam                 1935  90Hs-3D7-gel-plus-Act  2012-06-11 13:52:53   \n",
       "8002_4#6.bam                 1935            PK76-GE-WGA  2012-06-11 13:52:53   \n",
       "8002_4#7.bam                 1935           3D7-MspJ-dig  2012-06-11 13:52:53   \n",
       "\n",
       "                        description instrument_model instrument_model  \\\n",
       "irods_filename                                                          \n",
       "12135_6#66.cram         qc complete            HiSeq            HiSeq   \n",
       "12135_7#67.cram         qc complete            HiSeq            HiSeq   \n",
       "12135_7#68.cram         qc complete            HiSeq            HiSeq   \n",
       "12135_7#69.cram         qc complete            HiSeq            HiSeq   \n",
       "12135_7#70.cram         qc complete            HiSeq            HiSeq   \n",
       "12135_7#71.cram         qc complete            HiSeq            HiSeq   \n",
       "3530_8_nonhuman.bam     qc complete               HK               HK   \n",
       "5408_2_nonhuman.bam     qc complete               HK               HK   \n",
       "5408_5_nonhuman.bam     qc complete               HK               HK   \n",
       "5535_3_nonhuman.bam     qc complete               HK               HK   \n",
       "5562_2_nonhuman.bam     qc complete               HK               HK   \n",
       "5598_6_nonhuman.bam     qc complete               HK               HK   \n",
       "5598_7_nonhuman.bam     qc complete               HK               HK   \n",
       "5701_8_nonhuman#1.bam   qc complete               HK               HK   \n",
       "5701_8_nonhuman#2.bam   qc complete               HK               HK   \n",
       "5701_8_nonhuman#3.bam   qc complete               HK               HK   \n",
       "5701_8_nonhuman#4.bam   qc complete               HK               HK   \n",
       "5701_8_nonhuman#5.bam   qc complete               HK               HK   \n",
       "5701_8_nonhuman#6.bam   qc complete               HK               HK   \n",
       "5704_1_nonhuman.bam     qc complete               HK               HK   \n",
       "5756_8_nonhuman#11.bam  qc complete               HK               HK   \n",
       "5756_8_nonhuman#12.bam  qc complete               HK               HK   \n",
       "5928_8_nonhuman#1.bam   qc complete               HK               HK   \n",
       "5928_8_nonhuman#2.bam   qc complete               HK               HK   \n",
       "5928_8_nonhuman#3.bam   qc complete               HK               HK   \n",
       "5928_8_nonhuman#4.bam   qc complete               HK               HK   \n",
       "5941_1_nonhuman#1.bam   qc complete               HK               HK   \n",
       "5941_1_nonhuman#2.bam   qc complete               HK               HK   \n",
       "5941_1_nonhuman#3.bam   qc complete               HK               HK   \n",
       "5941_1_nonhuman#4.bam   qc complete               HK               HK   \n",
       "...                             ...              ...              ...   \n",
       "6346_3_nonhuman#1.bam   qc complete               HK               HK   \n",
       "6346_3_nonhuman#2.bam   qc complete               HK               HK   \n",
       "6346_3_nonhuman#3.bam   qc complete               HK               HK   \n",
       "6346_3_nonhuman#4.bam   qc complete               HK               HK   \n",
       "6346_5_nonhuman#1.bam   qc complete               HK               HK   \n",
       "6346_5_nonhuman#2.bam   qc complete               HK               HK   \n",
       "6346_5_nonhuman#3.bam   qc complete               HK               HK   \n",
       "6346_5_nonhuman#4.bam   qc complete               HK               HK   \n",
       "6901_8_nonhuman#10.bam  qc complete            HiSeq            HiSeq   \n",
       "6901_8_nonhuman#11.bam  qc complete            HiSeq            HiSeq   \n",
       "6901_8_nonhuman#2.bam   qc complete            HiSeq            HiSeq   \n",
       "6901_8_nonhuman#3.bam   qc complete            HiSeq            HiSeq   \n",
       "7560_8#1.bam            qc complete            HiSeq            HiSeq   \n",
       "7560_8#10.bam           qc complete            HiSeq            HiSeq   \n",
       "7560_8#11.bam           qc complete            HiSeq            HiSeq   \n",
       "7560_8#12.bam           qc complete            HiSeq            HiSeq   \n",
       "7560_8#4.bam            qc complete            HiSeq            HiSeq   \n",
       "7560_8#5.bam            qc complete            HiSeq            HiSeq   \n",
       "7560_8#6.bam            qc complete            HiSeq            HiSeq   \n",
       "7560_8#7.bam            qc complete            HiSeq            HiSeq   \n",
       "8002_4#1.bam            qc complete            HiSeq            HiSeq   \n",
       "8002_4#2.bam            qc complete            HiSeq            HiSeq   \n",
       "8002_4#21.bam           qc complete            HiSeq            HiSeq   \n",
       "8002_4#22.bam           qc complete            HiSeq            HiSeq   \n",
       "8002_4#23.bam           qc complete            HiSeq            HiSeq   \n",
       "8002_4#3.bam            qc complete            HiSeq            HiSeq   \n",
       "8002_4#4.bam            qc complete            HiSeq            HiSeq   \n",
       "8002_4#5.bam            qc complete            HiSeq            HiSeq   \n",
       "8002_4#6.bam            qc complete            HiSeq            HiSeq   \n",
       "8002_4#7.bam            qc complete            HiSeq            HiSeq   \n",
       "\n",
       "                        forward_read_length  requested_insert_size_from  \\\n",
       "irods_filename                                                            \n",
       "12135_6#66.cram                         100                       200.0   \n",
       "12135_7#67.cram                         100                       200.0   \n",
       "12135_7#68.cram                         100                       200.0   \n",
       "12135_7#69.cram                         100                       200.0   \n",
       "12135_7#70.cram                         100                       200.0   \n",
       "12135_7#71.cram                         100                       200.0   \n",
       "3530_8_nonhuman.bam                      76                       150.0   \n",
       "5408_2_nonhuman.bam                      76                         0.0   \n",
       "5408_5_nonhuman.bam                      76                         0.0   \n",
       "5535_3_nonhuman.bam                      76                       200.0   \n",
       "5562_2_nonhuman.bam                      76                       200.0   \n",
       "5598_6_nonhuman.bam                      76                       200.0   \n",
       "5598_7_nonhuman.bam                      76                       200.0   \n",
       "5701_8_nonhuman#1.bam                    76                       100.0   \n",
       "5701_8_nonhuman#2.bam                    76                       100.0   \n",
       "5701_8_nonhuman#3.bam                    76                       100.0   \n",
       "5701_8_nonhuman#4.bam                    76                       100.0   \n",
       "5701_8_nonhuman#5.bam                    76                       100.0   \n",
       "5701_8_nonhuman#6.bam                    76                       100.0   \n",
       "5704_1_nonhuman.bam                      76                       100.0   \n",
       "5756_8_nonhuman#11.bam                   76                       200.0   \n",
       "5756_8_nonhuman#12.bam                   76                       200.0   \n",
       "5928_8_nonhuman#1.bam                    76                       100.0   \n",
       "5928_8_nonhuman#2.bam                    76                       100.0   \n",
       "5928_8_nonhuman#3.bam                    76                       100.0   \n",
       "5928_8_nonhuman#4.bam                    76                       100.0   \n",
       "5941_1_nonhuman#1.bam                    76                       100.0   \n",
       "5941_1_nonhuman#2.bam                    76                       100.0   \n",
       "5941_1_nonhuman#3.bam                    76                       100.0   \n",
       "5941_1_nonhuman#4.bam                    76                       100.0   \n",
       "...                                     ...                         ...   \n",
       "6346_3_nonhuman#1.bam                    76                       100.0   \n",
       "6346_3_nonhuman#2.bam                    76                       100.0   \n",
       "6346_3_nonhuman#3.bam                    76                       100.0   \n",
       "6346_3_nonhuman#4.bam                    76                       100.0   \n",
       "6346_5_nonhuman#1.bam                    76                       100.0   \n",
       "6346_5_nonhuman#2.bam                    76                       100.0   \n",
       "6346_5_nonhuman#3.bam                    76                       100.0   \n",
       "6346_5_nonhuman#4.bam                    76                       100.0   \n",
       "6901_8_nonhuman#10.bam                  100                       200.0   \n",
       "6901_8_nonhuman#11.bam                  100                       200.0   \n",
       "6901_8_nonhuman#2.bam                   100                       200.0   \n",
       "6901_8_nonhuman#3.bam                   100                       200.0   \n",
       "7560_8#1.bam                             75                       150.0   \n",
       "7560_8#10.bam                            75                       150.0   \n",
       "7560_8#11.bam                            75                       150.0   \n",
       "7560_8#12.bam                            75                       150.0   \n",
       "7560_8#4.bam                             75                       150.0   \n",
       "7560_8#5.bam                             75                       150.0   \n",
       "7560_8#6.bam                             75                       150.0   \n",
       "7560_8#7.bam                             75                       150.0   \n",
       "8002_4#1.bam                             75                       150.0   \n",
       "8002_4#2.bam                             75                       150.0   \n",
       "8002_4#21.bam                            75                       150.0   \n",
       "8002_4#22.bam                            75                       150.0   \n",
       "8002_4#23.bam                            75                       150.0   \n",
       "8002_4#3.bam                             75                       150.0   \n",
       "8002_4#4.bam                             75                       150.0   \n",
       "8002_4#5.bam                             75                       150.0   \n",
       "8002_4#6.bam                             75                       150.0   \n",
       "8002_4#7.bam                             75                       150.0   \n",
       "\n",
       "                       ebi_run_acc                lane  \n",
       "irods_filename                                          \n",
       "12135_6#66.cram          ERR490344          12135_6#66  \n",
       "12135_7#67.cram          ERR490345          12135_7#67  \n",
       "12135_7#68.cram          ERR490346          12135_7#68  \n",
       "12135_7#69.cram          ERR490347          12135_7#69  \n",
       "12135_7#70.cram          ERR490348          12135_7#70  \n",
       "12135_7#71.cram          ERR490349          12135_7#71  \n",
       "3530_8_nonhuman.bam      ERR012749     3530_8_nonhuman  \n",
       "5408_2_nonhuman.bam            NaN     5408_2_nonhuman  \n",
       "5408_5_nonhuman.bam            NaN     5408_5_nonhuman  \n",
       "5535_3_nonhuman.bam            NaN     5535_3_nonhuman  \n",
       "5562_2_nonhuman.bam            NaN     5562_2_nonhuman  \n",
       "5598_6_nonhuman.bam            NaN     5598_6_nonhuman  \n",
       "5598_7_nonhuman.bam            NaN     5598_7_nonhuman  \n",
       "5701_8_nonhuman#1.bam          NaN   5701_8_nonhuman#1  \n",
       "5701_8_nonhuman#2.bam          NaN   5701_8_nonhuman#2  \n",
       "5701_8_nonhuman#3.bam          NaN   5701_8_nonhuman#3  \n",
       "5701_8_nonhuman#4.bam          NaN   5701_8_nonhuman#4  \n",
       "5701_8_nonhuman#5.bam          NaN   5701_8_nonhuman#5  \n",
       "5701_8_nonhuman#6.bam          NaN   5701_8_nonhuman#6  \n",
       "5704_1_nonhuman.bam            NaN     5704_1_nonhuman  \n",
       "5756_8_nonhuman#11.bam   ERR043381  5756_8_nonhuman#11  \n",
       "5756_8_nonhuman#12.bam   ERR043382  5756_8_nonhuman#12  \n",
       "5928_8_nonhuman#1.bam         None   5928_8_nonhuman#1  \n",
       "5928_8_nonhuman#2.bam    ERR047177   5928_8_nonhuman#2  \n",
       "5928_8_nonhuman#3.bam    ERR047178   5928_8_nonhuman#3  \n",
       "5928_8_nonhuman#4.bam    ERR047179   5928_8_nonhuman#4  \n",
       "5941_1_nonhuman#1.bam         None   5941_1_nonhuman#1  \n",
       "5941_1_nonhuman#2.bam         None   5941_1_nonhuman#2  \n",
       "5941_1_nonhuman#3.bam         None   5941_1_nonhuman#3  \n",
       "5941_1_nonhuman#4.bam         None   5941_1_nonhuman#4  \n",
       "...                            ...                 ...  \n",
       "6346_3_nonhuman#1.bam    ERR047184   6346_3_nonhuman#1  \n",
       "6346_3_nonhuman#2.bam    ERR047185   6346_3_nonhuman#2  \n",
       "6346_3_nonhuman#3.bam    ERR047186   6346_3_nonhuman#3  \n",
       "6346_3_nonhuman#4.bam    ERR047187   6346_3_nonhuman#4  \n",
       "6346_5_nonhuman#1.bam    ERR047188   6346_5_nonhuman#1  \n",
       "6346_5_nonhuman#2.bam    ERR047189   6346_5_nonhuman#2  \n",
       "6346_5_nonhuman#3.bam    ERR047190   6346_5_nonhuman#3  \n",
       "6346_5_nonhuman#4.bam    ERR047191   6346_5_nonhuman#4  \n",
       "6901_8_nonhuman#10.bam   ERR205946  6901_8_nonhuman#10  \n",
       "6901_8_nonhuman#11.bam   ERR205947  6901_8_nonhuman#11  \n",
       "6901_8_nonhuman#2.bam    ERR205938   6901_8_nonhuman#2  \n",
       "6901_8_nonhuman#3.bam    ERR205939   6901_8_nonhuman#3  \n",
       "7560_8#1.bam                  None            7560_8#1  \n",
       "7560_8#10.bam                 None           7560_8#10  \n",
       "7560_8#11.bam                 None           7560_8#11  \n",
       "7560_8#12.bam                 None           7560_8#12  \n",
       "7560_8#4.bam                  None            7560_8#4  \n",
       "7560_8#5.bam                  None            7560_8#5  \n",
       "7560_8#6.bam                  None            7560_8#6  \n",
       "7560_8#7.bam                  None            7560_8#7  \n",
       "8002_4#1.bam                  None            8002_4#1  \n",
       "8002_4#2.bam                  None            8002_4#2  \n",
       "8002_4#21.bam                 None           8002_4#21  \n",
       "8002_4#22.bam                 None           8002_4#22  \n",
       "8002_4#23.bam                 None           8002_4#23  \n",
       "8002_4#3.bam                  None            8002_4#3  \n",
       "8002_4#4.bam                  None            8002_4#4  \n",
       "8002_4#5.bam                  None            8002_4#5  \n",
       "8002_4#6.bam                  None            8002_4#6  \n",
       "8002_4#7.bam                  None            8002_4#7  \n",
       "\n",
       "[63 rows x 11 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_output.to_csv('lanelets_with_invalid_codes.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Conclusion\n",
    "Only three of the files in Magnus's list of 13 are in an \"invalid\" study. We will need to look elsewhere for the other 10."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
