{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "See https://github.com/wtsi-team112/fits/issues/4\n",
    "\n",
    "This notebook examines discrepancies between an initial Pf6.2 manifest created by Magnus from fits which has 23,542 files, and one created by Richard which has 17,001."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       ".container {\n",
       "    width: 100%;\n",
       "}\n",
       "# div#notebook {\n",
       "#     padding-top: 1em;\n",
       "# }\n",
       "# #header-container {\n",
       "#     display: none;\n",
       "# }\n",
       "# #header-bar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #maintoolbar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #menubar-container {\n",
       "#     position: fixed;\n",
       "#     margin-top: 0;\n",
       "# }\n",
       "site {\n",
       "    height: auto !important;\n",
       "}\n",
       "</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) \n",
      "[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]\n",
      "numpy 1.13.3\n",
      "scipy 0.19.1\n",
      "pandas 0.22.0\n",
      "numexpr 2.6.4\n",
      "pysam 0.8.4\n",
      "pysamstats 0.24.3\n",
      "petlx 1.0.3\n",
      "vcf 0.6.8\n",
      "h5py 2.7.0\n",
      "tables 3.4.2\n",
      "zarr 2.2.0\n",
      "scikit-allel 1.1.10\n"
     ]
    }
   ],
   "source": [
    "%run ../setup.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "richard_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/Pipelines_issues/work/65_Pf_6_2_Pv4_like_manifest/Pf_62_Pv4_like_manifest_20180906.txt'\n",
    "magnus_fn = './Pf_62.12-09-2018'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read in data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(17001, 26)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>previous_release</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>/seq/1276/1276_5_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1276_5_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_5_nonhuman.srf</td>\n",
       "      <td>5.464048e+09</td>\n",
       "      <td>ERR012350</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/1276/1276_6_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1276_6_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_6_nonhuman.srf</td>\n",
       "      <td>5.479595e+09</td>\n",
       "      <td>ERR012360</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/1333/1333_8_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1333_8_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1333</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1333_8_nonhuman.srf</td>\n",
       "      <td>5.136273e+09</td>\n",
       "      <td>ERR012305</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            path  \\\n",
       "irods_path                                                                         \n",
       "/seq/1276/1276_5_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/1276/1276_6_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/1333/1333_8_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "\n",
       "                                           study    sample             lane  \\\n",
       "irods_path                                                                    \n",
       "/seq/1276/1276_5_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_5_nonhuman   \n",
       "/seq/1276/1276_6_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_6_nonhuman   \n",
       "/seq/1333/1333_8_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1333_8_nonhuman   \n",
       "\n",
       "                               reads  paired sanger_sample_id  taxon_id  \\\n",
       "irods_path                                                                \n",
       "/seq/1276/1276_5_nonhuman.bam     -1       1              NaN      5833   \n",
       "/seq/1276/1276_6_nonhuman.bam     -1       1              NaN      5833   \n",
       "/seq/1333/1333_8_nonhuman.bam     -1       1              NaN      5833   \n",
       "\n",
       "                               study_lims  \\\n",
       "irods_path                                  \n",
       "/seq/1276/1276_5_nonhuman.bam         578   \n",
       "/seq/1276/1276_6_nonhuman.bam         578   \n",
       "/seq/1333/1333_8_nonhuman.bam         578   \n",
       "\n",
       "                                                             study_name  \\\n",
       "irods_path                                                                \n",
       "/seq/1276/1276_5_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "/seq/1276/1276_6_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "/seq/1333/1333_8_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "\n",
       "                               id_run  position  tag_index qc_complete  \\\n",
       "irods_path                                                               \n",
       "/seq/1276/1276_5_nonhuman.bam    1276         5        NaN  2008-09-24   \n",
       "/seq/1276/1276_6_nonhuman.bam    1276         6        NaN  2008-09-24   \n",
       "/seq/1333/1333_8_nonhuman.bam    1333         8        NaN  2008-09-24   \n",
       "\n",
       "                               manual_qc description instrument_name  \\\n",
       "irods_path                                                             \n",
       "/seq/1276/1276_5_nonhuman.bam        NaN         NaN             NaN   \n",
       "/seq/1276/1276_6_nonhuman.bam        NaN         NaN             NaN   \n",
       "/seq/1333/1333_8_nonhuman.bam        NaN         NaN             NaN   \n",
       "\n",
       "                              instrument_model  forward_read_length  \\\n",
       "irods_path                                                            \n",
       "/seq/1276/1276_5_nonhuman.bam            HiSeq                  100   \n",
       "/seq/1276/1276_6_nonhuman.bam            HiSeq                  100   \n",
       "/seq/1333/1333_8_nonhuman.bam            HiSeq                  100   \n",
       "\n",
       "                               requested_insert_size_from  \\\n",
       "irods_path                                                  \n",
       "/seq/1276/1276_5_nonhuman.bam                         NaN   \n",
       "/seq/1276/1276_6_nonhuman.bam                         NaN   \n",
       "/seq/1333/1333_8_nonhuman.bam                         NaN   \n",
       "\n",
       "                               requested_insert_size_to  human_percent_mapped  \\\n",
       "irods_path                                                                      \n",
       "/seq/1276/1276_5_nonhuman.bam                       NaN                   NaN   \n",
       "/seq/1276/1276_6_nonhuman.bam                       NaN                   NaN   \n",
       "/seq/1333/1333_8_nonhuman.bam                       NaN                   NaN   \n",
       "\n",
       "                                 subtrack_filename  subtrack_files_bytes  \\\n",
       "irods_path                                                                 \n",
       "/seq/1276/1276_5_nonhuman.bam  1276_5_nonhuman.srf          5.464048e+09   \n",
       "/seq/1276/1276_6_nonhuman.bam  1276_6_nonhuman.srf          5.479595e+09   \n",
       "/seq/1333/1333_8_nonhuman.bam  1333_8_nonhuman.srf          5.136273e+09   \n",
       "\n",
       "                              ebi_run_acc previous_release  \n",
       "irods_path                                                  \n",
       "/seq/1276/1276_5_nonhuman.bam   ERR012350            Pf6.0  \n",
       "/seq/1276/1276_6_nonhuman.bam   ERR012360            Pf6.0  \n",
       "/seq/1333/1333_8_nonhuman.bam   ERR012305            Pf6.0  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_columns = 30\n",
    "df_richard = pd.read_csv(richard_fn, sep='\\t', index_col=6)\n",
    "print(df_richard.shape)\n",
    "df_richard[0:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(23542, 26)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>previous_release</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>/seq/8563/8563_8#49.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1017-PF-GH-AMENGA-ETEGO</td>\n",
       "      <td>PF0386-C</td>\n",
       "      <td>8563_8#49</td>\n",
       "      <td>31300752.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PF0386-C</td>\n",
       "      <td>5833</td>\n",
       "      <td>578.0</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>8563.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HS24</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR216455</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/8563/8563_8#50.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0078-C</td>\n",
       "      <td>8563_8#50</td>\n",
       "      <td>18722556.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PM0078-C</td>\n",
       "      <td>5833</td>\n",
       "      <td>578.0</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>8563.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HS24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR216456</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/8563/8563_8#51.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1017-PF-GH-AMENGA-ETEGO</td>\n",
       "      <td>PF0397-C</td>\n",
       "      <td>8563_8#51</td>\n",
       "      <td>35463112.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>PF0397-C</td>\n",
       "      <td>5833</td>\n",
       "      <td>578.0</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>8563.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HS24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR216457</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                      path  \\\n",
       "irods_path                                                                   \n",
       "/seq/8563/8563_8#49.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/8563/8563_8#50.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/8563/8563_8#51.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "\n",
       "                                           study    sample       lane  \\\n",
       "irods_path                                                              \n",
       "/seq/8563/8563_8#49.bam  1017-PF-GH-AMENGA-ETEGO  PF0386-C  8563_8#49   \n",
       "/seq/8563/8563_8#50.bam        1001-PF-ML-DJIMDE  PM0078-C  8563_8#50   \n",
       "/seq/8563/8563_8#51.bam  1017-PF-GH-AMENGA-ETEGO  PF0397-C  8563_8#51   \n",
       "\n",
       "                              reads  paired sanger_sample_id  taxon_id  \\\n",
       "irods_path                                                               \n",
       "/seq/8563/8563_8#49.bam  31300752.0     1.0         PF0386-C      5833   \n",
       "/seq/8563/8563_8#50.bam  18722556.0     1.0         PM0078-C      5833   \n",
       "/seq/8563/8563_8#51.bam  35463112.0     1.0         PF0397-C      5833   \n",
       "\n",
       "                         study_lims                                study_name  \\\n",
       "irods_path                                                                      \n",
       "/seq/8563/8563_8#49.bam       578.0  Plasmodium falciparum genome variation 1   \n",
       "/seq/8563/8563_8#50.bam       578.0  Plasmodium falciparum genome variation 1   \n",
       "/seq/8563/8563_8#51.bam       578.0  Plasmodium falciparum genome variation 1   \n",
       "\n",
       "                         id_run  position  tag_index qc_complete  manual_qc  \\\n",
       "irods_path                                                                    \n",
       "/seq/8563/8563_8#49.bam  8563.0       8.0       49.0         NaN        1.0   \n",
       "/seq/8563/8563_8#50.bam  8563.0       8.0       50.0         NaN        1.0   \n",
       "/seq/8563/8563_8#51.bam  8563.0       8.0       51.0         NaN        1.0   \n",
       "\n",
       "                        description instrument_name instrument_model  \\\n",
       "irods_path                                                             \n",
       "/seq/8563/8563_8#49.bam         NaN            HS24            HiSeq   \n",
       "/seq/8563/8563_8#50.bam         NaN            HS24              NaN   \n",
       "/seq/8563/8563_8#51.bam         NaN            HS24              NaN   \n",
       "\n",
       "                         forward_read_length  requested_insert_size_from  \\\n",
       "irods_path                                                                 \n",
       "/seq/8563/8563_8#49.bam                100.0                       200.0   \n",
       "/seq/8563/8563_8#50.bam                100.0                       200.0   \n",
       "/seq/8563/8563_8#51.bam                100.0                       200.0   \n",
       "\n",
       "                         requested_insert_size_to human_percent_mapped  \\\n",
       "irods_path                                                               \n",
       "/seq/8563/8563_8#49.bam                     300.0                 TODO   \n",
       "/seq/8563/8563_8#50.bam                     300.0                 TODO   \n",
       "/seq/8563/8563_8#51.bam                     300.0                 TODO   \n",
       "\n",
       "                        subtrack_filename subtrack_files_bytes ebi_run_acc  \\\n",
       "irods_path                                                                   \n",
       "/seq/8563/8563_8#49.bam              TODO                 TODO   ERR216455   \n",
       "/seq/8563/8563_8#50.bam              TODO                 TODO   ERR216456   \n",
       "/seq/8563/8563_8#51.bam              TODO                 TODO   ERR216457   \n",
       "\n",
       "                        previous_release  \n",
       "irods_path                                \n",
       "/seq/8563/8563_8#49.bam          UNKNOWN  \n",
       "/seq/8563/8563_8#50.bam          UNKNOWN  \n",
       "/seq/8563/8563_8#51.bam          UNKNOWN  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus = pd.read_csv(magnus_fn, index_col=6)\n",
    "print(df_magnus.shape)\n",
    "df_magnus[0:3]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Do files contain unique lanelets?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17001 17001\n"
     ]
    }
   ],
   "source": [
    "print(df_richard['lane'].size, df_richard['lane'].nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23542 23530\n"
     ]
    }
   ],
   "source": [
    "print(df_magnus['lane'].size, df_magnus['lane'].nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#1.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0668-C</td>\n",
       "      <td>10868_3#1</td>\n",
       "      <td>42478110.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#2.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0669-C</td>\n",
       "      <td>10868_3#2</td>\n",
       "      <td>36974150.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#3.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0673-C</td>\n",
       "      <td>10868_3#3</td>\n",
       "      <td>30283304.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#4.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0674-C</td>\n",
       "      <td>10868_3#4</td>\n",
       "      <td>36744668.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#5.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0675-C</td>\n",
       "      <td>10868_3#5</td>\n",
       "      <td>35900900.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#6.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0687-C</td>\n",
       "      <td>10868_3#6</td>\n",
       "      <td>20195200.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#7.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0693-C</td>\n",
       "      <td>10868_3#7</td>\n",
       "      <td>40069810.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#8.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0696-C</td>\n",
       "      <td>10868_3#8</td>\n",
       "      <td>44304020.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#9.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0701-C</td>\n",
       "      <td>10868_3#9</td>\n",
       "      <td>44454962.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#10.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0704-C</td>\n",
       "      <td>10868_3#10</td>\n",
       "      <td>33727438.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#11.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0706-C</td>\n",
       "      <td>10868_3#11</td>\n",
       "      <td>41042808.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/10868_3#12.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0712-C</td>\n",
       "      <td>10868_3#12</td>\n",
       "      <td>38882374.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#1.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0668-C</td>\n",
       "      <td>10868_3#1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#10.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0704-C</td>\n",
       "      <td>10868_3#10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#11.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0706-C</td>\n",
       "      <td>10868_3#11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#12.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0712-C</td>\n",
       "      <td>10868_3#12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#2.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0669-C</td>\n",
       "      <td>10868_3#2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#3.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0673-C</td>\n",
       "      <td>10868_3#3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#4.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0674-C</td>\n",
       "      <td>10868_3#4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#5.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0675-C</td>\n",
       "      <td>10868_3#5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#6.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0687-C</td>\n",
       "      <td>10868_3#6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#7.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0693-C</td>\n",
       "      <td>10868_3#7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#8.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0696-C</td>\n",
       "      <td>10868_3#8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/10868/no_cal/10868_3#9.cram</th>\n",
       "      <td>1083-PF-GH-CONWAY</td>\n",
       "      <td>PF0701-C</td>\n",
       "      <td>10868_3#9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               study    sample        lane  \\\n",
       "irods_path                                                                   \n",
       "/seq/10868/10868_3#1.cram          1083-PF-GH-CONWAY  PF0668-C   10868_3#1   \n",
       "/seq/10868/10868_3#2.cram          1083-PF-GH-CONWAY  PF0669-C   10868_3#2   \n",
       "/seq/10868/10868_3#3.cram          1083-PF-GH-CONWAY  PF0673-C   10868_3#3   \n",
       "/seq/10868/10868_3#4.cram          1083-PF-GH-CONWAY  PF0674-C   10868_3#4   \n",
       "/seq/10868/10868_3#5.cram          1083-PF-GH-CONWAY  PF0675-C   10868_3#5   \n",
       "/seq/10868/10868_3#6.cram          1083-PF-GH-CONWAY  PF0687-C   10868_3#6   \n",
       "/seq/10868/10868_3#7.cram          1083-PF-GH-CONWAY  PF0693-C   10868_3#7   \n",
       "/seq/10868/10868_3#8.cram          1083-PF-GH-CONWAY  PF0696-C   10868_3#8   \n",
       "/seq/10868/10868_3#9.cram          1083-PF-GH-CONWAY  PF0701-C   10868_3#9   \n",
       "/seq/10868/10868_3#10.cram         1083-PF-GH-CONWAY  PF0704-C  10868_3#10   \n",
       "/seq/10868/10868_3#11.cram         1083-PF-GH-CONWAY  PF0706-C  10868_3#11   \n",
       "/seq/10868/10868_3#12.cram         1083-PF-GH-CONWAY  PF0712-C  10868_3#12   \n",
       "/seq/10868/no_cal/10868_3#1.cram   1083-PF-GH-CONWAY  PF0668-C   10868_3#1   \n",
       "/seq/10868/no_cal/10868_3#10.cram  1083-PF-GH-CONWAY  PF0704-C  10868_3#10   \n",
       "/seq/10868/no_cal/10868_3#11.cram  1083-PF-GH-CONWAY  PF0706-C  10868_3#11   \n",
       "/seq/10868/no_cal/10868_3#12.cram  1083-PF-GH-CONWAY  PF0712-C  10868_3#12   \n",
       "/seq/10868/no_cal/10868_3#2.cram   1083-PF-GH-CONWAY  PF0669-C   10868_3#2   \n",
       "/seq/10868/no_cal/10868_3#3.cram   1083-PF-GH-CONWAY  PF0673-C   10868_3#3   \n",
       "/seq/10868/no_cal/10868_3#4.cram   1083-PF-GH-CONWAY  PF0674-C   10868_3#4   \n",
       "/seq/10868/no_cal/10868_3#5.cram   1083-PF-GH-CONWAY  PF0675-C   10868_3#5   \n",
       "/seq/10868/no_cal/10868_3#6.cram   1083-PF-GH-CONWAY  PF0687-C   10868_3#6   \n",
       "/seq/10868/no_cal/10868_3#7.cram   1083-PF-GH-CONWAY  PF0693-C   10868_3#7   \n",
       "/seq/10868/no_cal/10868_3#8.cram   1083-PF-GH-CONWAY  PF0696-C   10868_3#8   \n",
       "/seq/10868/no_cal/10868_3#9.cram   1083-PF-GH-CONWAY  PF0701-C   10868_3#9   \n",
       "\n",
       "                                        reads  paired  \n",
       "irods_path                                             \n",
       "/seq/10868/10868_3#1.cram          42478110.0     1.0  \n",
       "/seq/10868/10868_3#2.cram          36974150.0     1.0  \n",
       "/seq/10868/10868_3#3.cram          30283304.0     1.0  \n",
       "/seq/10868/10868_3#4.cram          36744668.0     1.0  \n",
       "/seq/10868/10868_3#5.cram          35900900.0     1.0  \n",
       "/seq/10868/10868_3#6.cram          20195200.0     1.0  \n",
       "/seq/10868/10868_3#7.cram          40069810.0     1.0  \n",
       "/seq/10868/10868_3#8.cram          44304020.0     1.0  \n",
       "/seq/10868/10868_3#9.cram          44454962.0     1.0  \n",
       "/seq/10868/10868_3#10.cram         33727438.0     1.0  \n",
       "/seq/10868/10868_3#11.cram         41042808.0     1.0  \n",
       "/seq/10868/10868_3#12.cram         38882374.0     1.0  \n",
       "/seq/10868/no_cal/10868_3#1.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#10.cram         NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#11.cram         NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#12.cram         NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#2.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#3.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#4.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#5.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#6.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#7.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#8.cram          NaN     NaN  \n",
       "/seq/10868/no_cal/10868_3#9.cram          NaN     NaN  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus.loc[df_magnus['lane'].duplicated(keep=False), ['study', 'sample', 'lane', 'reads', 'paired']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(12, 26)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus.loc[df_magnus.index.to_series().str.contains('no_cal')].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summary\n",
    "Magnus file contains 12 lanelets which are duplicated, with one version containing 'no_cal' in file path"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lanelets in richard but not magnus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2385, 26)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>previous_release</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>/seq/1276/1276_5_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1276_5_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_5_nonhuman.srf</td>\n",
       "      <td>5.464048e+09</td>\n",
       "      <td>ERR012350</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/1276/1276_6_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1276_6_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_6_nonhuman.srf</td>\n",
       "      <td>5.479595e+09</td>\n",
       "      <td>ERR012360</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/1333/1333_8_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1333_8_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1333</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1333_8_nonhuman.srf</td>\n",
       "      <td>5.136273e+09</td>\n",
       "      <td>ERR012305</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/1276/1276_7_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0002-C</td>\n",
       "      <td>1276_7_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_7_nonhuman.srf</td>\n",
       "      <td>1.839657e+09</td>\n",
       "      <td>ERR012327</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/1276/1276_8_nonhuman.bam</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0002-C</td>\n",
       "      <td>1276_8_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_8_nonhuman.srf</td>\n",
       "      <td>1.672178e+09</td>\n",
       "      <td>ERR012520</td>\n",
       "      <td>Pf6.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            path  \\\n",
       "irods_path                                                                         \n",
       "/seq/1276/1276_5_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/1276/1276_6_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/1333/1333_8_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/1276/1276_7_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/1276/1276_8_nonhuman.bam  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "\n",
       "                                           study    sample             lane  \\\n",
       "irods_path                                                                    \n",
       "/seq/1276/1276_5_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_5_nonhuman   \n",
       "/seq/1276/1276_6_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1276_6_nonhuman   \n",
       "/seq/1333/1333_8_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C  1333_8_nonhuman   \n",
       "/seq/1276/1276_7_nonhuman.bam  1001-PF-ML-DJIMDE  PM0002-C  1276_7_nonhuman   \n",
       "/seq/1276/1276_8_nonhuman.bam  1001-PF-ML-DJIMDE  PM0002-C  1276_8_nonhuman   \n",
       "\n",
       "                               reads  paired sanger_sample_id  taxon_id  \\\n",
       "irods_path                                                                \n",
       "/seq/1276/1276_5_nonhuman.bam     -1       1              NaN      5833   \n",
       "/seq/1276/1276_6_nonhuman.bam     -1       1              NaN      5833   \n",
       "/seq/1333/1333_8_nonhuman.bam     -1       1              NaN      5833   \n",
       "/seq/1276/1276_7_nonhuman.bam     -1       1              NaN      5833   \n",
       "/seq/1276/1276_8_nonhuman.bam     -1       1              NaN      5833   \n",
       "\n",
       "                               study_lims  \\\n",
       "irods_path                                  \n",
       "/seq/1276/1276_5_nonhuman.bam         578   \n",
       "/seq/1276/1276_6_nonhuman.bam         578   \n",
       "/seq/1333/1333_8_nonhuman.bam         578   \n",
       "/seq/1276/1276_7_nonhuman.bam         578   \n",
       "/seq/1276/1276_8_nonhuman.bam         578   \n",
       "\n",
       "                                                             study_name  \\\n",
       "irods_path                                                                \n",
       "/seq/1276/1276_5_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "/seq/1276/1276_6_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "/seq/1333/1333_8_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "/seq/1276/1276_7_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "/seq/1276/1276_8_nonhuman.bam  Plasmodium falciparum genome variation 1   \n",
       "\n",
       "                               id_run  position  tag_index qc_complete  \\\n",
       "irods_path                                                               \n",
       "/seq/1276/1276_5_nonhuman.bam    1276         5        NaN  2008-09-24   \n",
       "/seq/1276/1276_6_nonhuman.bam    1276         6        NaN  2008-09-24   \n",
       "/seq/1333/1333_8_nonhuman.bam    1333         8        NaN  2008-09-24   \n",
       "/seq/1276/1276_7_nonhuman.bam    1276         7        NaN  2008-09-24   \n",
       "/seq/1276/1276_8_nonhuman.bam    1276         8        NaN  2008-09-24   \n",
       "\n",
       "                               manual_qc description instrument_name  \\\n",
       "irods_path                                                             \n",
       "/seq/1276/1276_5_nonhuman.bam        NaN         NaN             NaN   \n",
       "/seq/1276/1276_6_nonhuman.bam        NaN         NaN             NaN   \n",
       "/seq/1333/1333_8_nonhuman.bam        NaN         NaN             NaN   \n",
       "/seq/1276/1276_7_nonhuman.bam        NaN         NaN             NaN   \n",
       "/seq/1276/1276_8_nonhuman.bam        NaN         NaN             NaN   \n",
       "\n",
       "                              instrument_model  forward_read_length  \\\n",
       "irods_path                                                            \n",
       "/seq/1276/1276_5_nonhuman.bam            HiSeq                  100   \n",
       "/seq/1276/1276_6_nonhuman.bam            HiSeq                  100   \n",
       "/seq/1333/1333_8_nonhuman.bam            HiSeq                  100   \n",
       "/seq/1276/1276_7_nonhuman.bam            HiSeq                  100   \n",
       "/seq/1276/1276_8_nonhuman.bam            HiSeq                  100   \n",
       "\n",
       "                               requested_insert_size_from  \\\n",
       "irods_path                                                  \n",
       "/seq/1276/1276_5_nonhuman.bam                         NaN   \n",
       "/seq/1276/1276_6_nonhuman.bam                         NaN   \n",
       "/seq/1333/1333_8_nonhuman.bam                         NaN   \n",
       "/seq/1276/1276_7_nonhuman.bam                         NaN   \n",
       "/seq/1276/1276_8_nonhuman.bam                         NaN   \n",
       "\n",
       "                               requested_insert_size_to  human_percent_mapped  \\\n",
       "irods_path                                                                      \n",
       "/seq/1276/1276_5_nonhuman.bam                       NaN                   NaN   \n",
       "/seq/1276/1276_6_nonhuman.bam                       NaN                   NaN   \n",
       "/seq/1333/1333_8_nonhuman.bam                       NaN                   NaN   \n",
       "/seq/1276/1276_7_nonhuman.bam                       NaN                   NaN   \n",
       "/seq/1276/1276_8_nonhuman.bam                       NaN                   NaN   \n",
       "\n",
       "                                 subtrack_filename  subtrack_files_bytes  \\\n",
       "irods_path                                                                 \n",
       "/seq/1276/1276_5_nonhuman.bam  1276_5_nonhuman.srf          5.464048e+09   \n",
       "/seq/1276/1276_6_nonhuman.bam  1276_6_nonhuman.srf          5.479595e+09   \n",
       "/seq/1333/1333_8_nonhuman.bam  1333_8_nonhuman.srf          5.136273e+09   \n",
       "/seq/1276/1276_7_nonhuman.bam  1276_7_nonhuman.srf          1.839657e+09   \n",
       "/seq/1276/1276_8_nonhuman.bam  1276_8_nonhuman.srf          1.672178e+09   \n",
       "\n",
       "                              ebi_run_acc previous_release  \n",
       "irods_path                                                  \n",
       "/seq/1276/1276_5_nonhuman.bam   ERR012350            Pf6.0  \n",
       "/seq/1276/1276_6_nonhuman.bam   ERR012360            Pf6.0  \n",
       "/seq/1333/1333_8_nonhuman.bam   ERR012305            Pf6.0  \n",
       "/seq/1276/1276_7_nonhuman.bam   ERR012327            Pf6.0  \n",
       "/seq/1276/1276_8_nonhuman.bam   ERR012520            Pf6.0  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_richard_not_magnus = df_richard.loc[~df_richard.index.to_series().isin(df_magnus.index)]\n",
    "print(df_richard_not_magnus.shape)\n",
    "df_richard_not_magnus[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True     2167\n",
       "False     218\n",
       "Name: sanger_sample_id, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_richard_not_magnus['sanger_sample_id'].isnull().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Pf6.0    2383\n",
       "Pf6.1       2\n",
       "Name: previous_release, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_richard_not_magnus['previous_release'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1017-PF-GH-AMENGA-ETEGO          299\n",
       "1044-PF-KH-FAIRHURST             251\n",
       "1001-PF-ML-DJIMDE                186\n",
       "1031-PF-SEA-PLOWE                179\n",
       "1010-PF-TH-ANDERSON              134\n",
       "1026-PF-GN-CONWAY                132\n",
       "1004-PF-BF-OUEDRAOGO             128\n",
       "1011-PF-KH-SU                    110\n",
       "1016-PF-TH-NOSTEN                 92\n",
       "1147-PF-MR-CONWAY                 86\n",
       "1015-PF-KE-NZILA                  83\n",
       "1006-PF-GM-CONWAY                 75\n",
       "1155-PF-ID-PRICE                  70\n",
       "1021-PF-PG-MUELLER                60\n",
       "1043-PF-GB-RAYNER                 58\n",
       "1022-PF-MW-OCHOLLA                53\n",
       "1052-PF-TRAC-WHITE                44\n",
       "1032-PF-BRHN-SMITHEE              42\n",
       "1041-PF-US-FERDIG                 40\n",
       "1037-PF-GM-CONWAY                 31\n",
       "1012-PF-KH-WHITE                  27\n",
       "1042-PF-HN-RANFORD-CARTWRIGHT     22\n",
       "1013-PF-PEGB-BRANCH               21\n",
       "1098-PF-ET-GOLASSA                19\n",
       "1023-PF-CO-ECHEVERRI-GARCIA       17\n",
       "1020-PF-VN-BONI                   16\n",
       "1014-PF-SSA-SUTHERLAND            16\n",
       "1007-PF-TZ-DUFFY                  16\n",
       "1153-PF-Pf3KLAB-KWIATKOWSKI       15\n",
       "1024-PF-UG-BOUSEMA                12\n",
       "1018-PF-GB-NEWBOLD                10\n",
       "1025-PF-KH-PLOWE                  10\n",
       "1033-PF-BRGHSEA-SU                 8\n",
       "1038-PF-SD-HVIID                   7\n",
       "1009-PF-KH-PLOWE                   5\n",
       "1150-PF-Pf3KLAB-CONWAY             4\n",
       "1039-PF-PFLAB9-BAKER               3\n",
       "1034-PF-BRSEA-RAYNER               3\n",
       "1036-PF-VN-NZILA                   1\n",
       "Name: study, dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_richard_not_magnus['study'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6345     84\n",
       "6263     78\n",
       "6466     74\n",
       "6295     72\n",
       "6222     69\n",
       "17234    60\n",
       "6936     46\n",
       "6901     45\n",
       "6007     39\n",
       "17397    38\n",
       "Name: id_run, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_richard_not_magnus['id_run'].value_counts(dropna=False).head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>previous_release</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [path, study, sample, lane, reads, paired, sanger_sample_id, taxon_id, study_lims, study_name, id_run, position, tag_index, qc_complete, manual_qc, description, instrument_name, instrument_model, forward_read_length, requested_insert_size_from, requested_insert_size_to, human_percent_mapped, subtrack_filename, subtrack_files_bytes, ebi_run_acc, previous_release]\n",
       "Index: []"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus.loc[df_magnus['id_run'] == 6345.0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>previous_release</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [path, study, sample, lane, reads, paired, sanger_sample_id, taxon_id, study_lims, study_name, id_run, position, tag_index, qc_complete, manual_qc, description, instrument_name, instrument_model, forward_read_length, requested_insert_size_from, requested_insert_size_to, human_percent_mapped, subtrack_filename, subtrack_files_bytes, ebi_run_acc, previous_release]\n",
       "Index: []"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus.loc[df_magnus['id_run'] == 6263.0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>previous_release</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [path, study, sample, lane, reads, paired, sanger_sample_id, taxon_id, study_lims, study_name, id_run, position, tag_index, qc_complete, manual_qc, description, instrument_name, instrument_model, forward_read_length, requested_insert_size_from, requested_insert_size_to, human_percent_mapped, subtrack_filename, subtrack_files_bytes, ebi_run_acc, previous_release]\n",
       "Index: []"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus.loc[df_magnus['id_run'] == 6466.0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(10, 26)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus.loc[df_magnus['id_run'] == 17234.0].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(70, 26)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_richard.loc[df_richard['id_run'] == 17234.0].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summary\n",
    "There are 2,385 files in Richard's manifest, which aren't in Magnus's file. All of these were in previous releases so clearly should be present. I could see no obvious pattern in terms of studies or runs which were missing. For run 17234 for example, Magnus file had 10 lanelets, whereas Richard's had 70. The majority of the differences had no sanger_sample_id."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lanelets in magnus but not richard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(8926, 26)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "      <th>previous_release</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_path</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>/seq/23384/23384_1#1.cram</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SPT09376</td>\n",
       "      <td>23384_1#1</td>\n",
       "      <td>30821088.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4660STDY6750851</td>\n",
       "      <td>5833</td>\n",
       "      <td>4660.0</td>\n",
       "      <td>1204-PF-GM-CP1</td>\n",
       "      <td>23384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>HX10</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR2364729</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/23384/23384_1#2.cram</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SPT09379</td>\n",
       "      <td>23384_1#2</td>\n",
       "      <td>38921494.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4660STDY6750859</td>\n",
       "      <td>5833</td>\n",
       "      <td>4660.0</td>\n",
       "      <td>1204-PF-GM-CP1</td>\n",
       "      <td>23384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>HX10</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR2364730</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/23384/23384_1#3.cram</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SPT09398</td>\n",
       "      <td>23384_1#3</td>\n",
       "      <td>31690686.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4660STDY6750868</td>\n",
       "      <td>5833</td>\n",
       "      <td>4660.0</td>\n",
       "      <td>1204-PF-GM-CP1</td>\n",
       "      <td>23384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>HX10</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR2364731</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/23384/23384_1#4.cram</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SPT09396</td>\n",
       "      <td>23384_1#4</td>\n",
       "      <td>29677456.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>HX10</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR2364732</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>/seq/23384/23384_1#5.cram</th>\n",
       "      <td>/lustre/scratch118/malaria/team112/pipelines/s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SPT09413</td>\n",
       "      <td>23384_1#5</td>\n",
       "      <td>30322306.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4660STDY6750861</td>\n",
       "      <td>5833</td>\n",
       "      <td>4660.0</td>\n",
       "      <td>1204-PF-GM-CP1</td>\n",
       "      <td>23384.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2017-08-21 14:30:49</td>\n",
       "      <td>HX10</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>TODO</td>\n",
       "      <td>ERR2364733</td>\n",
       "      <td>UNKNOWN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                        path  \\\n",
       "irods_path                                                                     \n",
       "/seq/23384/23384_1#1.cram  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/23384/23384_1#2.cram  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/23384/23384_1#3.cram  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/23384/23384_1#4.cram  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "/seq/23384/23384_1#5.cram  /lustre/scratch118/malaria/team112/pipelines/s...   \n",
       "\n",
       "                          study    sample       lane       reads  paired  \\\n",
       "irods_path                                                                 \n",
       "/seq/23384/23384_1#1.cram   NaN  SPT09376  23384_1#1  30821088.0     1.0   \n",
       "/seq/23384/23384_1#2.cram   NaN  SPT09379  23384_1#2  38921494.0     1.0   \n",
       "/seq/23384/23384_1#3.cram   NaN  SPT09398  23384_1#3  31690686.0     1.0   \n",
       "/seq/23384/23384_1#4.cram   NaN  SPT09396  23384_1#4  29677456.0     1.0   \n",
       "/seq/23384/23384_1#5.cram   NaN  SPT09413  23384_1#5  30322306.0     1.0   \n",
       "\n",
       "                          sanger_sample_id  taxon_id  study_lims  \\\n",
       "irods_path                                                         \n",
       "/seq/23384/23384_1#1.cram  4660STDY6750851      5833      4660.0   \n",
       "/seq/23384/23384_1#2.cram  4660STDY6750859      5833      4660.0   \n",
       "/seq/23384/23384_1#3.cram  4660STDY6750868      5833      4660.0   \n",
       "/seq/23384/23384_1#4.cram              NaN      5833         NaN   \n",
       "/seq/23384/23384_1#5.cram  4660STDY6750861      5833      4660.0   \n",
       "\n",
       "                               study_name   id_run  position  tag_index  \\\n",
       "irods_path                                                                \n",
       "/seq/23384/23384_1#1.cram  1204-PF-GM-CP1  23384.0       1.0        1.0   \n",
       "/seq/23384/23384_1#2.cram  1204-PF-GM-CP1  23384.0       1.0        2.0   \n",
       "/seq/23384/23384_1#3.cram  1204-PF-GM-CP1  23384.0       1.0        3.0   \n",
       "/seq/23384/23384_1#4.cram             NaN  23384.0       1.0        4.0   \n",
       "/seq/23384/23384_1#5.cram  1204-PF-GM-CP1  23384.0       1.0        5.0   \n",
       "\n",
       "                                   qc_complete  manual_qc  \\\n",
       "irods_path                                                  \n",
       "/seq/23384/23384_1#1.cram  2017-08-21 14:30:49        1.0   \n",
       "/seq/23384/23384_1#2.cram  2017-08-21 14:30:49        1.0   \n",
       "/seq/23384/23384_1#3.cram  2017-08-21 14:30:49        1.0   \n",
       "/seq/23384/23384_1#4.cram  2017-08-21 14:30:49        1.0   \n",
       "/seq/23384/23384_1#5.cram  2017-08-21 14:30:49        1.0   \n",
       "\n",
       "                                   description instrument_name  \\\n",
       "irods_path                                                       \n",
       "/seq/23384/23384_1#1.cram  2017-08-21 14:30:49            HX10   \n",
       "/seq/23384/23384_1#2.cram  2017-08-21 14:30:49            HX10   \n",
       "/seq/23384/23384_1#3.cram  2017-08-21 14:30:49            HX10   \n",
       "/seq/23384/23384_1#4.cram  2017-08-21 14:30:49            HX10   \n",
       "/seq/23384/23384_1#5.cram  2017-08-21 14:30:49            HX10   \n",
       "\n",
       "                          instrument_model  forward_read_length  \\\n",
       "irods_path                                                        \n",
       "/seq/23384/23384_1#1.cram           HiSeqX                150.0   \n",
       "/seq/23384/23384_1#2.cram           HiSeqX                150.0   \n",
       "/seq/23384/23384_1#3.cram           HiSeqX                150.0   \n",
       "/seq/23384/23384_1#4.cram           HiSeqX                150.0   \n",
       "/seq/23384/23384_1#5.cram           HiSeqX                150.0   \n",
       "\n",
       "                           requested_insert_size_from  \\\n",
       "irods_path                                              \n",
       "/seq/23384/23384_1#1.cram                       450.0   \n",
       "/seq/23384/23384_1#2.cram                       450.0   \n",
       "/seq/23384/23384_1#3.cram                       450.0   \n",
       "/seq/23384/23384_1#4.cram                       450.0   \n",
       "/seq/23384/23384_1#5.cram                       450.0   \n",
       "\n",
       "                           requested_insert_size_to human_percent_mapped  \\\n",
       "irods_path                                                                 \n",
       "/seq/23384/23384_1#1.cram                     450.0                 TODO   \n",
       "/seq/23384/23384_1#2.cram                     450.0                 TODO   \n",
       "/seq/23384/23384_1#3.cram                     450.0                 TODO   \n",
       "/seq/23384/23384_1#4.cram                     450.0                 TODO   \n",
       "/seq/23384/23384_1#5.cram                     450.0                 TODO   \n",
       "\n",
       "                          subtrack_filename subtrack_files_bytes ebi_run_acc  \\\n",
       "irods_path                                                                     \n",
       "/seq/23384/23384_1#1.cram              TODO                 TODO  ERR2364729   \n",
       "/seq/23384/23384_1#2.cram              TODO                 TODO  ERR2364730   \n",
       "/seq/23384/23384_1#3.cram              TODO                 TODO  ERR2364731   \n",
       "/seq/23384/23384_1#4.cram              TODO                 TODO  ERR2364732   \n",
       "/seq/23384/23384_1#5.cram              TODO                 TODO  ERR2364733   \n",
       "\n",
       "                          previous_release  \n",
       "irods_path                                  \n",
       "/seq/23384/23384_1#1.cram          UNKNOWN  \n",
       "/seq/23384/23384_1#2.cram          UNKNOWN  \n",
       "/seq/23384/23384_1#3.cram          UNKNOWN  \n",
       "/seq/23384/23384_1#4.cram          UNKNOWN  \n",
       "/seq/23384/23384_1#5.cram          UNKNOWN  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus_not_richard = df_magnus.loc[~df_magnus.index.to_series().isin(df_richard.index)]\n",
    "print(df_magnus_not_richard.shape)\n",
    "df_magnus_not_richard[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1176-PF-KE-CP1                                        2458\n",
       "1204-PF-GM-CP1                                        2310\n",
       "Malaria Programme R&D                                 2192\n",
       "NaN                                                    824\n",
       "Team 112 R&D                                           594\n",
       "Malaria R&D                                            473\n",
       "AT-rich amplification                                   18\n",
       "1083-PF-GH-CONWAY                                       12\n",
       "IHTP_PWGS 1175-VO-KH-STLAURENT                          12\n",
       "ILB_1044-PF-KH-FAIRHURST                                12\n",
       "ILB_1126-PF-LAB-FAIRHURST                                8\n",
       "1148-PF-BD-MAUDE                                         6\n",
       "Test sequencing of high level human contamination.       6\n",
       "ILB_1106-PV-MULTI-PRICE_INDONESIA                        1\n",
       "Name: study_name, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus_not_richard['study_name'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NaN                        17488\n",
       "1052-PF-TRAC-WHITE          1052\n",
       "1125-PF-TH-NOSTEN            674\n",
       "1040-PF-GB-HAMILTON          636\n",
       "1022-PF-MW-OCHOLLA           318\n",
       "1095-PF-TZ-ISHENGOMA         300\n",
       "1044-PF-KH-FAIRHURST         288\n",
       "1094-PF-GH-AMENGA-ETEGO      256\n",
       "1093-PF-CM-APINJOH           239\n",
       "1008-PF-SEA-RINGWALD         234\n",
       "Name: study, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_magnus['study'].value_counts(dropna=False).head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Conclusion\n",
    "There are multiple problems with this version of the Pf6.2 manifest from fits:\n",
    "- A small number of lanelets have multiple rows, with one version containing \"no_cal\" in the file path\n",
    "- Most lanelets have no study associated. This is required for all rows\n",
    "- There are many lanelets from studies which should have been excluded, e.g. 1176-PF-KE-CP1, 1204-PF-GM-CP1 and R&D studies\n",
    "- Most worryingly, there are a large number of lanelets that are missing but that were included in previous builds. In Richard's version of the manifest most, but not all of these were missing sanger_sample_id. I couldn't spot any other obvious pattern.\n",
    "\n",
    "Note that I used a file to map sequencescape studies to Alfresco studies. This can be found in ```/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/sequencescape_alfresco_study_mappings.txt``` and was created with https://github.com/wtsi-team112/SIMS/blob/master/notebooks/rp7/20180203_mlwh_exceptions_files.ipynb\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
