{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "See https://github.com/wtsi-team112/fits/issues/4\n",
    "\n",
    "Magnus sent a file with the following files that are in FITS but not in the Pf 6.2 manifest:\n",
    "```\n",
    "sample\tstudy_name\tirods_path\tsanger_sample_id\n",
    "NULL\tPlasmodium falciparum genome variation 1\t/seq/6901/6901_8_nonhuman#2.bam\tTr5E\n",
    "NULL\tPlasmodium falciparum genome variation 1\t/seq/6901/6901_8_nonhuman#11.bam\tTr3E\n",
    "NULL\tPlasmodium falciparum genome variation 1\t/seq/6901/6901_8_nonhuman#10.bam\tTr5E_1\n",
    "NULL\tPlasmodium falciparum genome variation 1\t/seq/6901/6901_8_nonhuman#3.bam\tTr3E_1\n",
    "PD0338-C\tTest sequencing of high level human contamination.\t/seq/12135/12135_6#66.cram\t2876STDY5684799\n",
    "QE0008-C\tTest sequencing of high level human contamination.\t/seq/12135/12135_7#68.cram\t2876STDY5684801\n",
    "QE0022-C\tTest sequencing of high level human contamination.\t/seq/12135/12135_7#70.cram\t2876STDY5684803\n",
    "QE0145-C\tTest sequencing of high level human contamination.\t/seq/12135/12135_7#69.cram\t2876STDY5684802\n",
    "QE0196-C\tTest sequencing of high level human contamination.\t/seq/12135/12135_7#67.cram\t2876STDY5684800\n",
    "QE0212-C\tTest sequencing of high level human contamination.\t/seq/12135/12135_7#71.cram\t2876STDY5684804\n",
    "```\n",
    "I couldn't see an obvious explanation as to why these weren't in Pf 6.2. Here I investigate these."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       ".container {\n",
       "    width: 100%;\n",
       "}\n",
       "# div#notebook {\n",
       "#     padding-top: 1em;\n",
       "# }\n",
       "# #header-container {\n",
       "#     display: none;\n",
       "# }\n",
       "# #header-bar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #maintoolbar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #menubar-container {\n",
       "#     position: fixed;\n",
       "#     margin-top: 0;\n",
       "# }\n",
       "site {\n",
       "    height: auto !important;\n",
       "}\n",
       "</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) \n",
      "[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]\n",
      "numpy 1.13.3\n",
      "scipy 0.19.1\n",
      "pandas 0.22.0\n",
      "numexpr 2.6.4\n",
      "pysam 0.8.4\n",
      "pysamstats 0.24.3\n",
      "petlx 1.0.3\n",
      "vcf 0.6.8\n",
      "h5py 2.7.0\n",
      "tables 3.4.2\n",
      "zarr 2.2.0\n",
      "scikit-allel 1.1.10\n"
     ]
    }
   ],
   "source": [
    "%run ../setup.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Inputs\n",
    "mlwh_missing_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_missing_exceptions.txt'\n",
    "mlwh_study_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_study_exceptions.txt'\n",
    "mlwh_sample_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_sample_exceptions.txt'\n",
    "mlwh_taxon_exceptions_fn = '/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/mlwh_taxon_exceptions.txt'\n",
    "sequencescape_alfresco_study_mappings_fn='/nfs/team112_internal/rp7/src/github/wtsi-team112/SIMS/meta/mlwh/sequencescape_alfresco_study_mappings.txt'\n",
    "\n",
    "# Outputs\n",
    "output_fn = 'Pf_62_Pv4_like_manifest_20180906.txt'\n",
    "manifest_symlink = 'pf_62_manifest.txt'\n",
    "input_dir = '/lustre/scratch118/malaria/team112/pipelines/setups/pf_62/input'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Does the mlwh query pull in these files?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def set_irods_name(row):\n",
    "    # First calculate prefix using nonhuman dependent on id_run and species (taxon)\n",
    "    # Then calculate tag string (empty string if no tag)\n",
    "    # Then calculate file extension\n",
    "    # Then concat these three strings\n",
    "    if (\n",
    "        (row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 6750)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    elif (\n",
    "        ~(row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 7100)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    else:\n",
    "        prefix = \"%d_%d\" % (row['id_run'], row['position'])\n",
    "        \n",
    "    if np.isnan(row['tag_index']):\n",
    "        tag_string = ''\n",
    "    else:\n",
    "        tag_string = '#%d' % row['tag_index']\n",
    "    \n",
    "    if row['id_run'] <= 10300:\n",
    "        file_extension = '.bam'\n",
    "    else:\n",
    "        file_extension = '.cram'\n",
    "    \n",
    "    irods_filename = prefix + tag_string + file_extension\n",
    "    return(irods_filename)\n",
    "\n",
    "def set_sample_id(row):\n",
    "    if row['sample'] is None:\n",
    "        return('DS_%s' % row['name'])\n",
    "    else:\n",
    "        return('DS_%s' % row['sample'])\n",
    "\n",
    "def is_valid_alfresco_code(s):\n",
    "    try: \n",
    "        int(s[0:4])\n",
    "        return True\n",
    "    except ValueError:\n",
    "        return False\n",
    "\n",
    "def create_build_manifest(\n",
    "#     taxon_ids=[5833, 5855, 5858, 5821, 7165, 7173, 30066], # Pf, Pv, Pm, Pb, Ag, A. arabiensis, A. merus\n",
    "    taxon_ids=[5833, 36329, 5847, 57267, 137071], # Pf, 3D7, V1, Dd2, HB3\n",
    "    # 1089 is excluded as this is R&D samples. 1204 and 1176 are excluded as these are CP1 samples. 1175 is excluded as some Pf R&D samples were incorrectly tagged with this study\n",
    "    studies_to_exclude = ['1089-R&amp;D-GB-ALCOCK', '1204-PF-GM-CP1', '1176-PF-KE-CP1', '1175-VO-KH-STLAURENT'],\n",
    "    miseq_runs_to_allow = [13809, 13810], # Two Miseq runs within 24 samples on each from study 1095-PF-TZ-ISHENGOMA that were included in Pf 6.0\n",
    "    input_dir = input_dir,\n",
    "    mlwh_missing_exceptions_fn = mlwh_missing_exceptions_fn,\n",
    "    mlwh_study_exceptions_fn = mlwh_study_exceptions_fn,\n",
    "    mlwh_sample_exceptions_fn = mlwh_sample_exceptions_fn,\n",
    "    mlwh_taxon_exceptions_fn = mlwh_taxon_exceptions_fn,\n",
    "    sequencescape_alfresco_study_mappings_fn = sequencescape_alfresco_study_mappings_fn,\n",
    "    output_columns = [\n",
    "        'path',\n",
    "        'study',\n",
    "        'sample',\n",
    "        'lane',\n",
    "        'reads',\n",
    "        'paired',\n",
    "        'irods_path',\n",
    "        'sanger_sample_id',\n",
    "        'taxon_id',\n",
    "        'study_lims',\n",
    "        'study_name',\n",
    "        'id_run',\n",
    "        'position',\n",
    "        'tag_index',\n",
    "        'qc_complete',\n",
    "        'manual_qc',\n",
    "        'description',\n",
    "        'instrument_name',\n",
    "        'instrument_model',\n",
    "        'forward_read_length',\n",
    "        'requested_insert_size_from',\n",
    "        'requested_insert_size_to',\n",
    "        'human_percent_mapped',\n",
    "        'subtrack_filename',\n",
    "        'subtrack_files_bytes',\n",
    "        'ebi_run_acc'\n",
    "    ]\n",
    "):\n",
    "    \"\"\"Create a DataFrame to be used as a build manifest.\n",
    "    \n",
    "    A build manifest here is a list of all the \"lanelets\" that need to be\n",
    "    included in a build. The output will typically be written to a\n",
    "    tab-delmited file, either to use as input to vr-pipe, or perhaps as\n",
    "    input to a vr-track DB.\n",
    "\n",
    "    Args:\n",
    "        taxon_ids (list of int): The taxons of the build (P. falciparum is '5833',\n",
    "            P. vivax is '5855').\n",
    "        sequencescape_alfresco_study_mappings_fn (str): filename of\n",
    "            mappings from sequencscape to alfresco study mappings\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The build manifest.\n",
    "\n",
    "    \"\"\"\n",
    "    \n",
    "    # Read in taxon exceptions file\n",
    "    df_mlwh_taxon_exceptions = pd.read_csv(mlwh_taxon_exceptions_fn, sep='\\t', dtype={'tag_index': str, 'taxon_id': int}, index_col='irods_filename')\n",
    "    df_mlwh_taxon_exceptions['tag_index'].fillna('', inplace=True)\n",
    "    # Identify which samples in exceptions file match the taxon of current build, and which don't\n",
    "    df_mlwh_exceptions_this_taxon = df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)]\n",
    "    df_mlwh_exceptions_other_taxa = df_mlwh_taxon_exceptions.loc[~(df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids))]\n",
    "\n",
    "    # Read in sample exceptions file\n",
    "    df_mlwh_sample_exceptions = pd.read_csv(mlwh_sample_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in study exceptions file\n",
    "    df_mlwh_study_exceptions = pd.read_csv(mlwh_study_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in missing exceptions file\n",
    "    df_mlwh_missing_exceptions = pd.read_csv(mlwh_missing_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "    df_mlwh_missing_exceptions['derivative_sample_id'] = 'DS_' + df_mlwh_missing_exceptions['sample_id']\n",
    "    df_mlwh_missing_exceptions = df_mlwh_missing_exceptions.loc[\n",
    "        df_mlwh_missing_exceptions['taxon_id'].isin(taxon_ids)\n",
    "    ]\n",
    "\n",
    "    # Read in SequenceScape-Alfresco study mappings\n",
    "    df_sequencescape_alfresco_study_mappings = pd.read_csv(sequencescape_alfresco_study_mappings_fn, sep='\\t', index_col='seqscape_study_id')\n",
    "    sequencescape_alfresco_study_mappings_dict = df_sequencescape_alfresco_study_mappings.loc[\n",
    "        :,\n",
    "#         df_sequencescape_alfresco_study_mappings['build_flag']==1,\n",
    "        'alfresco_study_code'\n",
    "    ].to_dict()\n",
    "    \n",
    "    # Read in data from mlwh matching this taxon, plus samples from exceptions file matching this taxon\n",
    "    conn = pymysql.connect(\n",
    "        host='mlwh-db',\n",
    "        user='mlwh_malaria',\n",
    "        password='Solaris&2015',\n",
    "        db='mlwarehouse',\n",
    "        port=3435\n",
    "    )\n",
    "    \n",
    "    sql_query = 'SELECT \\\n",
    "        study.name as study_name, \\\n",
    "        study.id_study_lims as study_lims, \\\n",
    "        sample.supplier_name as sample, \\\n",
    "        sample.name, \\\n",
    "        sample.sanger_sample_id, \\\n",
    "        sample.taxon_id, \\\n",
    "        iseq_product_metrics.id_run, \\\n",
    "        iseq_product_metrics.position, \\\n",
    "        iseq_product_metrics.tag_index, \\\n",
    "        iseq_product_metrics.num_reads, \\\n",
    "        iseq_product_metrics.human_percent_mapped, \\\n",
    "        iseq_run_lane_metrics.instrument_name, \\\n",
    "        iseq_run_lane_metrics.instrument_model, \\\n",
    "        iseq_run_lane_metrics.paired_read, \\\n",
    "        iseq_run_lane_metrics.qc_complete, \\\n",
    "        iseq_flowcell.manual_qc, \\\n",
    "        iseq_flowcell.requested_insert_size_from, \\\n",
    "        iseq_flowcell.requested_insert_size_to, \\\n",
    "        iseq_flowcell.forward_read_length, \\\n",
    "        iseq_run_status_dict.description \\\n",
    "    FROM \\\n",
    "        study, \\\n",
    "        iseq_flowcell, \\\n",
    "        sample, \\\n",
    "        iseq_product_metrics, \\\n",
    "        iseq_run_status, \\\n",
    "        iseq_run_lane_metrics, \\\n",
    "        iseq_run_status_dict \\\n",
    "    WHERE \\\n",
    "        study.id_study_tmp = iseq_flowcell.id_study_tmp and \\\n",
    "        iseq_flowcell.id_sample_tmp = sample.id_sample_tmp and \\\n",
    "        iseq_flowcell.manual_qc = 1 and \\\n",
    "        iseq_product_metrics.id_iseq_flowcell_tmp = iseq_flowcell.id_iseq_flowcell_tmp and \\\n",
    "        iseq_run_status.id_run = iseq_product_metrics.id_run and \\\n",
    "        iseq_product_metrics.id_run = iseq_run_lane_metrics.id_run and \\\n",
    "        iseq_product_metrics.position = iseq_run_lane_metrics.position and \\\n",
    "        iseq_run_status.iscurrent = 1 and \\\n",
    "        ( ( iseq_run_lane_metrics.instrument_model != \"MiSeq\" ) or (iseq_product_metrics.id_run in (%s)) ) and \\\n",
    "        iseq_run_status.id_run_status_dict = iseq_run_status_dict.id_run_status_dict and \\\n",
    "        study.faculty_sponsor = \"Dominic Kwiatkowski\" and \\\n",
    "        ( ( sample.taxon_id in (%s) ) or ' % (\n",
    "            ', '.join([str(x) for x in miseq_runs_to_allow]),\n",
    "            ', '.join([str(x) for x in taxon_ids])\n",
    "        )\n",
    "    sql_query = sql_query + ' or '.join(\n",
    "        df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)].apply(\n",
    "            lambda x: '(iseq_product_metrics.id_run=\"%s\" and iseq_product_metrics.position=\"%s\" and iseq_product_metrics.tag_index=\"%s\")' % (x['id_run'], x['position'], x['tag_index']),\n",
    "            1\n",
    "        )\n",
    "    )\n",
    "    sql_query = sql_query + ')'\n",
    "#         (iseq_flowcell.manual_qc = 1 or iseq_flowcell.manual_qc is null) and \\\n",
    "    \n",
    "    df_return = pd.read_sql(sql_query, conn)\n",
    "\n",
    "    # Replace missing taxon_id with -1 (can't have missing int values in pandas Series)\n",
    "    df_return['taxon_id'] = df_return['taxon_id'].fillna(-1).astype('int32')\n",
    "\n",
    "    # Determine file name in iRods\n",
    "    df_return['irods_filename'] = df_return.apply(set_irods_name, 1)\n",
    "    df_return.set_index('irods_filename', inplace=True)\n",
    "\n",
    "    return(df_return)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_name</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>sample</th>\n",
       "      <th>name</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>num_reads</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>paired_read</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1337_1_nonhuman.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>AK121_D80 300</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>1337</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2745194</td>\n",
       "      <td>60.35</td>\n",
       "      <td>IL29</td>\n",
       "      <td>HK</td>\n",
       "      <td>1</td>\n",
       "      <td>2008-09-24 15:54:04</td>\n",
       "      <td>1</td>\n",
       "      <td>300.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>37</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1337_2_nonhuman.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>AK121_D80 200</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>1337</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4310986</td>\n",
       "      <td>58.68</td>\n",
       "      <td>IL29</td>\n",
       "      <td>HK</td>\n",
       "      <td>1</td>\n",
       "      <td>2008-09-24 15:54:04</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>37</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1337_3_nonhuman.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>AK121_D80 200</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>1337</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14488586</td>\n",
       "      <td>57.19</td>\n",
       "      <td>IL29</td>\n",
       "      <td>HK</td>\n",
       "      <td>1</td>\n",
       "      <td>2008-09-24 15:54:04</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>200.0</td>\n",
       "      <td>37</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   study_name study_lims  \\\n",
       "irods_filename                                                             \n",
       "1337_1_nonhuman.bam  Plasmodium falciparum genome variation 1        578   \n",
       "1337_2_nonhuman.bam  Plasmodium falciparum genome variation 1        578   \n",
       "1337_3_nonhuman.bam  Plasmodium falciparum genome variation 1        578   \n",
       "\n",
       "                    sample           name sanger_sample_id  taxon_id  id_run  \\\n",
       "irods_filename                                                                 \n",
       "1337_1_nonhuman.bam   None  AK121_D80 300             None      5833    1337   \n",
       "1337_2_nonhuman.bam   None  AK121_D80 200             None      5833    1337   \n",
       "1337_3_nonhuman.bam   None  AK121_D80 200             None      5833    1337   \n",
       "\n",
       "                     position  tag_index  num_reads  human_percent_mapped  \\\n",
       "irods_filename                                                              \n",
       "1337_1_nonhuman.bam         1        NaN    2745194                 60.35   \n",
       "1337_2_nonhuman.bam         2        NaN    4310986                 58.68   \n",
       "1337_3_nonhuman.bam         3        NaN   14488586                 57.19   \n",
       "\n",
       "                    instrument_name instrument_model  paired_read  \\\n",
       "irods_filename                                                      \n",
       "1337_1_nonhuman.bam            IL29               HK            1   \n",
       "1337_2_nonhuman.bam            IL29               HK            1   \n",
       "1337_3_nonhuman.bam            IL29               HK            1   \n",
       "\n",
       "                            qc_complete  manual_qc  \\\n",
       "irods_filename                                       \n",
       "1337_1_nonhuman.bam 2008-09-24 15:54:04          1   \n",
       "1337_2_nonhuman.bam 2008-09-24 15:54:04          1   \n",
       "1337_3_nonhuman.bam 2008-09-24 15:54:04          1   \n",
       "\n",
       "                     requested_insert_size_from  requested_insert_size_to  \\\n",
       "irods_filename                                                              \n",
       "1337_1_nonhuman.bam                       300.0                     300.0   \n",
       "1337_2_nonhuman.bam                       200.0                     200.0   \n",
       "1337_3_nonhuman.bam                       200.0                     200.0   \n",
       "\n",
       "                     forward_read_length  description  \n",
       "irods_filename                                         \n",
       "1337_1_nonhuman.bam                   37  qc complete  \n",
       "1337_2_nonhuman.bam                   37  qc complete  \n",
       "1337_3_nonhuman.bam                   37  qc complete  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mlwh = create_build_manifest()\n",
    "df_mlwh[0:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_name</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>sample</th>\n",
       "      <th>name</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>num_reads</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>paired_read</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#1.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0477-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15195722</td>\n",
       "      <td>89.33</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#2.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0479-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>22994194</td>\n",
       "      <td>90.78</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#3.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0473-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>18145404</td>\n",
       "      <td>80.67</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#4.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0522-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>4.0</td>\n",
       "      <td>22990098</td>\n",
       "      <td>74.86</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#5.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0463-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>5.0</td>\n",
       "      <td>15795786</td>\n",
       "      <td>85.70</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#6.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0531-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>6.0</td>\n",
       "      <td>16618420</td>\n",
       "      <td>81.78</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#7.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0512-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20197362</td>\n",
       "      <td>87.67</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#8.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0454-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>8.0</td>\n",
       "      <td>19465296</td>\n",
       "      <td>91.49</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#9.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0472-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>9.0</td>\n",
       "      <td>14194180</td>\n",
       "      <td>96.48</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#10.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0504-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>10.0</td>\n",
       "      <td>23747352</td>\n",
       "      <td>79.99</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#11.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0464-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>11.0</td>\n",
       "      <td>22377650</td>\n",
       "      <td>74.10</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_3_nonhuman#12.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PF0456-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>3</td>\n",
       "      <td>12.0</td>\n",
       "      <td>18611384</td>\n",
       "      <td>92.69</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#13.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0001-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>13.0</td>\n",
       "      <td>13525420</td>\n",
       "      <td>99.37</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#14.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0002-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>14.0</td>\n",
       "      <td>21008120</td>\n",
       "      <td>79.66</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#15.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0003-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>15.0</td>\n",
       "      <td>14580386</td>\n",
       "      <td>92.25</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#16.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0004-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>16.0</td>\n",
       "      <td>25527232</td>\n",
       "      <td>98.64</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#17.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0005-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>17.0</td>\n",
       "      <td>13915156</td>\n",
       "      <td>94.43</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#18.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0006-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>18.0</td>\n",
       "      <td>14123162</td>\n",
       "      <td>99.09</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#19.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0007-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>19.0</td>\n",
       "      <td>15825566</td>\n",
       "      <td>99.38</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#20.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0008-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>20.0</td>\n",
       "      <td>12100760</td>\n",
       "      <td>99.27</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#21.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0009-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>21.0</td>\n",
       "      <td>16239972</td>\n",
       "      <td>98.70</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#22.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0010-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>22.0</td>\n",
       "      <td>14206554</td>\n",
       "      <td>99.29</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#23.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0011-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>23.0</td>\n",
       "      <td>24807566</td>\n",
       "      <td>96.47</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_4_nonhuman#24.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PZ0012_C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>4</td>\n",
       "      <td>24.0</td>\n",
       "      <td>974524</td>\n",
       "      <td>99.50</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#25.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0493-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>25.0</td>\n",
       "      <td>2561942</td>\n",
       "      <td>75.94</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#26.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0494-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>26.0</td>\n",
       "      <td>33483224</td>\n",
       "      <td>88.06</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#27.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0495-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>27.0</td>\n",
       "      <td>20744206</td>\n",
       "      <td>77.76</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#28.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0496-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>28.0</td>\n",
       "      <td>26696498</td>\n",
       "      <td>75.53</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#29.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0497-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>29.0</td>\n",
       "      <td>18872142</td>\n",
       "      <td>66.63</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#30.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0498-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>30.0</td>\n",
       "      <td>17575026</td>\n",
       "      <td>67.07</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#31.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0499-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>31.0</td>\n",
       "      <td>25563546</td>\n",
       "      <td>68.43</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#32.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0500-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>32.0</td>\n",
       "      <td>18842286</td>\n",
       "      <td>72.96</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#33.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0501-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>33.0</td>\n",
       "      <td>25921218</td>\n",
       "      <td>73.40</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#34.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PG0120-CW</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>34.0</td>\n",
       "      <td>19275848</td>\n",
       "      <td>96.78</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_5_nonhuman#35.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PG0121-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>5</td>\n",
       "      <td>35.0</td>\n",
       "      <td>25600178</td>\n",
       "      <td>98.65</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#1.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0540-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4969844</td>\n",
       "      <td>99.42</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#2.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>Tr5E</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>2.0</td>\n",
       "      <td>52856</td>\n",
       "      <td>93.42</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#3.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>Tr3E_1</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>3.0</td>\n",
       "      <td>353834</td>\n",
       "      <td>98.46</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#4.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0545-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>4.0</td>\n",
       "      <td>13389882</td>\n",
       "      <td>99.41</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#5.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0535-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2341808</td>\n",
       "      <td>99.48</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#6.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0526-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4612274</td>\n",
       "      <td>99.47</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#7.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0530-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7985930</td>\n",
       "      <td>99.43</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#8.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0532-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>8.0</td>\n",
       "      <td>14623356</td>\n",
       "      <td>99.40</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#9.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0546-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>9.0</td>\n",
       "      <td>3781224</td>\n",
       "      <td>99.44</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#10.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>Tr5E_1</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>10.0</td>\n",
       "      <td>68518</td>\n",
       "      <td>96.61</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#11.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>Tr3E</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>11.0</td>\n",
       "      <td>289576</td>\n",
       "      <td>94.41</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#12.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0529-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>12.0</td>\n",
       "      <td>5091710</td>\n",
       "      <td>99.44</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#13.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0527-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>13.0</td>\n",
       "      <td>27655836</td>\n",
       "      <td>99.32</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#14.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>PH0543-C</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3000612</td>\n",
       "      <td>99.46</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      study_name study_lims  \\\n",
       "irods_filename                                                                \n",
       "6901_3_nonhuman#1.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#2.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#3.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#4.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#5.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#6.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#7.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#8.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#9.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#10.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#11.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_3_nonhuman#12.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#13.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#14.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#15.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#16.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#17.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#18.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#19.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#20.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#21.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#22.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#23.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_4_nonhuman#24.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#25.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#26.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#27.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#28.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#29.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#30.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#31.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#32.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#33.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#34.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_5_nonhuman#35.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#1.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#2.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#3.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#4.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#5.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#6.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#7.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#8.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#9.bam   Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#10.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#11.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#12.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#13.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#14.bam  Plasmodium falciparum genome variation 1        578   \n",
       "\n",
       "                       sample       name sanger_sample_id  taxon_id  id_run  \\\n",
       "irods_filename                                                                \n",
       "6901_3_nonhuman#1.bam    None   PF0477-C             None      5833    6901   \n",
       "6901_3_nonhuman#2.bam    None   PF0479-C             None      5833    6901   \n",
       "6901_3_nonhuman#3.bam    None   PF0473-C             None      5833    6901   \n",
       "6901_3_nonhuman#4.bam    None   PF0522-C             None      5833    6901   \n",
       "6901_3_nonhuman#5.bam    None   PF0463-C             None      5833    6901   \n",
       "6901_3_nonhuman#6.bam    None   PF0531-C             None      5833    6901   \n",
       "6901_3_nonhuman#7.bam    None   PF0512-C             None      5833    6901   \n",
       "6901_3_nonhuman#8.bam    None   PF0454-C             None      5833    6901   \n",
       "6901_3_nonhuman#9.bam    None   PF0472-C             None      5833    6901   \n",
       "6901_3_nonhuman#10.bam   None   PF0504-C             None      5833    6901   \n",
       "6901_3_nonhuman#11.bam   None   PF0464-C             None      5833    6901   \n",
       "6901_3_nonhuman#12.bam   None   PF0456-C             None      5833    6901   \n",
       "6901_4_nonhuman#13.bam   None  PZ0001-CW             None      5833    6901   \n",
       "6901_4_nonhuman#14.bam   None  PZ0002-CW             None      5833    6901   \n",
       "6901_4_nonhuman#15.bam   None  PZ0003-CW             None      5833    6901   \n",
       "6901_4_nonhuman#16.bam   None  PZ0004-CW             None      5833    6901   \n",
       "6901_4_nonhuman#17.bam   None  PZ0005-CW             None      5833    6901   \n",
       "6901_4_nonhuman#18.bam   None  PZ0006-CW             None      5833    6901   \n",
       "6901_4_nonhuman#19.bam   None   PZ0007-C             None      5833    6901   \n",
       "6901_4_nonhuman#20.bam   None  PZ0008-CW             None      5833    6901   \n",
       "6901_4_nonhuman#21.bam   None  PZ0009-CW             None      5833    6901   \n",
       "6901_4_nonhuman#22.bam   None  PZ0010-CW             None      5833    6901   \n",
       "6901_4_nonhuman#23.bam   None  PZ0011-CW             None      5833    6901   \n",
       "6901_4_nonhuman#24.bam   None   PZ0012_C             None      5833    6901   \n",
       "6901_5_nonhuman#25.bam   None  PH0493-CW             None      5833    6901   \n",
       "6901_5_nonhuman#26.bam   None  PH0494-CW             None      5833    6901   \n",
       "6901_5_nonhuman#27.bam   None  PH0495-CW             None      5833    6901   \n",
       "6901_5_nonhuman#28.bam   None  PH0496-CW             None      5833    6901   \n",
       "6901_5_nonhuman#29.bam   None  PH0497-CW             None      5833    6901   \n",
       "6901_5_nonhuman#30.bam   None  PH0498-CW             None      5833    6901   \n",
       "6901_5_nonhuman#31.bam   None  PH0499-CW             None      5833    6901   \n",
       "6901_5_nonhuman#32.bam   None  PH0500-CW             None      5833    6901   \n",
       "6901_5_nonhuman#33.bam   None  PH0501-CW             None      5833    6901   \n",
       "6901_5_nonhuman#34.bam   None  PG0120-CW             None      5833    6901   \n",
       "6901_5_nonhuman#35.bam   None   PG0121-C             None      5833    6901   \n",
       "6901_8_nonhuman#1.bam    None   PH0540-C             None      5833    6901   \n",
       "6901_8_nonhuman#2.bam    None       Tr5E             None      5833    6901   \n",
       "6901_8_nonhuman#3.bam    None     Tr3E_1             None      5833    6901   \n",
       "6901_8_nonhuman#4.bam    None   PH0545-C             None      5833    6901   \n",
       "6901_8_nonhuman#5.bam    None   PH0535-C             None      5833    6901   \n",
       "6901_8_nonhuman#6.bam    None   PH0526-C             None      5833    6901   \n",
       "6901_8_nonhuman#7.bam    None   PH0530-C             None      5833    6901   \n",
       "6901_8_nonhuman#8.bam    None   PH0532-C             None      5833    6901   \n",
       "6901_8_nonhuman#9.bam    None   PH0546-C             None      5833    6901   \n",
       "6901_8_nonhuman#10.bam   None     Tr5E_1             None      5833    6901   \n",
       "6901_8_nonhuman#11.bam   None       Tr3E             None      5833    6901   \n",
       "6901_8_nonhuman#12.bam   None   PH0529-C             None      5833    6901   \n",
       "6901_8_nonhuman#13.bam   None   PH0527-C             None      5833    6901   \n",
       "6901_8_nonhuman#14.bam   None   PH0543-C             None      5833    6901   \n",
       "\n",
       "                        position  tag_index  num_reads  human_percent_mapped  \\\n",
       "irods_filename                                                                 \n",
       "6901_3_nonhuman#1.bam          3        1.0   15195722                 89.33   \n",
       "6901_3_nonhuman#2.bam          3        2.0   22994194                 90.78   \n",
       "6901_3_nonhuman#3.bam          3        3.0   18145404                 80.67   \n",
       "6901_3_nonhuman#4.bam          3        4.0   22990098                 74.86   \n",
       "6901_3_nonhuman#5.bam          3        5.0   15795786                 85.70   \n",
       "6901_3_nonhuman#6.bam          3        6.0   16618420                 81.78   \n",
       "6901_3_nonhuman#7.bam          3        7.0   20197362                 87.67   \n",
       "6901_3_nonhuman#8.bam          3        8.0   19465296                 91.49   \n",
       "6901_3_nonhuman#9.bam          3        9.0   14194180                 96.48   \n",
       "6901_3_nonhuman#10.bam         3       10.0   23747352                 79.99   \n",
       "6901_3_nonhuman#11.bam         3       11.0   22377650                 74.10   \n",
       "6901_3_nonhuman#12.bam         3       12.0   18611384                 92.69   \n",
       "6901_4_nonhuman#13.bam         4       13.0   13525420                 99.37   \n",
       "6901_4_nonhuman#14.bam         4       14.0   21008120                 79.66   \n",
       "6901_4_nonhuman#15.bam         4       15.0   14580386                 92.25   \n",
       "6901_4_nonhuman#16.bam         4       16.0   25527232                 98.64   \n",
       "6901_4_nonhuman#17.bam         4       17.0   13915156                 94.43   \n",
       "6901_4_nonhuman#18.bam         4       18.0   14123162                 99.09   \n",
       "6901_4_nonhuman#19.bam         4       19.0   15825566                 99.38   \n",
       "6901_4_nonhuman#20.bam         4       20.0   12100760                 99.27   \n",
       "6901_4_nonhuman#21.bam         4       21.0   16239972                 98.70   \n",
       "6901_4_nonhuman#22.bam         4       22.0   14206554                 99.29   \n",
       "6901_4_nonhuman#23.bam         4       23.0   24807566                 96.47   \n",
       "6901_4_nonhuman#24.bam         4       24.0     974524                 99.50   \n",
       "6901_5_nonhuman#25.bam         5       25.0    2561942                 75.94   \n",
       "6901_5_nonhuman#26.bam         5       26.0   33483224                 88.06   \n",
       "6901_5_nonhuman#27.bam         5       27.0   20744206                 77.76   \n",
       "6901_5_nonhuman#28.bam         5       28.0   26696498                 75.53   \n",
       "6901_5_nonhuman#29.bam         5       29.0   18872142                 66.63   \n",
       "6901_5_nonhuman#30.bam         5       30.0   17575026                 67.07   \n",
       "6901_5_nonhuman#31.bam         5       31.0   25563546                 68.43   \n",
       "6901_5_nonhuman#32.bam         5       32.0   18842286                 72.96   \n",
       "6901_5_nonhuman#33.bam         5       33.0   25921218                 73.40   \n",
       "6901_5_nonhuman#34.bam         5       34.0   19275848                 96.78   \n",
       "6901_5_nonhuman#35.bam         5       35.0   25600178                 98.65   \n",
       "6901_8_nonhuman#1.bam          8        1.0    4969844                 99.42   \n",
       "6901_8_nonhuman#2.bam          8        2.0      52856                 93.42   \n",
       "6901_8_nonhuman#3.bam          8        3.0     353834                 98.46   \n",
       "6901_8_nonhuman#4.bam          8        4.0   13389882                 99.41   \n",
       "6901_8_nonhuman#5.bam          8        5.0    2341808                 99.48   \n",
       "6901_8_nonhuman#6.bam          8        6.0    4612274                 99.47   \n",
       "6901_8_nonhuman#7.bam          8        7.0    7985930                 99.43   \n",
       "6901_8_nonhuman#8.bam          8        8.0   14623356                 99.40   \n",
       "6901_8_nonhuman#9.bam          8        9.0    3781224                 99.44   \n",
       "6901_8_nonhuman#10.bam         8       10.0      68518                 96.61   \n",
       "6901_8_nonhuman#11.bam         8       11.0     289576                 94.41   \n",
       "6901_8_nonhuman#12.bam         8       12.0    5091710                 99.44   \n",
       "6901_8_nonhuman#13.bam         8       13.0   27655836                 99.32   \n",
       "6901_8_nonhuman#14.bam         8       14.0    3000612                 99.46   \n",
       "\n",
       "                       instrument_name instrument_model  paired_read  \\\n",
       "irods_filename                                                         \n",
       "6901_3_nonhuman#1.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#2.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#3.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#4.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#5.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#6.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#7.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#8.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#9.bam             HS13            HiSeq            1   \n",
       "6901_3_nonhuman#10.bam            HS13            HiSeq            1   \n",
       "6901_3_nonhuman#11.bam            HS13            HiSeq            1   \n",
       "6901_3_nonhuman#12.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#13.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#14.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#15.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#16.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#17.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#18.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#19.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#20.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#21.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#22.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#23.bam            HS13            HiSeq            1   \n",
       "6901_4_nonhuman#24.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#25.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#26.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#27.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#28.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#29.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#30.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#31.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#32.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#33.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#34.bam            HS13            HiSeq            1   \n",
       "6901_5_nonhuman#35.bam            HS13            HiSeq            1   \n",
       "6901_8_nonhuman#1.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#2.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#3.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#4.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#5.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#6.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#7.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#8.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#9.bam             HS13            HiSeq            1   \n",
       "6901_8_nonhuman#10.bam            HS13            HiSeq            1   \n",
       "6901_8_nonhuman#11.bam            HS13            HiSeq            1   \n",
       "6901_8_nonhuman#12.bam            HS13            HiSeq            1   \n",
       "6901_8_nonhuman#13.bam            HS13            HiSeq            1   \n",
       "6901_8_nonhuman#14.bam            HS13            HiSeq            1   \n",
       "\n",
       "                               qc_complete  manual_qc  \\\n",
       "irods_filename                                          \n",
       "6901_3_nonhuman#1.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#2.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#3.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#4.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#5.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#6.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#7.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#8.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#9.bam  2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#10.bam 2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#11.bam 2011-11-01 12:52:25          1   \n",
       "6901_3_nonhuman#12.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#13.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#14.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#15.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#16.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#17.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#18.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#19.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#20.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#21.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#22.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#23.bam 2011-11-01 12:52:25          1   \n",
       "6901_4_nonhuman#24.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#25.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#26.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#27.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#28.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#29.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#30.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#31.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#32.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#33.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#34.bam 2011-11-01 12:52:25          1   \n",
       "6901_5_nonhuman#35.bam 2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#1.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#2.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#3.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#4.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#5.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#6.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#7.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#8.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#9.bam  2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#10.bam 2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#11.bam 2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#12.bam 2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#13.bam 2011-11-01 12:52:25          1   \n",
       "6901_8_nonhuman#14.bam 2011-11-01 12:52:25          1   \n",
       "\n",
       "                        requested_insert_size_from  requested_insert_size_to  \\\n",
       "irods_filename                                                                 \n",
       "6901_3_nonhuman#1.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#2.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#3.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#4.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#5.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#6.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#7.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#8.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#9.bam                        200.0                     300.0   \n",
       "6901_3_nonhuman#10.bam                       200.0                     300.0   \n",
       "6901_3_nonhuman#11.bam                       200.0                     300.0   \n",
       "6901_3_nonhuman#12.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#13.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#14.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#15.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#16.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#17.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#18.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#19.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#20.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#21.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#22.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#23.bam                       200.0                     300.0   \n",
       "6901_4_nonhuman#24.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#25.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#26.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#27.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#28.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#29.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#30.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#31.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#32.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#33.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#34.bam                       200.0                     300.0   \n",
       "6901_5_nonhuman#35.bam                       200.0                     300.0   \n",
       "6901_8_nonhuman#1.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#2.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#3.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#4.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#5.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#6.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#7.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#8.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#9.bam                        200.0                     300.0   \n",
       "6901_8_nonhuman#10.bam                       200.0                     300.0   \n",
       "6901_8_nonhuman#11.bam                       200.0                     300.0   \n",
       "6901_8_nonhuman#12.bam                       200.0                     300.0   \n",
       "6901_8_nonhuman#13.bam                       200.0                     300.0   \n",
       "6901_8_nonhuman#14.bam                       200.0                     300.0   \n",
       "\n",
       "                        forward_read_length  description  \n",
       "irods_filename                                            \n",
       "6901_3_nonhuman#1.bam                   100  qc complete  \n",
       "6901_3_nonhuman#2.bam                   100  qc complete  \n",
       "6901_3_nonhuman#3.bam                   100  qc complete  \n",
       "6901_3_nonhuman#4.bam                   100  qc complete  \n",
       "6901_3_nonhuman#5.bam                   100  qc complete  \n",
       "6901_3_nonhuman#6.bam                   100  qc complete  \n",
       "6901_3_nonhuman#7.bam                   100  qc complete  \n",
       "6901_3_nonhuman#8.bam                   100  qc complete  \n",
       "6901_3_nonhuman#9.bam                   100  qc complete  \n",
       "6901_3_nonhuman#10.bam                  100  qc complete  \n",
       "6901_3_nonhuman#11.bam                  100  qc complete  \n",
       "6901_3_nonhuman#12.bam                  100  qc complete  \n",
       "6901_4_nonhuman#13.bam                  100  qc complete  \n",
       "6901_4_nonhuman#14.bam                  100  qc complete  \n",
       "6901_4_nonhuman#15.bam                  100  qc complete  \n",
       "6901_4_nonhuman#16.bam                  100  qc complete  \n",
       "6901_4_nonhuman#17.bam                  100  qc complete  \n",
       "6901_4_nonhuman#18.bam                  100  qc complete  \n",
       "6901_4_nonhuman#19.bam                  100  qc complete  \n",
       "6901_4_nonhuman#20.bam                  100  qc complete  \n",
       "6901_4_nonhuman#21.bam                  100  qc complete  \n",
       "6901_4_nonhuman#22.bam                  100  qc complete  \n",
       "6901_4_nonhuman#23.bam                  100  qc complete  \n",
       "6901_4_nonhuman#24.bam                  100  qc complete  \n",
       "6901_5_nonhuman#25.bam                  100  qc complete  \n",
       "6901_5_nonhuman#26.bam                  100  qc complete  \n",
       "6901_5_nonhuman#27.bam                  100  qc complete  \n",
       "6901_5_nonhuman#28.bam                  100  qc complete  \n",
       "6901_5_nonhuman#29.bam                  100  qc complete  \n",
       "6901_5_nonhuman#30.bam                  100  qc complete  \n",
       "6901_5_nonhuman#31.bam                  100  qc complete  \n",
       "6901_5_nonhuman#32.bam                  100  qc complete  \n",
       "6901_5_nonhuman#33.bam                  100  qc complete  \n",
       "6901_5_nonhuman#34.bam                  100  qc complete  \n",
       "6901_5_nonhuman#35.bam                  100  qc complete  \n",
       "6901_8_nonhuman#1.bam                   100  qc complete  \n",
       "6901_8_nonhuman#2.bam                   100  qc complete  \n",
       "6901_8_nonhuman#3.bam                   100  qc complete  \n",
       "6901_8_nonhuman#4.bam                   100  qc complete  \n",
       "6901_8_nonhuman#5.bam                   100  qc complete  \n",
       "6901_8_nonhuman#6.bam                   100  qc complete  \n",
       "6901_8_nonhuman#7.bam                   100  qc complete  \n",
       "6901_8_nonhuman#8.bam                   100  qc complete  \n",
       "6901_8_nonhuman#9.bam                   100  qc complete  \n",
       "6901_8_nonhuman#10.bam                  100  qc complete  \n",
       "6901_8_nonhuman#11.bam                  100  qc complete  \n",
       "6901_8_nonhuman#12.bam                  100  qc complete  \n",
       "6901_8_nonhuman#13.bam                  100  qc complete  \n",
       "6901_8_nonhuman#14.bam                  100  qc complete  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mlwh.loc[df_mlwh['id_run'] == 6901]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'6901_8_nonhuman#1.bam' in df_mlwh.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'6901_8_nonhuman#2.bam' in df_mlwh.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_name</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>sample</th>\n",
       "      <th>name</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>num_reads</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>paired_read</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12135_1#31.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0138-C</td>\n",
       "      <td>2876STDY5684764</td>\n",
       "      <td>2876STDY5684764</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>1</td>\n",
       "      <td>31.0</td>\n",
       "      <td>2512866</td>\n",
       "      <td>99.55</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_1#32.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0139-C</td>\n",
       "      <td>2876STDY5684765</td>\n",
       "      <td>2876STDY5684765</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>1</td>\n",
       "      <td>32.0</td>\n",
       "      <td>2193340</td>\n",
       "      <td>99.22</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_1#33.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0140-C</td>\n",
       "      <td>2876STDY5684766</td>\n",
       "      <td>2876STDY5684766</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>1</td>\n",
       "      <td>33.0</td>\n",
       "      <td>2098154</td>\n",
       "      <td>99.49</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_1#34.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0141-C</td>\n",
       "      <td>2876STDY5684767</td>\n",
       "      <td>2876STDY5684767</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>1</td>\n",
       "      <td>34.0</td>\n",
       "      <td>2209976</td>\n",
       "      <td>99.53</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_1#35.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0142-C</td>\n",
       "      <td>2876STDY5684768</td>\n",
       "      <td>2876STDY5684768</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>1</td>\n",
       "      <td>35.0</td>\n",
       "      <td>3432242</td>\n",
       "      <td>99.52</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_1#36.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0143-C</td>\n",
       "      <td>2876STDY5684769</td>\n",
       "      <td>2876STDY5684769</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>1</td>\n",
       "      <td>36.0</td>\n",
       "      <td>2457978</td>\n",
       "      <td>99.52</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_2#37.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0144-C</td>\n",
       "      <td>2876STDY5684770</td>\n",
       "      <td>2876STDY5684770</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>2</td>\n",
       "      <td>37.0</td>\n",
       "      <td>2077158</td>\n",
       "      <td>99.52</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_2#38.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0145-C</td>\n",
       "      <td>2876STDY5684771</td>\n",
       "      <td>2876STDY5684771</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>2</td>\n",
       "      <td>38.0</td>\n",
       "      <td>2133750</td>\n",
       "      <td>99.49</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_2#39.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0146-C</td>\n",
       "      <td>2876STDY5684772</td>\n",
       "      <td>2876STDY5684772</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>2</td>\n",
       "      <td>39.0</td>\n",
       "      <td>2447318</td>\n",
       "      <td>99.53</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_2#40.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0147-C</td>\n",
       "      <td>2876STDY5684773</td>\n",
       "      <td>2876STDY5684773</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>2</td>\n",
       "      <td>40.0</td>\n",
       "      <td>2246078</td>\n",
       "      <td>99.51</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_2#41.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0148-C</td>\n",
       "      <td>2876STDY5684774</td>\n",
       "      <td>2876STDY5684774</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>2</td>\n",
       "      <td>41.0</td>\n",
       "      <td>1850042</td>\n",
       "      <td>99.52</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_2#42.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0149-C</td>\n",
       "      <td>2876STDY5684775</td>\n",
       "      <td>2876STDY5684775</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>2</td>\n",
       "      <td>42.0</td>\n",
       "      <td>2018266</td>\n",
       "      <td>99.51</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_3#43.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0150-C</td>\n",
       "      <td>2876STDY5684776</td>\n",
       "      <td>2876STDY5684776</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>3</td>\n",
       "      <td>43.0</td>\n",
       "      <td>2145746</td>\n",
       "      <td>99.55</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_3#44.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0151-C</td>\n",
       "      <td>2876STDY5684777</td>\n",
       "      <td>2876STDY5684777</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>3</td>\n",
       "      <td>44.0</td>\n",
       "      <td>1995634</td>\n",
       "      <td>99.58</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_3#45.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0152-C</td>\n",
       "      <td>2876STDY5684778</td>\n",
       "      <td>2876STDY5684778</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>3</td>\n",
       "      <td>45.0</td>\n",
       "      <td>2773312</td>\n",
       "      <td>99.54</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_3#46.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0153-C</td>\n",
       "      <td>2876STDY5684779</td>\n",
       "      <td>2876STDY5684779</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>3</td>\n",
       "      <td>46.0</td>\n",
       "      <td>2076032</td>\n",
       "      <td>99.54</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_3#47.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0154-C</td>\n",
       "      <td>2876STDY5684780</td>\n",
       "      <td>2876STDY5684780</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>3</td>\n",
       "      <td>47.0</td>\n",
       "      <td>2169918</td>\n",
       "      <td>99.57</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_3#48.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0155-C</td>\n",
       "      <td>2876STDY5684781</td>\n",
       "      <td>2876STDY5684781</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>3</td>\n",
       "      <td>48.0</td>\n",
       "      <td>2070780</td>\n",
       "      <td>99.55</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_4#49.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0156-C</td>\n",
       "      <td>2876STDY5684782</td>\n",
       "      <td>2876STDY5684782</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>4</td>\n",
       "      <td>49.0</td>\n",
       "      <td>2899830</td>\n",
       "      <td>99.56</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_4#50.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0157-C</td>\n",
       "      <td>2876STDY5684783</td>\n",
       "      <td>2876STDY5684783</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>4</td>\n",
       "      <td>50.0</td>\n",
       "      <td>2115480</td>\n",
       "      <td>99.53</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_4#51.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0158-C</td>\n",
       "      <td>2876STDY5684784</td>\n",
       "      <td>2876STDY5684784</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>4</td>\n",
       "      <td>51.0</td>\n",
       "      <td>2179082</td>\n",
       "      <td>99.58</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_4#52.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0159-C</td>\n",
       "      <td>2876STDY5684785</td>\n",
       "      <td>2876STDY5684785</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>4</td>\n",
       "      <td>52.0</td>\n",
       "      <td>2224284</td>\n",
       "      <td>99.59</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_4#53.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0160-C</td>\n",
       "      <td>2876STDY5684786</td>\n",
       "      <td>2876STDY5684786</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>4</td>\n",
       "      <td>53.0</td>\n",
       "      <td>1923730</td>\n",
       "      <td>99.54</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_4#54.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0161-C</td>\n",
       "      <td>2876STDY5684787</td>\n",
       "      <td>2876STDY5684787</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>4</td>\n",
       "      <td>54.0</td>\n",
       "      <td>2508814</td>\n",
       "      <td>99.54</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_5#55.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PC0162-C</td>\n",
       "      <td>2876STDY5684788</td>\n",
       "      <td>2876STDY5684788</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>5</td>\n",
       "      <td>55.0</td>\n",
       "      <td>3000932</td>\n",
       "      <td>99.54</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_5#56.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PE0092-C</td>\n",
       "      <td>2876STDY5684789</td>\n",
       "      <td>2876STDY5684789</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>5</td>\n",
       "      <td>56.0</td>\n",
       "      <td>10818680</td>\n",
       "      <td>99.36</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_5#57.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PM0129-C</td>\n",
       "      <td>2876STDY5684790</td>\n",
       "      <td>2876STDY5684790</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>5</td>\n",
       "      <td>57.0</td>\n",
       "      <td>6512292</td>\n",
       "      <td>99.45</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_5#58.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PM0077-C</td>\n",
       "      <td>2876STDY5684791</td>\n",
       "      <td>2876STDY5684791</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>5</td>\n",
       "      <td>58.0</td>\n",
       "      <td>12493066</td>\n",
       "      <td>99.49</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_5#59.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PF0812-C</td>\n",
       "      <td>2876STDY5684792</td>\n",
       "      <td>2876STDY5684792</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>5</td>\n",
       "      <td>59.0</td>\n",
       "      <td>3306334</td>\n",
       "      <td>99.04</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_5#60.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PE0093-C</td>\n",
       "      <td>2876STDY5684793</td>\n",
       "      <td>2876STDY5684793</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>5</td>\n",
       "      <td>60.0</td>\n",
       "      <td>7780290</td>\n",
       "      <td>99.49</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_6#61.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PE0108-C</td>\n",
       "      <td>2876STDY5684794</td>\n",
       "      <td>2876STDY5684794</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>6</td>\n",
       "      <td>61.0</td>\n",
       "      <td>6752670</td>\n",
       "      <td>99.53</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_6#62.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PM0161-C</td>\n",
       "      <td>2876STDY5684795</td>\n",
       "      <td>2876STDY5684795</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>6</td>\n",
       "      <td>62.0</td>\n",
       "      <td>14254540</td>\n",
       "      <td>99.20</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_6#63.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PF0266-C</td>\n",
       "      <td>2876STDY5684796</td>\n",
       "      <td>2876STDY5684796</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>6</td>\n",
       "      <td>63.0</td>\n",
       "      <td>6991004</td>\n",
       "      <td>99.52</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_6#64.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PH0963-C</td>\n",
       "      <td>2876STDY5684797</td>\n",
       "      <td>2876STDY5684797</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>6</td>\n",
       "      <td>64.0</td>\n",
       "      <td>6460296</td>\n",
       "      <td>99.51</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_6#65.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PH0963-Cx</td>\n",
       "      <td>2876STDY5684798</td>\n",
       "      <td>2876STDY5684798</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>6</td>\n",
       "      <td>65.0</td>\n",
       "      <td>6109270</td>\n",
       "      <td>99.48</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_6#66.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>PD0338-C</td>\n",
       "      <td>2876STDY5684799</td>\n",
       "      <td>2876STDY5684799</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>6</td>\n",
       "      <td>66.0</td>\n",
       "      <td>4663282</td>\n",
       "      <td>99.44</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#67.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0196-C</td>\n",
       "      <td>2876STDY5684800</td>\n",
       "      <td>2876STDY5684800</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>7</td>\n",
       "      <td>67.0</td>\n",
       "      <td>3498108</td>\n",
       "      <td>99.57</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#68.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0008-C</td>\n",
       "      <td>2876STDY5684801</td>\n",
       "      <td>2876STDY5684801</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>7</td>\n",
       "      <td>68.0</td>\n",
       "      <td>6843374</td>\n",
       "      <td>99.50</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#69.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0145-C</td>\n",
       "      <td>2876STDY5684802</td>\n",
       "      <td>2876STDY5684802</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>7</td>\n",
       "      <td>69.0</td>\n",
       "      <td>3067626</td>\n",
       "      <td>99.54</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#70.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0022-C</td>\n",
       "      <td>2876STDY5684803</td>\n",
       "      <td>2876STDY5684803</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>7</td>\n",
       "      <td>70.0</td>\n",
       "      <td>16264426</td>\n",
       "      <td>99.49</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_7#71.cram</th>\n",
       "      <td>Test sequencing of high level human contaminat...</td>\n",
       "      <td>2876</td>\n",
       "      <td>QE0212-C</td>\n",
       "      <td>2876STDY5684804</td>\n",
       "      <td>2876STDY5684804</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>7</td>\n",
       "      <td>71.0</td>\n",
       "      <td>5631900</td>\n",
       "      <td>99.54</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_8#73.cram</th>\n",
       "      <td>1102-PF-MG-RANDRIANARIVELOJOSIA</td>\n",
       "      <td>2886</td>\n",
       "      <td>QM0005-C</td>\n",
       "      <td>2886STDY5684862</td>\n",
       "      <td>2886STDY5684862</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>8</td>\n",
       "      <td>73.0</td>\n",
       "      <td>32819612</td>\n",
       "      <td>99.17</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_8#74.cram</th>\n",
       "      <td>1102-PF-MG-RANDRIANARIVELOJOSIA</td>\n",
       "      <td>2886</td>\n",
       "      <td>QM0006-C</td>\n",
       "      <td>2886STDY5684870</td>\n",
       "      <td>2886STDY5684870</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>8</td>\n",
       "      <td>74.0</td>\n",
       "      <td>29713518</td>\n",
       "      <td>98.86</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_8#75.cram</th>\n",
       "      <td>1102-PF-MG-RANDRIANARIVELOJOSIA</td>\n",
       "      <td>2886</td>\n",
       "      <td>QM0007-C</td>\n",
       "      <td>2886STDY5684878</td>\n",
       "      <td>2886STDY5684878</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>8</td>\n",
       "      <td>75.0</td>\n",
       "      <td>51806556</td>\n",
       "      <td>93.66</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_8#76.cram</th>\n",
       "      <td>1102-PF-MG-RANDRIANARIVELOJOSIA</td>\n",
       "      <td>2886</td>\n",
       "      <td>QM0008-C</td>\n",
       "      <td>2886STDY5684886</td>\n",
       "      <td>2886STDY5684886</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>8</td>\n",
       "      <td>76.0</td>\n",
       "      <td>47580480</td>\n",
       "      <td>98.20</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_8#77.cram</th>\n",
       "      <td>1102-PF-MG-RANDRIANARIVELOJOSIA</td>\n",
       "      <td>2886</td>\n",
       "      <td>QM0009-C</td>\n",
       "      <td>2886STDY5684894</td>\n",
       "      <td>2886STDY5684894</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>8</td>\n",
       "      <td>77.0</td>\n",
       "      <td>50653168</td>\n",
       "      <td>97.35</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_8#78.cram</th>\n",
       "      <td>1102-PF-MG-RANDRIANARIVELOJOSIA</td>\n",
       "      <td>2886</td>\n",
       "      <td>QM0011-C</td>\n",
       "      <td>2886STDY5684910</td>\n",
       "      <td>2886STDY5684910</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>8</td>\n",
       "      <td>78.0</td>\n",
       "      <td>52981598</td>\n",
       "      <td>96.60</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12135_8#79.cram</th>\n",
       "      <td>1102-PF-MG-RANDRIANARIVELOJOSIA</td>\n",
       "      <td>2886</td>\n",
       "      <td>QM0012-C</td>\n",
       "      <td>2886STDY5684918</td>\n",
       "      <td>2886STDY5684918</td>\n",
       "      <td>5833</td>\n",
       "      <td>12135</td>\n",
       "      <td>8</td>\n",
       "      <td>79.0</td>\n",
       "      <td>48504104</td>\n",
       "      <td>97.99</td>\n",
       "      <td>HS16</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2014-02-28 12:53:57</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        study_name study_lims  \\\n",
       "irods_filename                                                                  \n",
       "12135_1#31.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_1#32.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_1#33.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_1#34.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_1#35.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_1#36.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_2#37.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_2#38.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_2#39.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_2#40.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_2#41.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_2#42.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_3#43.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_3#44.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_3#45.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_3#46.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_3#47.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_3#48.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_4#49.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_4#50.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_4#51.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_4#52.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_4#53.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_4#54.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_5#55.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_5#56.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_5#57.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_5#58.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_5#59.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_5#60.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_6#61.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_6#62.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_6#63.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_6#64.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_6#65.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_6#66.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_7#67.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_7#68.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_7#69.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_7#70.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_7#71.cram  Test sequencing of high level human contaminat...       2876   \n",
       "12135_8#73.cram                    1102-PF-MG-RANDRIANARIVELOJOSIA       2886   \n",
       "12135_8#74.cram                    1102-PF-MG-RANDRIANARIVELOJOSIA       2886   \n",
       "12135_8#75.cram                    1102-PF-MG-RANDRIANARIVELOJOSIA       2886   \n",
       "12135_8#76.cram                    1102-PF-MG-RANDRIANARIVELOJOSIA       2886   \n",
       "12135_8#77.cram                    1102-PF-MG-RANDRIANARIVELOJOSIA       2886   \n",
       "12135_8#78.cram                    1102-PF-MG-RANDRIANARIVELOJOSIA       2886   \n",
       "12135_8#79.cram                    1102-PF-MG-RANDRIANARIVELOJOSIA       2886   \n",
       "\n",
       "                    sample             name sanger_sample_id  taxon_id  \\\n",
       "irods_filename                                                           \n",
       "12135_1#31.cram   PC0138-C  2876STDY5684764  2876STDY5684764      5833   \n",
       "12135_1#32.cram   PC0139-C  2876STDY5684765  2876STDY5684765      5833   \n",
       "12135_1#33.cram   PC0140-C  2876STDY5684766  2876STDY5684766      5833   \n",
       "12135_1#34.cram   PC0141-C  2876STDY5684767  2876STDY5684767      5833   \n",
       "12135_1#35.cram   PC0142-C  2876STDY5684768  2876STDY5684768      5833   \n",
       "12135_1#36.cram   PC0143-C  2876STDY5684769  2876STDY5684769      5833   \n",
       "12135_2#37.cram   PC0144-C  2876STDY5684770  2876STDY5684770      5833   \n",
       "12135_2#38.cram   PC0145-C  2876STDY5684771  2876STDY5684771      5833   \n",
       "12135_2#39.cram   PC0146-C  2876STDY5684772  2876STDY5684772      5833   \n",
       "12135_2#40.cram   PC0147-C  2876STDY5684773  2876STDY5684773      5833   \n",
       "12135_2#41.cram   PC0148-C  2876STDY5684774  2876STDY5684774      5833   \n",
       "12135_2#42.cram   PC0149-C  2876STDY5684775  2876STDY5684775      5833   \n",
       "12135_3#43.cram   PC0150-C  2876STDY5684776  2876STDY5684776      5833   \n",
       "12135_3#44.cram   PC0151-C  2876STDY5684777  2876STDY5684777      5833   \n",
       "12135_3#45.cram   PC0152-C  2876STDY5684778  2876STDY5684778      5833   \n",
       "12135_3#46.cram   PC0153-C  2876STDY5684779  2876STDY5684779      5833   \n",
       "12135_3#47.cram   PC0154-C  2876STDY5684780  2876STDY5684780      5833   \n",
       "12135_3#48.cram   PC0155-C  2876STDY5684781  2876STDY5684781      5833   \n",
       "12135_4#49.cram   PC0156-C  2876STDY5684782  2876STDY5684782      5833   \n",
       "12135_4#50.cram   PC0157-C  2876STDY5684783  2876STDY5684783      5833   \n",
       "12135_4#51.cram   PC0158-C  2876STDY5684784  2876STDY5684784      5833   \n",
       "12135_4#52.cram   PC0159-C  2876STDY5684785  2876STDY5684785      5833   \n",
       "12135_4#53.cram   PC0160-C  2876STDY5684786  2876STDY5684786      5833   \n",
       "12135_4#54.cram   PC0161-C  2876STDY5684787  2876STDY5684787      5833   \n",
       "12135_5#55.cram   PC0162-C  2876STDY5684788  2876STDY5684788      5833   \n",
       "12135_5#56.cram   PE0092-C  2876STDY5684789  2876STDY5684789      5833   \n",
       "12135_5#57.cram   PM0129-C  2876STDY5684790  2876STDY5684790      5833   \n",
       "12135_5#58.cram   PM0077-C  2876STDY5684791  2876STDY5684791      5833   \n",
       "12135_5#59.cram   PF0812-C  2876STDY5684792  2876STDY5684792      5833   \n",
       "12135_5#60.cram   PE0093-C  2876STDY5684793  2876STDY5684793      5833   \n",
       "12135_6#61.cram   PE0108-C  2876STDY5684794  2876STDY5684794      5833   \n",
       "12135_6#62.cram   PM0161-C  2876STDY5684795  2876STDY5684795      5833   \n",
       "12135_6#63.cram   PF0266-C  2876STDY5684796  2876STDY5684796      5833   \n",
       "12135_6#64.cram   PH0963-C  2876STDY5684797  2876STDY5684797      5833   \n",
       "12135_6#65.cram  PH0963-Cx  2876STDY5684798  2876STDY5684798      5833   \n",
       "12135_6#66.cram   PD0338-C  2876STDY5684799  2876STDY5684799      5833   \n",
       "12135_7#67.cram   QE0196-C  2876STDY5684800  2876STDY5684800      5833   \n",
       "12135_7#68.cram   QE0008-C  2876STDY5684801  2876STDY5684801      5833   \n",
       "12135_7#69.cram   QE0145-C  2876STDY5684802  2876STDY5684802      5833   \n",
       "12135_7#70.cram   QE0022-C  2876STDY5684803  2876STDY5684803      5833   \n",
       "12135_7#71.cram   QE0212-C  2876STDY5684804  2876STDY5684804      5833   \n",
       "12135_8#73.cram   QM0005-C  2886STDY5684862  2886STDY5684862      5833   \n",
       "12135_8#74.cram   QM0006-C  2886STDY5684870  2886STDY5684870      5833   \n",
       "12135_8#75.cram   QM0007-C  2886STDY5684878  2886STDY5684878      5833   \n",
       "12135_8#76.cram   QM0008-C  2886STDY5684886  2886STDY5684886      5833   \n",
       "12135_8#77.cram   QM0009-C  2886STDY5684894  2886STDY5684894      5833   \n",
       "12135_8#78.cram   QM0011-C  2886STDY5684910  2886STDY5684910      5833   \n",
       "12135_8#79.cram   QM0012-C  2886STDY5684918  2886STDY5684918      5833   \n",
       "\n",
       "                 id_run  position  tag_index  num_reads  human_percent_mapped  \\\n",
       "irods_filename                                                                  \n",
       "12135_1#31.cram   12135         1       31.0    2512866                 99.55   \n",
       "12135_1#32.cram   12135         1       32.0    2193340                 99.22   \n",
       "12135_1#33.cram   12135         1       33.0    2098154                 99.49   \n",
       "12135_1#34.cram   12135         1       34.0    2209976                 99.53   \n",
       "12135_1#35.cram   12135         1       35.0    3432242                 99.52   \n",
       "12135_1#36.cram   12135         1       36.0    2457978                 99.52   \n",
       "12135_2#37.cram   12135         2       37.0    2077158                 99.52   \n",
       "12135_2#38.cram   12135         2       38.0    2133750                 99.49   \n",
       "12135_2#39.cram   12135         2       39.0    2447318                 99.53   \n",
       "12135_2#40.cram   12135         2       40.0    2246078                 99.51   \n",
       "12135_2#41.cram   12135         2       41.0    1850042                 99.52   \n",
       "12135_2#42.cram   12135         2       42.0    2018266                 99.51   \n",
       "12135_3#43.cram   12135         3       43.0    2145746                 99.55   \n",
       "12135_3#44.cram   12135         3       44.0    1995634                 99.58   \n",
       "12135_3#45.cram   12135         3       45.0    2773312                 99.54   \n",
       "12135_3#46.cram   12135         3       46.0    2076032                 99.54   \n",
       "12135_3#47.cram   12135         3       47.0    2169918                 99.57   \n",
       "12135_3#48.cram   12135         3       48.0    2070780                 99.55   \n",
       "12135_4#49.cram   12135         4       49.0    2899830                 99.56   \n",
       "12135_4#50.cram   12135         4       50.0    2115480                 99.53   \n",
       "12135_4#51.cram   12135         4       51.0    2179082                 99.58   \n",
       "12135_4#52.cram   12135         4       52.0    2224284                 99.59   \n",
       "12135_4#53.cram   12135         4       53.0    1923730                 99.54   \n",
       "12135_4#54.cram   12135         4       54.0    2508814                 99.54   \n",
       "12135_5#55.cram   12135         5       55.0    3000932                 99.54   \n",
       "12135_5#56.cram   12135         5       56.0   10818680                 99.36   \n",
       "12135_5#57.cram   12135         5       57.0    6512292                 99.45   \n",
       "12135_5#58.cram   12135         5       58.0   12493066                 99.49   \n",
       "12135_5#59.cram   12135         5       59.0    3306334                 99.04   \n",
       "12135_5#60.cram   12135         5       60.0    7780290                 99.49   \n",
       "12135_6#61.cram   12135         6       61.0    6752670                 99.53   \n",
       "12135_6#62.cram   12135         6       62.0   14254540                 99.20   \n",
       "12135_6#63.cram   12135         6       63.0    6991004                 99.52   \n",
       "12135_6#64.cram   12135         6       64.0    6460296                 99.51   \n",
       "12135_6#65.cram   12135         6       65.0    6109270                 99.48   \n",
       "12135_6#66.cram   12135         6       66.0    4663282                 99.44   \n",
       "12135_7#67.cram   12135         7       67.0    3498108                 99.57   \n",
       "12135_7#68.cram   12135         7       68.0    6843374                 99.50   \n",
       "12135_7#69.cram   12135         7       69.0    3067626                 99.54   \n",
       "12135_7#70.cram   12135         7       70.0   16264426                 99.49   \n",
       "12135_7#71.cram   12135         7       71.0    5631900                 99.54   \n",
       "12135_8#73.cram   12135         8       73.0   32819612                 99.17   \n",
       "12135_8#74.cram   12135         8       74.0   29713518                 98.86   \n",
       "12135_8#75.cram   12135         8       75.0   51806556                 93.66   \n",
       "12135_8#76.cram   12135         8       76.0   47580480                 98.20   \n",
       "12135_8#77.cram   12135         8       77.0   50653168                 97.35   \n",
       "12135_8#78.cram   12135         8       78.0   52981598                 96.60   \n",
       "12135_8#79.cram   12135         8       79.0   48504104                 97.99   \n",
       "\n",
       "                instrument_name instrument_model  paired_read  \\\n",
       "irods_filename                                                  \n",
       "12135_1#31.cram            HS16            HiSeq            1   \n",
       "12135_1#32.cram            HS16            HiSeq            1   \n",
       "12135_1#33.cram            HS16            HiSeq            1   \n",
       "12135_1#34.cram            HS16            HiSeq            1   \n",
       "12135_1#35.cram            HS16            HiSeq            1   \n",
       "12135_1#36.cram            HS16            HiSeq            1   \n",
       "12135_2#37.cram            HS16            HiSeq            1   \n",
       "12135_2#38.cram            HS16            HiSeq            1   \n",
       "12135_2#39.cram            HS16            HiSeq            1   \n",
       "12135_2#40.cram            HS16            HiSeq            1   \n",
       "12135_2#41.cram            HS16            HiSeq            1   \n",
       "12135_2#42.cram            HS16            HiSeq            1   \n",
       "12135_3#43.cram            HS16            HiSeq            1   \n",
       "12135_3#44.cram            HS16            HiSeq            1   \n",
       "12135_3#45.cram            HS16            HiSeq            1   \n",
       "12135_3#46.cram            HS16            HiSeq            1   \n",
       "12135_3#47.cram            HS16            HiSeq            1   \n",
       "12135_3#48.cram            HS16            HiSeq            1   \n",
       "12135_4#49.cram            HS16            HiSeq            1   \n",
       "12135_4#50.cram            HS16            HiSeq            1   \n",
       "12135_4#51.cram            HS16            HiSeq            1   \n",
       "12135_4#52.cram            HS16            HiSeq            1   \n",
       "12135_4#53.cram            HS16            HiSeq            1   \n",
       "12135_4#54.cram            HS16            HiSeq            1   \n",
       "12135_5#55.cram            HS16            HiSeq            1   \n",
       "12135_5#56.cram            HS16            HiSeq            1   \n",
       "12135_5#57.cram            HS16            HiSeq            1   \n",
       "12135_5#58.cram            HS16            HiSeq            1   \n",
       "12135_5#59.cram            HS16            HiSeq            1   \n",
       "12135_5#60.cram            HS16            HiSeq            1   \n",
       "12135_6#61.cram            HS16            HiSeq            1   \n",
       "12135_6#62.cram            HS16            HiSeq            1   \n",
       "12135_6#63.cram            HS16            HiSeq            1   \n",
       "12135_6#64.cram            HS16            HiSeq            1   \n",
       "12135_6#65.cram            HS16            HiSeq            1   \n",
       "12135_6#66.cram            HS16            HiSeq            1   \n",
       "12135_7#67.cram            HS16            HiSeq            1   \n",
       "12135_7#68.cram            HS16            HiSeq            1   \n",
       "12135_7#69.cram            HS16            HiSeq            1   \n",
       "12135_7#70.cram            HS16            HiSeq            1   \n",
       "12135_7#71.cram            HS16            HiSeq            1   \n",
       "12135_8#73.cram            HS16            HiSeq            1   \n",
       "12135_8#74.cram            HS16            HiSeq            1   \n",
       "12135_8#75.cram            HS16            HiSeq            1   \n",
       "12135_8#76.cram            HS16            HiSeq            1   \n",
       "12135_8#77.cram            HS16            HiSeq            1   \n",
       "12135_8#78.cram            HS16            HiSeq            1   \n",
       "12135_8#79.cram            HS16            HiSeq            1   \n",
       "\n",
       "                        qc_complete  manual_qc  requested_insert_size_from  \\\n",
       "irods_filename                                                               \n",
       "12135_1#31.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_1#32.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_1#33.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_1#34.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_1#35.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_1#36.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_2#37.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_2#38.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_2#39.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_2#40.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_2#41.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_2#42.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_3#43.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_3#44.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_3#45.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_3#46.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_3#47.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_3#48.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_4#49.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_4#50.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_4#51.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_4#52.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_4#53.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_4#54.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_5#55.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_5#56.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_5#57.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_5#58.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_5#59.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_5#60.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_6#61.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_6#62.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_6#63.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_6#64.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_6#65.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_6#66.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_7#67.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_7#68.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_7#69.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_7#70.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_7#71.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_8#73.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_8#74.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_8#75.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_8#76.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_8#77.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_8#78.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "12135_8#79.cram 2014-02-28 12:53:57          1                       200.0   \n",
       "\n",
       "                 requested_insert_size_to  forward_read_length  description  \n",
       "irods_filename                                                               \n",
       "12135_1#31.cram                     300.0                  100  qc complete  \n",
       "12135_1#32.cram                     300.0                  100  qc complete  \n",
       "12135_1#33.cram                     300.0                  100  qc complete  \n",
       "12135_1#34.cram                     300.0                  100  qc complete  \n",
       "12135_1#35.cram                     300.0                  100  qc complete  \n",
       "12135_1#36.cram                     300.0                  100  qc complete  \n",
       "12135_2#37.cram                     300.0                  100  qc complete  \n",
       "12135_2#38.cram                     300.0                  100  qc complete  \n",
       "12135_2#39.cram                     300.0                  100  qc complete  \n",
       "12135_2#40.cram                     300.0                  100  qc complete  \n",
       "12135_2#41.cram                     300.0                  100  qc complete  \n",
       "12135_2#42.cram                     300.0                  100  qc complete  \n",
       "12135_3#43.cram                     300.0                  100  qc complete  \n",
       "12135_3#44.cram                     300.0                  100  qc complete  \n",
       "12135_3#45.cram                     300.0                  100  qc complete  \n",
       "12135_3#46.cram                     300.0                  100  qc complete  \n",
       "12135_3#47.cram                     300.0                  100  qc complete  \n",
       "12135_3#48.cram                     300.0                  100  qc complete  \n",
       "12135_4#49.cram                     300.0                  100  qc complete  \n",
       "12135_4#50.cram                     300.0                  100  qc complete  \n",
       "12135_4#51.cram                     300.0                  100  qc complete  \n",
       "12135_4#52.cram                     300.0                  100  qc complete  \n",
       "12135_4#53.cram                     300.0                  100  qc complete  \n",
       "12135_4#54.cram                     300.0                  100  qc complete  \n",
       "12135_5#55.cram                     300.0                  100  qc complete  \n",
       "12135_5#56.cram                     300.0                  100  qc complete  \n",
       "12135_5#57.cram                     300.0                  100  qc complete  \n",
       "12135_5#58.cram                     300.0                  100  qc complete  \n",
       "12135_5#59.cram                     300.0                  100  qc complete  \n",
       "12135_5#60.cram                     300.0                  100  qc complete  \n",
       "12135_6#61.cram                     300.0                  100  qc complete  \n",
       "12135_6#62.cram                     300.0                  100  qc complete  \n",
       "12135_6#63.cram                     300.0                  100  qc complete  \n",
       "12135_6#64.cram                     300.0                  100  qc complete  \n",
       "12135_6#65.cram                     300.0                  100  qc complete  \n",
       "12135_6#66.cram                     300.0                  100  qc complete  \n",
       "12135_7#67.cram                     300.0                  100  qc complete  \n",
       "12135_7#68.cram                     300.0                  100  qc complete  \n",
       "12135_7#69.cram                     300.0                  100  qc complete  \n",
       "12135_7#70.cram                     300.0                  100  qc complete  \n",
       "12135_7#71.cram                     300.0                  100  qc complete  \n",
       "12135_8#73.cram                     300.0                  100  qc complete  \n",
       "12135_8#74.cram                     300.0                  100  qc complete  \n",
       "12135_8#75.cram                     300.0                  100  qc complete  \n",
       "12135_8#76.cram                     300.0                  100  qc complete  \n",
       "12135_8#77.cram                     300.0                  100  qc complete  \n",
       "12135_8#78.cram                     300.0                  100  qc complete  \n",
       "12135_8#79.cram                     300.0                  100  qc complete  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mlwh.loc[df_mlwh['id_run'] == 12135]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'12135_6#65.cram' in df_mlwh.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'12135_6#66.cram' in df_mlwh.index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add some constraints"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def set_irods_name(row):\n",
    "    # First calculate prefix using nonhuman dependent on id_run and species (taxon)\n",
    "    # Then calculate tag string (empty string if no tag)\n",
    "    # Then calculate file extension\n",
    "    # Then concat these three strings\n",
    "    if (\n",
    "        (row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 6750)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    elif (\n",
    "        ~(row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 7100)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    else:\n",
    "        prefix = \"%d_%d\" % (row['id_run'], row['position'])\n",
    "        \n",
    "    if np.isnan(row['tag_index']):\n",
    "        tag_string = ''\n",
    "    else:\n",
    "        tag_string = '#%d' % row['tag_index']\n",
    "    \n",
    "    if row['id_run'] <= 10300:\n",
    "        file_extension = '.bam'\n",
    "    else:\n",
    "        file_extension = '.cram'\n",
    "    \n",
    "    irods_filename = prefix + tag_string + file_extension\n",
    "    return(irods_filename)\n",
    "\n",
    "def set_sample_id(row):\n",
    "    if row['sample'] is None:\n",
    "        return('DS_%s' % row['name'])\n",
    "    else:\n",
    "        return('DS_%s' % row['sample'])\n",
    "\n",
    "def is_valid_alfresco_code(s):\n",
    "    try: \n",
    "        int(s[0:4])\n",
    "        return True\n",
    "    except ValueError:\n",
    "        return False\n",
    "\n",
    "def create_build_manifest(\n",
    "#     taxon_ids=[5833, 5855, 5858, 5821, 7165, 7173, 30066], # Pf, Pv, Pm, Pb, Ag, A. arabiensis, A. merus\n",
    "    taxon_ids=[5833, 36329, 5847, 57267, 137071], # Pf, 3D7, V1, Dd2, HB3\n",
    "    # 1089 is excluded as this is R&D samples. 1204 and 1176 are excluded as these are CP1 samples. 1175 is excluded as some Pf R&D samples were incorrectly tagged with this study\n",
    "    studies_to_exclude = ['1089-R&amp;D-GB-ALCOCK', '1204-PF-GM-CP1', '1176-PF-KE-CP1', '1175-VO-KH-STLAURENT'],\n",
    "    miseq_runs_to_allow = [13809, 13810], # Two Miseq runs within 24 samples on each from study 1095-PF-TZ-ISHENGOMA that were included in Pf 6.0\n",
    "    input_dir = input_dir,\n",
    "    mlwh_missing_exceptions_fn = mlwh_missing_exceptions_fn,\n",
    "    mlwh_study_exceptions_fn = mlwh_study_exceptions_fn,\n",
    "    mlwh_sample_exceptions_fn = mlwh_sample_exceptions_fn,\n",
    "    mlwh_taxon_exceptions_fn = mlwh_taxon_exceptions_fn,\n",
    "    sequencescape_alfresco_study_mappings_fn = sequencescape_alfresco_study_mappings_fn,\n",
    "    output_columns = [\n",
    "        'path',\n",
    "        'study',\n",
    "        'sample',\n",
    "        'lane',\n",
    "        'reads',\n",
    "        'paired',\n",
    "        'irods_path',\n",
    "        'sanger_sample_id',\n",
    "        'taxon_id',\n",
    "        'study_lims',\n",
    "        'study_name',\n",
    "        'id_run',\n",
    "        'position',\n",
    "        'tag_index',\n",
    "        'qc_complete',\n",
    "        'manual_qc',\n",
    "        'description',\n",
    "        'instrument_name',\n",
    "        'instrument_model',\n",
    "        'forward_read_length',\n",
    "        'requested_insert_size_from',\n",
    "        'requested_insert_size_to',\n",
    "        'human_percent_mapped',\n",
    "        'subtrack_filename',\n",
    "        'subtrack_files_bytes',\n",
    "        'ebi_run_acc'\n",
    "    ]\n",
    "):\n",
    "    \"\"\"Create a DataFrame to be used as a build manifest.\n",
    "    \n",
    "    A build manifest here is a list of all the \"lanelets\" that need to be\n",
    "    included in a build. The output will typically be written to a\n",
    "    tab-delmited file, either to use as input to vr-pipe, or perhaps as\n",
    "    input to a vr-track DB.\n",
    "\n",
    "    Args:\n",
    "        taxon_ids (list of int): The taxons of the build (P. falciparum is '5833',\n",
    "            P. vivax is '5855').\n",
    "        sequencescape_alfresco_study_mappings_fn (str): filename of\n",
    "            mappings from sequencscape to alfresco study mappings\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The build manifest.\n",
    "\n",
    "    \"\"\"\n",
    "    \n",
    "    # Read in taxon exceptions file\n",
    "    df_mlwh_taxon_exceptions = pd.read_csv(mlwh_taxon_exceptions_fn, sep='\\t', dtype={'tag_index': str, 'taxon_id': int}, index_col='irods_filename')\n",
    "    df_mlwh_taxon_exceptions['tag_index'].fillna('', inplace=True)\n",
    "    # Identify which samples in exceptions file match the taxon of current build, and which don't\n",
    "    df_mlwh_exceptions_this_taxon = df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)]\n",
    "    df_mlwh_exceptions_other_taxa = df_mlwh_taxon_exceptions.loc[~(df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids))]\n",
    "\n",
    "    # Read in sample exceptions file\n",
    "    df_mlwh_sample_exceptions = pd.read_csv(mlwh_sample_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in study exceptions file\n",
    "    df_mlwh_study_exceptions = pd.read_csv(mlwh_study_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in missing exceptions file\n",
    "    df_mlwh_missing_exceptions = pd.read_csv(mlwh_missing_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "    df_mlwh_missing_exceptions['derivative_sample_id'] = 'DS_' + df_mlwh_missing_exceptions['sample_id']\n",
    "    df_mlwh_missing_exceptions = df_mlwh_missing_exceptions.loc[\n",
    "        df_mlwh_missing_exceptions['taxon_id'].isin(taxon_ids)\n",
    "    ]\n",
    "\n",
    "    # Read in SequenceScape-Alfresco study mappings\n",
    "    df_sequencescape_alfresco_study_mappings = pd.read_csv(sequencescape_alfresco_study_mappings_fn, sep='\\t', index_col='seqscape_study_id')\n",
    "    sequencescape_alfresco_study_mappings_dict = df_sequencescape_alfresco_study_mappings.loc[\n",
    "        :,\n",
    "#         df_sequencescape_alfresco_study_mappings['build_flag']==1,\n",
    "        'alfresco_study_code'\n",
    "    ].to_dict()\n",
    "    \n",
    "    # Read in data from mlwh matching this taxon, plus samples from exceptions file matching this taxon\n",
    "    conn = pymysql.connect(\n",
    "        host='mlwh-db',\n",
    "        user='mlwh_malaria',\n",
    "        password='Solaris&2015',\n",
    "        db='mlwarehouse',\n",
    "        port=3435\n",
    "    )\n",
    "    \n",
    "    sql_query = 'SELECT \\\n",
    "        study.name as study_name, \\\n",
    "        study.id_study_lims as study_lims, \\\n",
    "        sample.supplier_name as sample, \\\n",
    "        sample.name, \\\n",
    "        sample.sanger_sample_id, \\\n",
    "        sample.taxon_id, \\\n",
    "        iseq_product_metrics.id_run, \\\n",
    "        iseq_product_metrics.position, \\\n",
    "        iseq_product_metrics.tag_index, \\\n",
    "        iseq_product_metrics.num_reads, \\\n",
    "        iseq_product_metrics.human_percent_mapped, \\\n",
    "        iseq_run_lane_metrics.instrument_name, \\\n",
    "        iseq_run_lane_metrics.instrument_model, \\\n",
    "        iseq_run_lane_metrics.paired_read, \\\n",
    "        iseq_run_lane_metrics.qc_complete, \\\n",
    "        iseq_flowcell.manual_qc, \\\n",
    "        iseq_flowcell.requested_insert_size_from, \\\n",
    "        iseq_flowcell.requested_insert_size_to, \\\n",
    "        iseq_flowcell.forward_read_length, \\\n",
    "        iseq_run_status_dict.description \\\n",
    "    FROM \\\n",
    "        study, \\\n",
    "        iseq_flowcell, \\\n",
    "        sample, \\\n",
    "        iseq_product_metrics, \\\n",
    "        iseq_run_status, \\\n",
    "        iseq_run_lane_metrics, \\\n",
    "        iseq_run_status_dict \\\n",
    "    WHERE \\\n",
    "        study.id_study_tmp = iseq_flowcell.id_study_tmp and \\\n",
    "        iseq_flowcell.id_sample_tmp = sample.id_sample_tmp and \\\n",
    "        iseq_flowcell.manual_qc = 1 and \\\n",
    "        iseq_product_metrics.id_iseq_flowcell_tmp = iseq_flowcell.id_iseq_flowcell_tmp and \\\n",
    "        iseq_run_status.id_run = iseq_product_metrics.id_run and \\\n",
    "        iseq_product_metrics.id_run = iseq_run_lane_metrics.id_run and \\\n",
    "        iseq_product_metrics.position = iseq_run_lane_metrics.position and \\\n",
    "        iseq_run_status.iscurrent = 1 and \\\n",
    "        ( ( iseq_run_lane_metrics.instrument_model != \"MiSeq\" ) or (iseq_product_metrics.id_run in (%s)) ) and \\\n",
    "        iseq_run_status.id_run_status_dict = iseq_run_status_dict.id_run_status_dict and \\\n",
    "        study.faculty_sponsor = \"Dominic Kwiatkowski\" and \\\n",
    "        ( ( sample.taxon_id in (%s) ) or ' % (\n",
    "            ', '.join([str(x) for x in miseq_runs_to_allow]),\n",
    "            ', '.join([str(x) for x in taxon_ids])\n",
    "        )\n",
    "    sql_query = sql_query + ' or '.join(\n",
    "        df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)].apply(\n",
    "            lambda x: '(iseq_product_metrics.id_run=\"%s\" and iseq_product_metrics.position=\"%s\" and iseq_product_metrics.tag_index=\"%s\")' % (x['id_run'], x['position'], x['tag_index']),\n",
    "            1\n",
    "        )\n",
    "    )\n",
    "    sql_query = sql_query + ')'\n",
    "#         (iseq_flowcell.manual_qc = 1 or iseq_flowcell.manual_qc is null) and \\\n",
    "    \n",
    "    df_return = pd.read_sql(sql_query, conn)\n",
    "\n",
    "    # Replace missing taxon_id with -1 (can't have missing int values in pandas Series)\n",
    "    df_return['taxon_id'] = df_return['taxon_id'].fillna(-1).astype('int32')\n",
    "\n",
    "    # Determine file name in iRods\n",
    "    df_return['irods_filename'] = df_return.apply(set_irods_name, 1)\n",
    "    df_return.set_index('irods_filename', inplace=True)\n",
    "\n",
    "    # Determine alfresco study and change any incorrect studies\n",
    "    df_return['alfresco_study_code'] = df_return['study_lims'].apply(\n",
    "        lambda x: sequencescape_alfresco_study_mappings_dict[int(x)] if int(x) in sequencescape_alfresco_study_mappings_dict else ''\n",
    "    )\n",
    "    empty_alfresco_study_code = (df_return['alfresco_study_code'] == '')\n",
    "    df_return.loc[empty_alfresco_study_code, 'alfresco_study_code'] = df_return.loc[empty_alfresco_study_code, 'study_name']\n",
    "    \n",
    "    study_exception_indexes = df_mlwh_study_exceptions.index[\n",
    "        df_mlwh_study_exceptions.index.isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[study_exception_indexes, 'alfresco_study_code'] = df_mlwh_study_exceptions.loc[study_exception_indexes, 'alfresco_study_code']\n",
    "\n",
    "    # Change any incorrect taxon IDs\n",
    "    taxon_exception_indexes = df_mlwh_exceptions_this_taxon.index[\n",
    "        df_mlwh_exceptions_this_taxon.index.to_series().isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[taxon_exception_indexes, 'taxon_id'] = df_mlwh_exceptions_this_taxon.loc[taxon_exception_indexes, 'taxon_id']\n",
    "\n",
    "    # Determine sample ID and change any incorrect sample IDs\n",
    "    df_return['derivative_sample_id'] = df_return.apply(set_sample_id, 1)\n",
    "    sample_exception_indexes = df_mlwh_sample_exceptions.index[\n",
    "        df_mlwh_sample_exceptions.index.isin(df_return.index)\n",
    "    ]    \n",
    "    df_return.loc[sample_exception_indexes, 'derivative_sample_id'] = 'DS_' + df_mlwh_sample_exceptions.loc[sample_exception_indexes, 'sample_id']\n",
    "    df_return.drop(['sample', 'name'], axis=1, inplace=True)\n",
    "    \n",
    "    # Remove any samples that are not in this taxon\n",
    "    df_return = df_return.loc[~df_return.index.isin(df_mlwh_exceptions_other_taxa.index)]\n",
    "\n",
    "    return(df_return)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(23430, 20)\n",
      "True\n",
      "True\n",
      "True\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "df_mlwh_taxa_exceptions = create_build_manifest()\n",
    "print(df_mlwh_taxa_exceptions.shape)\n",
    "print('6901_8_nonhuman#1.bam' in df_mlwh_taxa_exceptions.index)\n",
    "print('6901_8_nonhuman#2.bam' in df_mlwh_taxa_exceptions.index)\n",
    "print('12135_6#65.cram' in df_mlwh_taxa_exceptions.index)\n",
    "print('12135_6#66.cram' in df_mlwh_taxa_exceptions.index)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_name</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>num_reads</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>paired_read</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>description</th>\n",
       "      <th>alfresco_study_code</th>\n",
       "      <th>derivative_sample_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#1.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4969844</td>\n",
       "      <td>99.42</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>1052-PF-TRAC-WHITE</td>\n",
       "      <td>DS_PH0540-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6901_8_nonhuman#2.bam</th>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>578</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>6901</td>\n",
       "      <td>8</td>\n",
       "      <td>2.0</td>\n",
       "      <td>52856</td>\n",
       "      <td>93.42</td>\n",
       "      <td>HS13</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>1</td>\n",
       "      <td>2011-11-01 12:52:25</td>\n",
       "      <td>1</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>100</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>DS_Tr5E</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     study_name study_lims  \\\n",
       "irods_filename                                                               \n",
       "6901_8_nonhuman#1.bam  Plasmodium falciparum genome variation 1        578   \n",
       "6901_8_nonhuman#2.bam  Plasmodium falciparum genome variation 1        578   \n",
       "\n",
       "                      sanger_sample_id  taxon_id  id_run  position  tag_index  \\\n",
       "irods_filename                                                                  \n",
       "6901_8_nonhuman#1.bam             None      5833    6901         8        1.0   \n",
       "6901_8_nonhuman#2.bam             None      5833    6901         8        2.0   \n",
       "\n",
       "                       num_reads  human_percent_mapped instrument_name  \\\n",
       "irods_filename                                                           \n",
       "6901_8_nonhuman#1.bam    4969844                 99.42            HS13   \n",
       "6901_8_nonhuman#2.bam      52856                 93.42            HS13   \n",
       "\n",
       "                      instrument_model  paired_read         qc_complete  \\\n",
       "irods_filename                                                            \n",
       "6901_8_nonhuman#1.bam            HiSeq            1 2011-11-01 12:52:25   \n",
       "6901_8_nonhuman#2.bam            HiSeq            1 2011-11-01 12:52:25   \n",
       "\n",
       "                       manual_qc  requested_insert_size_from  \\\n",
       "irods_filename                                                 \n",
       "6901_8_nonhuman#1.bam          1                       200.0   \n",
       "6901_8_nonhuman#2.bam          1                       200.0   \n",
       "\n",
       "                       requested_insert_size_to  forward_read_length  \\\n",
       "irods_filename                                                         \n",
       "6901_8_nonhuman#1.bam                     300.0                  100   \n",
       "6901_8_nonhuman#2.bam                     300.0                  100   \n",
       "\n",
       "                       description                       alfresco_study_code  \\\n",
       "irods_filename                                                                 \n",
       "6901_8_nonhuman#1.bam  qc complete                        1052-PF-TRAC-WHITE   \n",
       "6901_8_nonhuman#2.bam  qc complete  Plasmodium falciparum genome variation 1   \n",
       "\n",
       "                      derivative_sample_id  \n",
       "irods_filename                              \n",
       "6901_8_nonhuman#1.bam          DS_PH0540-C  \n",
       "6901_8_nonhuman#2.bam              DS_Tr5E  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mlwh_taxa_exceptions.loc[['6901_8_nonhuman#1.bam', '6901_8_nonhuman#2.bam']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summary\n",
    "Missing files still included here - add more constraints..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def create_build_manifest(\n",
    "#     taxon_ids=[5833, 5855, 5858, 5821, 7165, 7173, 30066], # Pf, Pv, Pm, Pb, Ag, A. arabiensis, A. merus\n",
    "    taxon_ids=[5833, 36329, 5847, 57267, 137071], # Pf, 3D7, V1, Dd2, HB3\n",
    "    # 1089 is excluded as this is R&D samples. 1204 and 1176 are excluded as these are CP1 samples. 1175 is excluded as some Pf R&D samples were incorrectly tagged with this study\n",
    "    studies_to_exclude = ['1089-R&amp;D-GB-ALCOCK', '1204-PF-GM-CP1', '1176-PF-KE-CP1', '1175-VO-KH-STLAURENT'],\n",
    "    miseq_runs_to_allow = [13809, 13810], # Two Miseq runs within 24 samples on each from study 1095-PF-TZ-ISHENGOMA that were included in Pf 6.0\n",
    "    input_dir = input_dir,\n",
    "    mlwh_missing_exceptions_fn = mlwh_missing_exceptions_fn,\n",
    "    mlwh_study_exceptions_fn = mlwh_study_exceptions_fn,\n",
    "    mlwh_sample_exceptions_fn = mlwh_sample_exceptions_fn,\n",
    "    mlwh_taxon_exceptions_fn = mlwh_taxon_exceptions_fn,\n",
    "    sequencescape_alfresco_study_mappings_fn = sequencescape_alfresco_study_mappings_fn,\n",
    "    output_columns = [\n",
    "        'path',\n",
    "        'study',\n",
    "        'sample',\n",
    "        'lane',\n",
    "        'reads',\n",
    "        'paired',\n",
    "        'irods_path',\n",
    "        'sanger_sample_id',\n",
    "        'taxon_id',\n",
    "        'study_lims',\n",
    "        'study_name',\n",
    "        'id_run',\n",
    "        'position',\n",
    "        'tag_index',\n",
    "        'qc_complete',\n",
    "        'manual_qc',\n",
    "        'description',\n",
    "        'instrument_name',\n",
    "        'instrument_model',\n",
    "        'forward_read_length',\n",
    "        'requested_insert_size_from',\n",
    "        'requested_insert_size_to',\n",
    "        'human_percent_mapped',\n",
    "        'subtrack_filename',\n",
    "        'subtrack_files_bytes',\n",
    "        'ebi_run_acc'\n",
    "    ]\n",
    "):\n",
    "    \"\"\"Create a DataFrame to be used as a build manifest.\n",
    "    \n",
    "    A build manifest here is a list of all the \"lanelets\" that need to be\n",
    "    included in a build. The output will typically be written to a\n",
    "    tab-delmited file, either to use as input to vr-pipe, or perhaps as\n",
    "    input to a vr-track DB.\n",
    "\n",
    "    Args:\n",
    "        taxon_ids (list of int): The taxons of the build (P. falciparum is '5833',\n",
    "            P. vivax is '5855').\n",
    "        sequencescape_alfresco_study_mappings_fn (str): filename of\n",
    "            mappings from sequencscape to alfresco study mappings\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The build manifest.\n",
    "\n",
    "    \"\"\"\n",
    "    \n",
    "    # Read in taxon exceptions file\n",
    "    df_mlwh_taxon_exceptions = pd.read_csv(mlwh_taxon_exceptions_fn, sep='\\t', dtype={'tag_index': str, 'taxon_id': int}, index_col='irods_filename')\n",
    "    df_mlwh_taxon_exceptions['tag_index'].fillna('', inplace=True)\n",
    "    # Identify which samples in exceptions file match the taxon of current build, and which don't\n",
    "    df_mlwh_exceptions_this_taxon = df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)]\n",
    "    df_mlwh_exceptions_other_taxa = df_mlwh_taxon_exceptions.loc[~(df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids))]\n",
    "\n",
    "    # Read in sample exceptions file\n",
    "    df_mlwh_sample_exceptions = pd.read_csv(mlwh_sample_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in study exceptions file\n",
    "    df_mlwh_study_exceptions = pd.read_csv(mlwh_study_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in missing exceptions file\n",
    "    df_mlwh_missing_exceptions = pd.read_csv(mlwh_missing_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "    df_mlwh_missing_exceptions['derivative_sample_id'] = 'DS_' + df_mlwh_missing_exceptions['sample_id']\n",
    "    df_mlwh_missing_exceptions = df_mlwh_missing_exceptions.loc[\n",
    "        df_mlwh_missing_exceptions['taxon_id'].isin(taxon_ids)\n",
    "    ]\n",
    "\n",
    "    # Read in SequenceScape-Alfresco study mappings\n",
    "    df_sequencescape_alfresco_study_mappings = pd.read_csv(sequencescape_alfresco_study_mappings_fn, sep='\\t', index_col='seqscape_study_id')\n",
    "    sequencescape_alfresco_study_mappings_dict = df_sequencescape_alfresco_study_mappings.loc[\n",
    "        :,\n",
    "#         df_sequencescape_alfresco_study_mappings['build_flag']==1,\n",
    "        'alfresco_study_code'\n",
    "    ].to_dict()\n",
    "    \n",
    "    # Read in data from mlwh matching this taxon, plus samples from exceptions file matching this taxon\n",
    "    conn = pymysql.connect(\n",
    "        host='mlwh-db',\n",
    "        user='mlwh_malaria',\n",
    "        password='Solaris&2015',\n",
    "        db='mlwarehouse',\n",
    "        port=3435\n",
    "    )\n",
    "    \n",
    "    sql_query = 'SELECT \\\n",
    "        study.name as study_name, \\\n",
    "        study.id_study_lims as study_lims, \\\n",
    "        sample.supplier_name as sample, \\\n",
    "        sample.name, \\\n",
    "        sample.sanger_sample_id, \\\n",
    "        sample.taxon_id, \\\n",
    "        iseq_product_metrics.id_run, \\\n",
    "        iseq_product_metrics.position, \\\n",
    "        iseq_product_metrics.tag_index, \\\n",
    "        iseq_product_metrics.num_reads, \\\n",
    "        iseq_product_metrics.human_percent_mapped, \\\n",
    "        iseq_run_lane_metrics.instrument_name, \\\n",
    "        iseq_run_lane_metrics.instrument_model, \\\n",
    "        iseq_run_lane_metrics.paired_read, \\\n",
    "        iseq_run_lane_metrics.qc_complete, \\\n",
    "        iseq_flowcell.manual_qc, \\\n",
    "        iseq_flowcell.requested_insert_size_from, \\\n",
    "        iseq_flowcell.requested_insert_size_to, \\\n",
    "        iseq_flowcell.forward_read_length, \\\n",
    "        iseq_run_status_dict.description \\\n",
    "    FROM \\\n",
    "        study, \\\n",
    "        iseq_flowcell, \\\n",
    "        sample, \\\n",
    "        iseq_product_metrics, \\\n",
    "        iseq_run_status, \\\n",
    "        iseq_run_lane_metrics, \\\n",
    "        iseq_run_status_dict \\\n",
    "    WHERE \\\n",
    "        study.id_study_tmp = iseq_flowcell.id_study_tmp and \\\n",
    "        iseq_flowcell.id_sample_tmp = sample.id_sample_tmp and \\\n",
    "        iseq_flowcell.manual_qc = 1 and \\\n",
    "        iseq_product_metrics.id_iseq_flowcell_tmp = iseq_flowcell.id_iseq_flowcell_tmp and \\\n",
    "        iseq_run_status.id_run = iseq_product_metrics.id_run and \\\n",
    "        iseq_product_metrics.id_run = iseq_run_lane_metrics.id_run and \\\n",
    "        iseq_product_metrics.position = iseq_run_lane_metrics.position and \\\n",
    "        iseq_run_status.iscurrent = 1 and \\\n",
    "        ( ( iseq_run_lane_metrics.instrument_model != \"MiSeq\" ) or (iseq_product_metrics.id_run in (%s)) ) and \\\n",
    "        iseq_run_status.id_run_status_dict = iseq_run_status_dict.id_run_status_dict and \\\n",
    "        study.faculty_sponsor = \"Dominic Kwiatkowski\" and \\\n",
    "        ( ( sample.taxon_id in (%s) ) or ' % (\n",
    "            ', '.join([str(x) for x in miseq_runs_to_allow]),\n",
    "            ', '.join([str(x) for x in taxon_ids])\n",
    "        )\n",
    "    sql_query = sql_query + ' or '.join(\n",
    "        df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)].apply(\n",
    "            lambda x: '(iseq_product_metrics.id_run=\"%s\" and iseq_product_metrics.position=\"%s\" and iseq_product_metrics.tag_index=\"%s\")' % (x['id_run'], x['position'], x['tag_index']),\n",
    "            1\n",
    "        )\n",
    "    )\n",
    "    sql_query = sql_query + ')'\n",
    "#         (iseq_flowcell.manual_qc = 1 or iseq_flowcell.manual_qc is null) and \\\n",
    "    \n",
    "    df_return = pd.read_sql(sql_query, conn)\n",
    "\n",
    "    # Replace missing taxon_id with -1 (can't have missing int values in pandas Series)\n",
    "    df_return['taxon_id'] = df_return['taxon_id'].fillna(-1).astype('int32')\n",
    "\n",
    "    # Determine file name in iRods\n",
    "    df_return['irods_filename'] = df_return.apply(set_irods_name, 1)\n",
    "    df_return.set_index('irods_filename', inplace=True)\n",
    "\n",
    "    # Determine alfresco study and change any incorrect studies\n",
    "    df_return['alfresco_study_code'] = df_return['study_lims'].apply(\n",
    "        lambda x: sequencescape_alfresco_study_mappings_dict[int(x)] if int(x) in sequencescape_alfresco_study_mappings_dict else ''\n",
    "    )\n",
    "    empty_alfresco_study_code = (df_return['alfresco_study_code'] == '')\n",
    "    df_return.loc[empty_alfresco_study_code, 'alfresco_study_code'] = df_return.loc[empty_alfresco_study_code, 'study_name']\n",
    "    \n",
    "    study_exception_indexes = df_mlwh_study_exceptions.index[\n",
    "        df_mlwh_study_exceptions.index.isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[study_exception_indexes, 'alfresco_study_code'] = df_mlwh_study_exceptions.loc[study_exception_indexes, 'alfresco_study_code']\n",
    "\n",
    "    # Change any incorrect taxon IDs\n",
    "    taxon_exception_indexes = df_mlwh_exceptions_this_taxon.index[\n",
    "        df_mlwh_exceptions_this_taxon.index.to_series().isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[taxon_exception_indexes, 'taxon_id'] = df_mlwh_exceptions_this_taxon.loc[taxon_exception_indexes, 'taxon_id']\n",
    "\n",
    "    # Determine sample ID and change any incorrect sample IDs\n",
    "    df_return['derivative_sample_id'] = df_return.apply(set_sample_id, 1)\n",
    "    sample_exception_indexes = df_mlwh_sample_exceptions.index[\n",
    "        df_mlwh_sample_exceptions.index.isin(df_return.index)\n",
    "    ]    \n",
    "    df_return.loc[sample_exception_indexes, 'derivative_sample_id'] = 'DS_' + df_mlwh_sample_exceptions.loc[sample_exception_indexes, 'sample_id']\n",
    "    df_return.drop(['sample', 'name'], axis=1, inplace=True)\n",
    "    \n",
    "    # Remove any samples that are not in this taxon\n",
    "    df_return = df_return.loc[~df_return.index.isin(df_mlwh_exceptions_other_taxa.index)]\n",
    "\n",
    "    # Merge in missing exceptions\n",
    "    df_return = df_return.append(df_mlwh_missing_exceptions.drop('study_group', 1))\n",
    "    \n",
    "    # Remove any samples that don't have an alfresco study\n",
    "    invalid_codes = ~df_return['alfresco_study_code'].apply(is_valid_alfresco_code)\n",
    "    if(np.count_nonzero(invalid_codes) > 0):\n",
    "        print('Removing %d files with invalid alfresco_study_code:' % np.count_nonzero(invalid_codes))\n",
    "        print(df_return['alfresco_study_code'].loc[invalid_codes].value_counts())\n",
    "        df_return = df_return.loc[~invalid_codes]\n",
    "\n",
    "    return(df_return)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing 63 files with invalid alfresco_study_code:\n",
      "AT-rich amplification                                 53\n",
      "Test sequencing of high level human contamination.     6\n",
      "Plasmodium falciparum genome variation 1               4\n",
      "Name: alfresco_study_code, dtype: int64\n",
      "(23773, 21)\n",
      "True\n",
      "False\n",
      "True\n",
      "False\n"
     ]
    }
   ],
   "source": [
    "df_mlwh_pre_subtrack = create_build_manifest()\n",
    "print(df_mlwh_pre_subtrack.shape)\n",
    "print('6901_8_nonhuman#1.bam' in df_mlwh_pre_subtrack.index)\n",
    "print('6901_8_nonhuman#2.bam' in df_mlwh_pre_subtrack.index)\n",
    "print('12135_6#65.cram' in df_mlwh_pre_subtrack.index)\n",
    "print('12135_6#66.cram' in df_mlwh_pre_subtrack.index)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "irods_filename\n",
       "6901_8_nonhuman#1.bam                                   1052-PF-TRAC-WHITE\n",
       "6901_8_nonhuman#2.bam             Plasmodium falciparum genome variation 1\n",
       "12135_6#65.cram                                       1044-PF-KH-FAIRHURST\n",
       "12135_6#66.cram          Test sequencing of high level human contaminat...\n",
       "Name: alfresco_study_code, dtype: object"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mlwh_taxa_exceptions.loc[['6901_8_nonhuman#1.bam', '6901_8_nonhuman#2.bam', '12135_6#65.cram', '12135_6#66.cram'], 'alfresco_study_code']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Conclusion\n",
    "The missing files were in sequencescape studies that don't have a direct mapping to Alfresco studies. As such we can only attach an Alfresco study if the file was in vw_vrpipe, but these 10 files weren't in vw_vrpipe. My script removes any sample without a valid Alfresco study, and hence these got removed.\n",
    "\n",
    "Would recommend adding these as manifest exceptions with the explanation \"In sequencescape study without corresponding Alfresco study and not in vw_vrpipe\"."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
