{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "See discussion at https://github.com/malariagen/fits/issues/62\n",
    "\n",
    "Here I am trying to determine whether there are files that I am identifying through mlwh queries, but that are not in subtrack. If so, does Magnus see these in \"production FITS\" but not \"Subtrack FITS\"?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       ".container {\n",
       "    width: 100%;\n",
       "}\n",
       "# div#notebook {\n",
       "#     padding-top: 1em;\n",
       "# }\n",
       "# #header-container {\n",
       "#     display: none;\n",
       "# }\n",
       "# #header-bar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #maintoolbar {\n",
       "#     display: none;\n",
       "# }\n",
       "# #menubar-container {\n",
       "#     position: fixed;\n",
       "#     margin-top: 0;\n",
       "# }\n",
       "site {\n",
       "    height: auto !important;\n",
       "}\n",
       "</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:53:06) \n",
      "[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]\n",
      "numpy 1.13.3\n",
      "scipy 0.19.1\n",
      "pandas 0.22.0\n",
      "numexpr 2.6.4\n",
      "pysam 0.8.4\n",
      "pysamstats 0.24.3\n",
      "petlx 1.0.3\n",
      "vcf 0.6.8\n",
      "h5py 2.7.0\n",
      "tables 3.4.2\n",
      "zarr 2.2.0\n",
      "scikit-allel 1.1.10\n"
     ]
    }
   ],
   "source": [
    "%run ../setup.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Inputs\n",
    "mlwh_missing_exceptions_fn = '/nfs/team112_internal/rp7/src/github/malariagen/SIMS/meta/mlwh/mlwh_missing_exceptions.txt'\n",
    "mlwh_study_exceptions_fn = '/nfs/team112_internal/rp7/src/github/malariagen/SIMS/meta/mlwh/mlwh_study_exceptions.txt'\n",
    "mlwh_sample_exceptions_fn = '/nfs/team112_internal/rp7/src/github/malariagen/SIMS/meta/mlwh/mlwh_sample_exceptions.txt'\n",
    "mlwh_taxon_exceptions_fn = '/nfs/team112_internal/rp7/src/github/malariagen/SIMS/meta/mlwh/mlwh_taxon_exceptions.txt'\n",
    "sequencescape_alfresco_study_mappings_fn='/nfs/team112_internal/rp7/src/github/malariagen/SIMS/meta/mlwh/sequencescape_alfresco_study_mappings.txt'\n",
    "input_dir = 'dummy' # only using this so I can reuse the script I wrote for Pf 6.2, i.e. https://github.com/malariagen/parasite-ops/blob/master/work/18_Pf_6_2_irods_manifest/20181023_%2318_Pf_6_2_irods_manifest.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create iRODS manifest for all MalariaGEN Pf samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def set_irods_name(row):\n",
    "    # First calculate prefix using nonhuman dependent on id_run and species (taxon)\n",
    "    # Then calculate tag string (empty string if no tag)\n",
    "    # Then calculate file extension\n",
    "    # Then concat these three strings\n",
    "    if (\n",
    "        (row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 6750)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    elif (\n",
    "        ~(row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 7100)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    else:\n",
    "        prefix = \"%d_%d\" % (row['id_run'], row['position'])\n",
    "        \n",
    "    if np.isnan(row['tag_index']):\n",
    "        tag_string = ''\n",
    "    else:\n",
    "        tag_string = '#%d' % row['tag_index']\n",
    "    \n",
    "    if row['id_run'] <= 10300:\n",
    "        file_extension = '.bam'\n",
    "    else:\n",
    "        file_extension = '.cram'\n",
    "    \n",
    "    irods_filename = prefix + tag_string + file_extension\n",
    "    return(irods_filename)\n",
    "\n",
    "def set_sample_id(row):\n",
    "    if row['sample'] is None:\n",
    "        return('DS_%s' % row['name'])\n",
    "    else:\n",
    "        return('DS_%s' % row['sample'])\n",
    "\n",
    "def is_valid_alfresco_code(s):\n",
    "    try: \n",
    "        int(s[0:4])\n",
    "        return True\n",
    "    except ValueError:\n",
    "        return False\n",
    "\n",
    "def create_build_manifest(\n",
    "#     taxon_ids=[5833, 5855, 5858, 5821, 7165, 7173, 30066], # Pf, Pv, Pm, Pb, Ag, A. arabiensis, A. merus\n",
    "    taxon_ids=[5833, 36329, 5847, 57267, 137071], # Pf, 3D7, V1, Dd2, HB3\n",
    "    # 1089 is excluded as this is R&D samples. 1204 and 1176 are excluded as these are CP1 samples. 1175 is excluded as some Pf R&D samples were incorrectly tagged with this study. 1157 is a Pv study - there are two suspected Pf samples in this study, but we need further investigation, and possible study change both in ROMA and in Sequencescape/mlwh/ENA/iRODS before we can include\n",
    "    studies_to_exclude = ['1089-R&amp;D-GB-ALCOCK', '1204-PF-GM-CP1', '1176-PF-KE-CP1', '1175-VO-KH-STLAURENT', '1157-PV-MULTI-PRICE'],\n",
    "    miseq_runs_to_allow = [13809, 13810], # Two Miseq runs within 24 samples on each from study 1095-PF-TZ-ISHENGOMA that were included in Pf 6.0\n",
    "    input_dir = input_dir,\n",
    "    mlwh_missing_exceptions_fn = mlwh_missing_exceptions_fn,\n",
    "    mlwh_study_exceptions_fn = mlwh_study_exceptions_fn,\n",
    "    mlwh_sample_exceptions_fn = mlwh_sample_exceptions_fn,\n",
    "    mlwh_taxon_exceptions_fn = mlwh_taxon_exceptions_fn,\n",
    "    sequencescape_alfresco_study_mappings_fn = sequencescape_alfresco_study_mappings_fn,\n",
    "    output_columns = [\n",
    "        'path',\n",
    "        'study',\n",
    "        'sample',\n",
    "        'lane',\n",
    "        'reads',\n",
    "        'paired',\n",
    "        'irods_path',\n",
    "        'sanger_sample_id',\n",
    "        'taxon_id',\n",
    "        'study_lims',\n",
    "        'study_name',\n",
    "        'id_run',\n",
    "        'position',\n",
    "        'tag_index',\n",
    "        'qc_complete',\n",
    "        'manual_qc',\n",
    "        'description',\n",
    "        'instrument_name',\n",
    "        'instrument_model',\n",
    "        'forward_read_length',\n",
    "        'requested_insert_size_from',\n",
    "        'requested_insert_size_to',\n",
    "        'human_percent_mapped',\n",
    "        'subtrack_filename',\n",
    "        'subtrack_files_bytes',\n",
    "        'ebi_run_acc'\n",
    "    ]\n",
    "):\n",
    "    \"\"\"Create a DataFrame to be used as a build manifest.\n",
    "    \n",
    "    A build manifest here is a list of all the \"lanelets\" that need to be\n",
    "    included in a build. The output will typically be written to a\n",
    "    tab-delmited file, either to use as input to vr-pipe, or perhaps as\n",
    "    input to a vr-track DB.\n",
    "\n",
    "    Args:\n",
    "        taxon_ids (list of int): The taxons of the build (P. falciparum is '5833',\n",
    "            P. vivax is '5855').\n",
    "        sequencescape_alfresco_study_mappings_fn (str): filename of\n",
    "            mappings from sequencscape to alfresco study mappings\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The build manifest.\n",
    "\n",
    "    \"\"\"\n",
    "    \n",
    "    # Read in taxon exceptions file\n",
    "    df_mlwh_taxon_exceptions = pd.read_csv(mlwh_taxon_exceptions_fn, sep='\\t', dtype={'tag_index': str, 'taxon_id': int}, index_col='irods_filename')\n",
    "    df_mlwh_taxon_exceptions['tag_index'].fillna('', inplace=True)\n",
    "    # Identify which samples in exceptions file match the taxon of current build, and which don't\n",
    "    df_mlwh_exceptions_this_taxon = df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)]\n",
    "    df_mlwh_exceptions_other_taxa = df_mlwh_taxon_exceptions.loc[~(df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids))]\n",
    "\n",
    "    # Read in sample exceptions file\n",
    "    df_mlwh_sample_exceptions = pd.read_csv(mlwh_sample_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in study exceptions file\n",
    "    df_mlwh_study_exceptions = pd.read_csv(mlwh_study_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in missing exceptions file\n",
    "    df_mlwh_missing_exceptions = pd.read_csv(mlwh_missing_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "    df_mlwh_missing_exceptions['derivative_sample_id'] = 'DS_' + df_mlwh_missing_exceptions['sample_id']\n",
    "    df_mlwh_missing_exceptions = df_mlwh_missing_exceptions.loc[\n",
    "        df_mlwh_missing_exceptions['taxon_id'].isin(taxon_ids)\n",
    "    ]\n",
    "\n",
    "    # Read in SequenceScape-Alfresco study mappings\n",
    "    df_sequencescape_alfresco_study_mappings = pd.read_csv(sequencescape_alfresco_study_mappings_fn, sep='\\t', index_col='seqscape_study_id')\n",
    "    sequencescape_alfresco_study_mappings_dict = df_sequencescape_alfresco_study_mappings.loc[\n",
    "        :,\n",
    "#         df_sequencescape_alfresco_study_mappings['build_flag']==1,\n",
    "        'alfresco_study_code'\n",
    "    ].to_dict()\n",
    "    \n",
    "    # Read in data from mlwh matching this taxon, plus samples from exceptions file matching this taxon\n",
    "    conn = pymysql.connect(\n",
    "        host='mlwh-db',\n",
    "        user='mlwh_malaria',\n",
    "        password='Solaris&2015',\n",
    "        db='mlwarehouse',\n",
    "        port=3435\n",
    "    )\n",
    "    \n",
    "    sql_query = 'SELECT \\\n",
    "        study.name as study_name, \\\n",
    "        study.id_study_lims as study_lims, \\\n",
    "        sample.supplier_name as sample, \\\n",
    "        sample.name, \\\n",
    "        sample.sanger_sample_id, \\\n",
    "        sample.taxon_id, \\\n",
    "        iseq_product_metrics.id_run, \\\n",
    "        iseq_product_metrics.position, \\\n",
    "        iseq_product_metrics.tag_index, \\\n",
    "        iseq_product_metrics.num_reads, \\\n",
    "        iseq_product_metrics.human_percent_mapped, \\\n",
    "        iseq_run_lane_metrics.instrument_name, \\\n",
    "        iseq_run_lane_metrics.instrument_model, \\\n",
    "        iseq_run_lane_metrics.paired_read, \\\n",
    "        iseq_run_lane_metrics.qc_complete, \\\n",
    "        iseq_flowcell.manual_qc, \\\n",
    "        iseq_flowcell.requested_insert_size_from, \\\n",
    "        iseq_flowcell.requested_insert_size_to, \\\n",
    "        iseq_flowcell.forward_read_length, \\\n",
    "        iseq_run_status_dict.description \\\n",
    "    FROM \\\n",
    "        study, \\\n",
    "        iseq_flowcell, \\\n",
    "        sample, \\\n",
    "        iseq_product_metrics, \\\n",
    "        iseq_run_status, \\\n",
    "        iseq_run_lane_metrics, \\\n",
    "        iseq_run_status_dict \\\n",
    "    WHERE \\\n",
    "        study.id_study_tmp = iseq_flowcell.id_study_tmp and \\\n",
    "        iseq_flowcell.id_sample_tmp = sample.id_sample_tmp and \\\n",
    "        iseq_flowcell.manual_qc = 1 and \\\n",
    "        iseq_product_metrics.id_iseq_flowcell_tmp = iseq_flowcell.id_iseq_flowcell_tmp and \\\n",
    "        iseq_run_status.id_run = iseq_product_metrics.id_run and \\\n",
    "        iseq_product_metrics.id_run = iseq_run_lane_metrics.id_run and \\\n",
    "        iseq_product_metrics.position = iseq_run_lane_metrics.position and \\\n",
    "        iseq_run_status.iscurrent = 1 and \\\n",
    "        ( ( iseq_run_lane_metrics.instrument_model != \"MiSeq\" ) or (iseq_product_metrics.id_run in (%s)) ) and \\\n",
    "        iseq_run_status.id_run_status_dict = iseq_run_status_dict.id_run_status_dict and \\\n",
    "        study.faculty_sponsor = \"Dominic Kwiatkowski\" and \\\n",
    "        ( ( sample.taxon_id in (%s) ) or ' % (\n",
    "            ', '.join([str(x) for x in miseq_runs_to_allow]),\n",
    "            ', '.join([str(x) for x in taxon_ids])\n",
    "        )\n",
    "    sql_query = sql_query + ' or '.join(\n",
    "        df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)].apply(\n",
    "            lambda x: '(iseq_product_metrics.id_run=\"%s\" and iseq_product_metrics.position=\"%s\" and iseq_product_metrics.tag_index=\"%s\")' % (x['id_run'], x['position'], x['tag_index']),\n",
    "            1\n",
    "        )\n",
    "    )\n",
    "    sql_query = sql_query + ')'\n",
    "#         (iseq_flowcell.manual_qc = 1 or iseq_flowcell.manual_qc is null) and \\\n",
    "    \n",
    "    df_return = pd.read_sql(sql_query, conn)\n",
    "\n",
    "    # Replace missing taxon_id with -1 (can't have missing int values in pandas Series)\n",
    "    df_return['taxon_id'] = df_return['taxon_id'].fillna(-1).astype('int32')\n",
    "\n",
    "    # Determine file name in iRods\n",
    "    df_return['irods_filename'] = df_return.apply(set_irods_name, 1)\n",
    "    df_return.set_index('irods_filename', inplace=True)\n",
    "\n",
    "    # Determine alfresco study and change any incorrect studies\n",
    "    df_return['alfresco_study_code'] = df_return['study_lims'].apply(\n",
    "        lambda x: sequencescape_alfresco_study_mappings_dict[int(x)] if int(x) in sequencescape_alfresco_study_mappings_dict else ''\n",
    "    )\n",
    "    empty_alfresco_study_code = (df_return['alfresco_study_code'] == '')\n",
    "    df_return.loc[empty_alfresco_study_code, 'alfresco_study_code'] = df_return.loc[empty_alfresco_study_code, 'study_name']\n",
    "    \n",
    "    study_exception_indexes = df_mlwh_study_exceptions.index[\n",
    "        df_mlwh_study_exceptions.index.isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[study_exception_indexes, 'alfresco_study_code'] = df_mlwh_study_exceptions.loc[study_exception_indexes, 'alfresco_study_code']\n",
    "\n",
    "    # Change any incorrect taxon IDs\n",
    "    taxon_exception_indexes = df_mlwh_exceptions_this_taxon.index[\n",
    "        df_mlwh_exceptions_this_taxon.index.to_series().isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[taxon_exception_indexes, 'taxon_id'] = df_mlwh_exceptions_this_taxon.loc[taxon_exception_indexes, 'taxon_id']\n",
    "\n",
    "    # Determine sample ID and change any incorrect sample IDs\n",
    "    df_return['derivative_sample_id'] = df_return.apply(set_sample_id, 1)\n",
    "    sample_exception_indexes = df_mlwh_sample_exceptions.index[\n",
    "        df_mlwh_sample_exceptions.index.isin(df_return.index)\n",
    "    ]    \n",
    "    df_return.loc[sample_exception_indexes, 'derivative_sample_id'] = 'DS_' + df_mlwh_sample_exceptions.loc[sample_exception_indexes, 'sample_id']\n",
    "    df_return.drop(['sample', 'name'], axis=1, inplace=True)\n",
    "    \n",
    "    # Remove any samples that are not in this taxon\n",
    "    df_return = df_return.loc[~df_return.index.isin(df_mlwh_exceptions_other_taxa.index)]\n",
    "    \n",
    "    # Merge in missing exceptions\n",
    "    df_return = df_return.append(df_mlwh_missing_exceptions.drop('study_group', 1))\n",
    "    \n",
    "    # Remove any samples that don't have an alfresco study\n",
    "    invalid_codes = ~df_return['alfresco_study_code'].apply(is_valid_alfresco_code)\n",
    "    if(np.count_nonzero(invalid_codes) > 0):\n",
    "        print('Removing %d files with invalid alfresco_study_code:' % np.count_nonzero(invalid_codes))\n",
    "        print(df_return['alfresco_study_code'].loc[invalid_codes].value_counts())\n",
    "        df_return = df_return.loc[~invalid_codes]\n",
    "\n",
    "    # Determine filename in subtrack (for very old samples, there is only a .srf file in subtrack)\n",
    "    # Note we determined the id_run cutoff (5750) for bam/srf after looking at previous runs. There seems\n",
    "    # to be a grey area between runs 5500 and 5750 where some files are srf and some are bam\n",
    "    df_return['subtrack_filename'] = df_return.index\n",
    "    df_return.loc[df_return['id_run'] < 5750, 'subtrack_filename'] = df_return.loc[df_return['id_run'] < 5750, 'subtrack_filename'].apply(lambda x: x.replace('.bam', '.srf'))\n",
    "    \n",
    "    # Read in data from substrack, most importantly to get run accessions\n",
    "    sql_query = \"\\\n",
    "    SELECT \\\n",
    "        files.file_name as irods_filename, \\\n",
    "        files.bytes as subtrack_files_bytes, \\\n",
    "        files.timestamp as subtrack_files_timestamp, \\\n",
    "        files.public_date as subtrack_files_public_date, \\\n",
    "        submission.id as subtrack_submission_id, \\\n",
    "        submission.created as subtrack_submission_created, \\\n",
    "        submission.release_date as subtrack_submission_release_date, \\\n",
    "        submission.timestamp as subtrack_submission_timestamp, \\\n",
    "        submission.ext_db as subtrack_submission_ext_db, \\\n",
    "        submission.ebi_sample_acc, \\\n",
    "        submission.ebi_exp_acc, \\\n",
    "        submission.ebi_study_acc, \\\n",
    "        submission.ebi_sub_acc, \\\n",
    "        submission.ebi_run_acc \\\n",
    "    FROM \\\n",
    "        submission, \\\n",
    "        files \\\n",
    "    WHERE \\\n",
    "        files.sub_id = submission.id AND \\\n",
    "        files.file_name in (%s)\\\n",
    "    \" % (\"'\" + \"', '\".join(df_return['subtrack_filename']) + \"'\")\n",
    "\n",
    "    conn= pymysql.connect(\n",
    "        host='shap',\n",
    "        user='ega_dataset',\n",
    "        password='ega_dataset',\n",
    "        db='subtrack',\n",
    "        port=3303\n",
    "    )\n",
    "\n",
    "    df_run_accessions = pd.read_sql(sql_query, conn).set_index('irods_filename')\n",
    "    \n",
    "    # Merge in subtrack data\n",
    "    df_return = df_return.join(df_run_accessions, on='subtrack_filename', how='left')\n",
    "\n",
    "    # Remove unwanted studies\n",
    "    df_return = df_return.loc[~df_return['alfresco_study_code'].isin(studies_to_exclude)]\n",
    "\n",
    "    # Create other columns\n",
    "    df_return.rename(columns={'alfresco_study_code': 'study', 'paired_read': 'paired'}, inplace=True)\n",
    "    df_return['lane'] = df_return.index.to_series().apply(lambda x: x.split('.')[0])\n",
    "    df_return['path'] = df_return.apply(lambda x: \"%s/%s\" % (input_dir, x.name), 1)\n",
    "    df_return['reads'] = df_return['num_reads'].fillna(-1).astype(int)\n",
    "    df_return['irods_path'] = df_return.apply(lambda x: \"/seq/%d/%s\" % (x['id_run'], x.name), 1)\n",
    "    df_return['sample'] = df_return['derivative_sample_id'].str.replace('DS_', '')\n",
    "\n",
    "    # Remove lanes with zero reads\n",
    "    df_return = df_return.loc[df_return['reads'] != 0]\n",
    "    \n",
    "    # Remove control samples (see emails from Sonia 03/05/2018 12:20 and from Richard 09/10/2018 13:26)\n",
    "#     df_return = df_return.loc[df_return['sample'].str.slice(0, 1) != 'C']\n",
    "    df_return = df_return.loc[(df_return['sample'].str.slice(0, 1) != 'C') & (df_return['sample'] != 'control')]\n",
    "    \n",
    "    # Restrict to certain columns\n",
    "    if output_columns is not None:\n",
    "        df_return = (\n",
    "            df_return[output_columns]\n",
    "            .sort_values(['study', 'sample', 'id_run', 'position', 'tag_index'])\n",
    "        )\n",
    "    \n",
    "    print(df_return.shape)\n",
    "\n",
    "    return(df_return) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing 63 files with invalid alfresco_study_code:\n",
      "AT-rich amplification                                 53\n",
      "Test sequencing of high level human contamination.     6\n",
      "Plasmodium falciparum genome variation 1               4\n",
      "Name: alfresco_study_code, dtype: int64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/nfs/users/nfs_r/rp7/miniconda3/envs/biipy/lib/python3.5/site-packages/pymysql/cursors.py:166: Warning: (3170, \"Memory capacity of 8388608 bytes for 'range_optimizer_max_mem_size' exceeded. Range optimization was not done for this query.\")\n",
      "  result = self._query(query)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(18130, 26)\n"
     ]
    }
   ],
   "source": [
    "df_assay_data = create_build_manifest()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>irods_path</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1276_5_nonhuman.bam</th>\n",
       "      <td>dummy/1276_5_nonhuman.bam</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1276_5_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/1276/1276_5_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_5_nonhuman.srf</td>\n",
       "      <td>5.464048e+09</td>\n",
       "      <td>ERR012350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1276_6_nonhuman.bam</th>\n",
       "      <td>dummy/1276_6_nonhuman.bam</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1276_6_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/1276/1276_6_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1276</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1276_6_nonhuman.srf</td>\n",
       "      <td>5.479595e+09</td>\n",
       "      <td>ERR012360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1333_8_nonhuman.bam</th>\n",
       "      <td>dummy/1333_8_nonhuman.bam</td>\n",
       "      <td>1001-PF-ML-DJIMDE</td>\n",
       "      <td>PM0001-C</td>\n",
       "      <td>1333_8_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/1333/1333_8_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>1333</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-09-24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1333_8_nonhuman.srf</td>\n",
       "      <td>5.136273e+09</td>\n",
       "      <td>ERR012305</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          path              study    sample  \\\n",
       "irods_filename                                                                \n",
       "1276_5_nonhuman.bam  dummy/1276_5_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C   \n",
       "1276_6_nonhuman.bam  dummy/1276_6_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C   \n",
       "1333_8_nonhuman.bam  dummy/1333_8_nonhuman.bam  1001-PF-ML-DJIMDE  PM0001-C   \n",
       "\n",
       "                                lane  reads  paired  \\\n",
       "irods_filename                                        \n",
       "1276_5_nonhuman.bam  1276_5_nonhuman     -1     NaN   \n",
       "1276_6_nonhuman.bam  1276_6_nonhuman     -1     NaN   \n",
       "1333_8_nonhuman.bam  1333_8_nonhuman     -1     NaN   \n",
       "\n",
       "                                        irods_path sanger_sample_id  taxon_id  \\\n",
       "irods_filename                                                                  \n",
       "1276_5_nonhuman.bam  /seq/1276/1276_5_nonhuman.bam              NaN      5833   \n",
       "1276_6_nonhuman.bam  /seq/1276/1276_6_nonhuman.bam              NaN      5833   \n",
       "1333_8_nonhuman.bam  /seq/1333/1333_8_nonhuman.bam              NaN      5833   \n",
       "\n",
       "                    study_lims                                study_name  \\\n",
       "irods_filename                                                             \n",
       "1276_5_nonhuman.bam        578  Plasmodium falciparum genome variation 1   \n",
       "1276_6_nonhuman.bam        578  Plasmodium falciparum genome variation 1   \n",
       "1333_8_nonhuman.bam        578  Plasmodium falciparum genome variation 1   \n",
       "\n",
       "                     id_run  position  tag_index qc_complete  manual_qc  \\\n",
       "irods_filename                                                            \n",
       "1276_5_nonhuman.bam    1276         5        NaN  2008-09-24        NaN   \n",
       "1276_6_nonhuman.bam    1276         6        NaN  2008-09-24        NaN   \n",
       "1333_8_nonhuman.bam    1333         8        NaN  2008-09-24        NaN   \n",
       "\n",
       "                    description instrument_name instrument_model  \\\n",
       "irods_filename                                                     \n",
       "1276_5_nonhuman.bam         NaN             NaN            HiSeq   \n",
       "1276_6_nonhuman.bam         NaN             NaN            HiSeq   \n",
       "1333_8_nonhuman.bam         NaN             NaN            HiSeq   \n",
       "\n",
       "                     forward_read_length  requested_insert_size_from  \\\n",
       "irods_filename                                                         \n",
       "1276_5_nonhuman.bam                  100                         NaN   \n",
       "1276_6_nonhuman.bam                  100                         NaN   \n",
       "1333_8_nonhuman.bam                  100                         NaN   \n",
       "\n",
       "                     requested_insert_size_to  human_percent_mapped  \\\n",
       "irods_filename                                                        \n",
       "1276_5_nonhuman.bam                       NaN                   NaN   \n",
       "1276_6_nonhuman.bam                       NaN                   NaN   \n",
       "1333_8_nonhuman.bam                       NaN                   NaN   \n",
       "\n",
       "                       subtrack_filename  subtrack_files_bytes ebi_run_acc  \n",
       "irods_filename                                                              \n",
       "1276_5_nonhuman.bam  1276_5_nonhuman.srf          5.464048e+09   ERR012350  \n",
       "1276_6_nonhuman.bam  1276_6_nonhuman.srf          5.479595e+09   ERR012360  \n",
       "1333_8_nonhuman.bam  1333_8_nonhuman.srf          5.136273e+09   ERR012305  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_columns = 50\n",
    "df_assay_data[0:3]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find files found using mlwh but not in subtrack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "id_run  study                  \n",
       "245     1032-PF-BRHN-SMITHEE        2\n",
       "368     1032-PF-BRHN-SMITHEE        2\n",
       "2128    1004-PF-BF-OUEDRAOGO        5\n",
       "2217    1013-PF-PEGB-BRANCH         1\n",
       "2575    1015-PF-KE-NZILA            1\n",
       "4060    1004-PF-BF-OUEDRAOGO        3\n",
       "        1016-PF-TH-NOSTEN           1\n",
       "        1032-PF-BRHN-SMITHEE        1\n",
       "5534    1043-PF-GB-RAYNER           1\n",
       "5554    1017-PF-GH-AMENGA-ETEGO     1\n",
       "5557    1017-PF-GH-AMENGA-ETEGO     1\n",
       "5597    1015-PF-KE-NZILA            2\n",
       "5696    1004-PF-BF-OUEDRAOGO        1\n",
       "        1012-PF-KH-WHITE            1\n",
       "5717    1044-PF-KH-FAIRHURST        1\n",
       "10901   1155-PF-ID-PRICE            6\n",
       "11019   1155-PF-ID-PRICE            2\n",
       "12299   1155-PF-ID-PRICE           14\n",
       "14323   1155-PF-ID-PRICE            1\n",
       "14324   1155-PF-ID-PRICE            3\n",
       "16166   1155-PF-ID-PRICE           36\n",
       "16189   1155-PF-ID-PRICE           11\n",
       "16229   1155-PF-ID-PRICE           19\n",
       "25237   1155-PF-ID-PRICE           37\n",
       "25552   1155-PF-ID-PRICE            3\n",
       "dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_assay_data.loc[df_assay_data['ebi_run_acc'].isnull(), ['id_run', 'study']].groupby(['id_run', 'study']).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>irods_path</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5554_1_nonhuman.bam</th>\n",
       "      <td>dummy/5554_1_nonhuman.bam</td>\n",
       "      <td>1017-PF-GH-AMENGA-ETEGO</td>\n",
       "      <td>PF0176-C</td>\n",
       "      <td>5554_1_nonhuman</td>\n",
       "      <td>62084346</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/5554/5554_1_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>5554</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2010-12-21 23:55:44</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL27</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>87.64</td>\n",
       "      <td>5554_1_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5557_1_nonhuman.bam</th>\n",
       "      <td>dummy/5557_1_nonhuman.bam</td>\n",
       "      <td>1017-PF-GH-AMENGA-ETEGO</td>\n",
       "      <td>PF0264-C</td>\n",
       "      <td>5557_1_nonhuman</td>\n",
       "      <td>60035853</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/5557/5557_1_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>5557</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2010-12-20 16:00:42</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL9</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>200.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>97.99</td>\n",
       "      <td>5557_1_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          path                    study  \\\n",
       "irods_filename                                                            \n",
       "5554_1_nonhuman.bam  dummy/5554_1_nonhuman.bam  1017-PF-GH-AMENGA-ETEGO   \n",
       "5557_1_nonhuman.bam  dummy/5557_1_nonhuman.bam  1017-PF-GH-AMENGA-ETEGO   \n",
       "\n",
       "                       sample             lane     reads  paired  \\\n",
       "irods_filename                                                     \n",
       "5554_1_nonhuman.bam  PF0176-C  5554_1_nonhuman  62084346     1.0   \n",
       "5557_1_nonhuman.bam  PF0264-C  5557_1_nonhuman  60035853     1.0   \n",
       "\n",
       "                                        irods_path sanger_sample_id  taxon_id  \\\n",
       "irods_filename                                                                  \n",
       "5554_1_nonhuman.bam  /seq/5554/5554_1_nonhuman.bam             None      5833   \n",
       "5557_1_nonhuman.bam  /seq/5557/5557_1_nonhuman.bam             None      5833   \n",
       "\n",
       "                    study_lims                                study_name  \\\n",
       "irods_filename                                                             \n",
       "5554_1_nonhuman.bam        578  Plasmodium falciparum genome variation 1   \n",
       "5557_1_nonhuman.bam        578  Plasmodium falciparum genome variation 1   \n",
       "\n",
       "                     id_run  position  tag_index          qc_complete  \\\n",
       "irods_filename                                                          \n",
       "5554_1_nonhuman.bam    5554         1        NaN  2010-12-21 23:55:44   \n",
       "5557_1_nonhuman.bam    5557         1        NaN  2010-12-20 16:00:42   \n",
       "\n",
       "                     manual_qc  description instrument_name instrument_model  \\\n",
       "irods_filename                                                                 \n",
       "5554_1_nonhuman.bam        1.0  qc complete            IL27               HK   \n",
       "5557_1_nonhuman.bam        1.0  qc complete             IL9               HK   \n",
       "\n",
       "                     forward_read_length  requested_insert_size_from  \\\n",
       "irods_filename                                                         \n",
       "5554_1_nonhuman.bam                   76                       200.0   \n",
       "5557_1_nonhuman.bam                   76                       200.0   \n",
       "\n",
       "                     requested_insert_size_to  human_percent_mapped  \\\n",
       "irods_filename                                                        \n",
       "5554_1_nonhuman.bam                     300.0                 87.64   \n",
       "5557_1_nonhuman.bam                     300.0                 97.99   \n",
       "\n",
       "                       subtrack_filename  subtrack_files_bytes ebi_run_acc  \n",
       "irods_filename                                                              \n",
       "5554_1_nonhuman.bam  5554_1_nonhuman.srf                   NaN         NaN  \n",
       "5557_1_nonhuman.bam  5557_1_nonhuman.srf                   NaN         NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_assay_data.loc[df_assay_data['ebi_run_acc'].isnull() & (df_assay_data['study'] == '1017-PF-GH-AMENGA-ETEGO')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>irods_path</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2128_7_nonhuman.bam</th>\n",
       "      <td>dummy/2128_7_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0061-C</td>\n",
       "      <td>2128_7_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/2128/2128_7_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>2128</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-02-20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2128_7_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2128_8_nonhuman.bam</th>\n",
       "      <td>dummy/2128_8_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0061-C</td>\n",
       "      <td>2128_8_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/2128/2128_8_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>2128</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-02-20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2128_8_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2128_2_nonhuman.bam</th>\n",
       "      <td>dummy/2128_2_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0067-C</td>\n",
       "      <td>2128_2_nonhuman</td>\n",
       "      <td>15813522</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/2128/2128_2_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>2128</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-02-20 15:24:50</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL8</td>\n",
       "      <td>HK</td>\n",
       "      <td>54</td>\n",
       "      <td>300.0</td>\n",
       "      <td>600.0</td>\n",
       "      <td>98.33</td>\n",
       "      <td>2128_2_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2128_3_nonhuman.bam</th>\n",
       "      <td>dummy/2128_3_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0067-C</td>\n",
       "      <td>2128_3_nonhuman</td>\n",
       "      <td>15907048</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/2128/2128_3_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>2128</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-02-20 15:24:50</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL8</td>\n",
       "      <td>HK</td>\n",
       "      <td>54</td>\n",
       "      <td>300.0</td>\n",
       "      <td>600.0</td>\n",
       "      <td>98.28</td>\n",
       "      <td>2128_3_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5696_2_nonhuman.bam</th>\n",
       "      <td>dummy/5696_2_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0067-C</td>\n",
       "      <td>5696_2_nonhuman</td>\n",
       "      <td>153418043</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/5696/5696_2_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>5696</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2011-02-01 03:44:19</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HS6</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>75</td>\n",
       "      <td>300.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>98.07</td>\n",
       "      <td>5696_2_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2128_6_nonhuman.bam</th>\n",
       "      <td>dummy/2128_6_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0068-C</td>\n",
       "      <td>2128_6_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/2128/2128_6_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>2128</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-02-20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HiSeq</td>\n",
       "      <td>100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2128_6_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4060_6_nonhuman.bam</th>\n",
       "      <td>dummy/4060_6_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0068-C</td>\n",
       "      <td>4060_6_nonhuman</td>\n",
       "      <td>39410448</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/4060/4060_6_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>578</td>\n",
       "      <td>Plasmodium falciparum genome variation 1</td>\n",
       "      <td>4060</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-11-25 14:39:42</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL12</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>300.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>93.11</td>\n",
       "      <td>4060_6_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4060_7_nonhuman.bam</th>\n",
       "      <td>dummy/4060_7_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0068-C</td>\n",
       "      <td>4060_7_nonhuman</td>\n",
       "      <td>40244830</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/4060/4060_7_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>546</td>\n",
       "      <td>Plasmodium falciparum Illumina sequencing R&amp;D 1</td>\n",
       "      <td>4060</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-11-25 14:39:42</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL12</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>300.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>86.54</td>\n",
       "      <td>4060_7_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4060_8_nonhuman.bam</th>\n",
       "      <td>dummy/4060_8_nonhuman.bam</td>\n",
       "      <td>1004-PF-BF-OUEDRAOGO</td>\n",
       "      <td>PK0068-C</td>\n",
       "      <td>4060_8_nonhuman</td>\n",
       "      <td>36472202</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/4060/4060_8_nonhuman.bam</td>\n",
       "      <td>None</td>\n",
       "      <td>5833</td>\n",
       "      <td>546</td>\n",
       "      <td>Plasmodium falciparum Illumina sequencing R&amp;D 1</td>\n",
       "      <td>4060</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2009-11-25 14:39:42</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>IL12</td>\n",
       "      <td>HK</td>\n",
       "      <td>76</td>\n",
       "      <td>300.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>88.40</td>\n",
       "      <td>4060_8_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          path                 study  \\\n",
       "irods_filename                                                         \n",
       "2128_7_nonhuman.bam  dummy/2128_7_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "2128_8_nonhuman.bam  dummy/2128_8_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "2128_2_nonhuman.bam  dummy/2128_2_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "2128_3_nonhuman.bam  dummy/2128_3_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "5696_2_nonhuman.bam  dummy/5696_2_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "2128_6_nonhuman.bam  dummy/2128_6_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "4060_6_nonhuman.bam  dummy/4060_6_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "4060_7_nonhuman.bam  dummy/4060_7_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "4060_8_nonhuman.bam  dummy/4060_8_nonhuman.bam  1004-PF-BF-OUEDRAOGO   \n",
       "\n",
       "                       sample             lane      reads  paired  \\\n",
       "irods_filename                                                      \n",
       "2128_7_nonhuman.bam  PK0061-C  2128_7_nonhuman         -1     NaN   \n",
       "2128_8_nonhuman.bam  PK0061-C  2128_8_nonhuman         -1     NaN   \n",
       "2128_2_nonhuman.bam  PK0067-C  2128_2_nonhuman   15813522     1.0   \n",
       "2128_3_nonhuman.bam  PK0067-C  2128_3_nonhuman   15907048     1.0   \n",
       "5696_2_nonhuman.bam  PK0067-C  5696_2_nonhuman  153418043     1.0   \n",
       "2128_6_nonhuman.bam  PK0068-C  2128_6_nonhuman         -1     NaN   \n",
       "4060_6_nonhuman.bam  PK0068-C  4060_6_nonhuman   39410448     1.0   \n",
       "4060_7_nonhuman.bam  PK0068-C  4060_7_nonhuman   40244830     1.0   \n",
       "4060_8_nonhuman.bam  PK0068-C  4060_8_nonhuman   36472202     1.0   \n",
       "\n",
       "                                        irods_path sanger_sample_id  taxon_id  \\\n",
       "irods_filename                                                                  \n",
       "2128_7_nonhuman.bam  /seq/2128/2128_7_nonhuman.bam              NaN      5833   \n",
       "2128_8_nonhuman.bam  /seq/2128/2128_8_nonhuman.bam              NaN      5833   \n",
       "2128_2_nonhuman.bam  /seq/2128/2128_2_nonhuman.bam             None      5833   \n",
       "2128_3_nonhuman.bam  /seq/2128/2128_3_nonhuman.bam             None      5833   \n",
       "5696_2_nonhuman.bam  /seq/5696/5696_2_nonhuman.bam             None      5833   \n",
       "2128_6_nonhuman.bam  /seq/2128/2128_6_nonhuman.bam              NaN      5833   \n",
       "4060_6_nonhuman.bam  /seq/4060/4060_6_nonhuman.bam             None      5833   \n",
       "4060_7_nonhuman.bam  /seq/4060/4060_7_nonhuman.bam             None      5833   \n",
       "4060_8_nonhuman.bam  /seq/4060/4060_8_nonhuman.bam             None      5833   \n",
       "\n",
       "                    study_lims  \\\n",
       "irods_filename                   \n",
       "2128_7_nonhuman.bam        578   \n",
       "2128_8_nonhuman.bam        578   \n",
       "2128_2_nonhuman.bam        578   \n",
       "2128_3_nonhuman.bam        578   \n",
       "5696_2_nonhuman.bam        578   \n",
       "2128_6_nonhuman.bam        578   \n",
       "4060_6_nonhuman.bam        578   \n",
       "4060_7_nonhuman.bam        546   \n",
       "4060_8_nonhuman.bam        546   \n",
       "\n",
       "                                                          study_name  id_run  \\\n",
       "irods_filename                                                                 \n",
       "2128_7_nonhuman.bam         Plasmodium falciparum genome variation 1    2128   \n",
       "2128_8_nonhuman.bam         Plasmodium falciparum genome variation 1    2128   \n",
       "2128_2_nonhuman.bam         Plasmodium falciparum genome variation 1    2128   \n",
       "2128_3_nonhuman.bam         Plasmodium falciparum genome variation 1    2128   \n",
       "5696_2_nonhuman.bam         Plasmodium falciparum genome variation 1    5696   \n",
       "2128_6_nonhuman.bam         Plasmodium falciparum genome variation 1    2128   \n",
       "4060_6_nonhuman.bam         Plasmodium falciparum genome variation 1    4060   \n",
       "4060_7_nonhuman.bam  Plasmodium falciparum Illumina sequencing R&D 1    4060   \n",
       "4060_8_nonhuman.bam  Plasmodium falciparum Illumina sequencing R&D 1    4060   \n",
       "\n",
       "                     position  tag_index          qc_complete  manual_qc  \\\n",
       "irods_filename                                                             \n",
       "2128_7_nonhuman.bam         7        NaN           2009-02-20        NaN   \n",
       "2128_8_nonhuman.bam         8        NaN           2009-02-20        NaN   \n",
       "2128_2_nonhuman.bam         2        NaN  2009-02-20 15:24:50        1.0   \n",
       "2128_3_nonhuman.bam         3        NaN  2009-02-20 15:24:50        1.0   \n",
       "5696_2_nonhuman.bam         2        NaN  2011-02-01 03:44:19        1.0   \n",
       "2128_6_nonhuman.bam         6        NaN           2009-02-20        NaN   \n",
       "4060_6_nonhuman.bam         6        NaN  2009-11-25 14:39:42        1.0   \n",
       "4060_7_nonhuman.bam         7        NaN  2009-11-25 14:39:42        1.0   \n",
       "4060_8_nonhuman.bam         8        NaN  2009-11-25 14:39:42        1.0   \n",
       "\n",
       "                     description instrument_name instrument_model  \\\n",
       "irods_filename                                                      \n",
       "2128_7_nonhuman.bam          NaN             NaN            HiSeq   \n",
       "2128_8_nonhuman.bam          NaN             NaN            HiSeq   \n",
       "2128_2_nonhuman.bam  qc complete             IL8               HK   \n",
       "2128_3_nonhuman.bam  qc complete             IL8               HK   \n",
       "5696_2_nonhuman.bam  qc complete             HS6            HiSeq   \n",
       "2128_6_nonhuman.bam          NaN             NaN            HiSeq   \n",
       "4060_6_nonhuman.bam  qc complete            IL12               HK   \n",
       "4060_7_nonhuman.bam  qc complete            IL12               HK   \n",
       "4060_8_nonhuman.bam  qc complete            IL12               HK   \n",
       "\n",
       "                     forward_read_length  requested_insert_size_from  \\\n",
       "irods_filename                                                         \n",
       "2128_7_nonhuman.bam                  100                         NaN   \n",
       "2128_8_nonhuman.bam                  100                         NaN   \n",
       "2128_2_nonhuman.bam                   54                       300.0   \n",
       "2128_3_nonhuman.bam                   54                       300.0   \n",
       "5696_2_nonhuman.bam                   75                       300.0   \n",
       "2128_6_nonhuman.bam                  100                         NaN   \n",
       "4060_6_nonhuman.bam                   76                       300.0   \n",
       "4060_7_nonhuman.bam                   76                       300.0   \n",
       "4060_8_nonhuman.bam                   76                       300.0   \n",
       "\n",
       "                     requested_insert_size_to  human_percent_mapped  \\\n",
       "irods_filename                                                        \n",
       "2128_7_nonhuman.bam                       NaN                   NaN   \n",
       "2128_8_nonhuman.bam                       NaN                   NaN   \n",
       "2128_2_nonhuman.bam                     600.0                 98.33   \n",
       "2128_3_nonhuman.bam                     600.0                 98.28   \n",
       "5696_2_nonhuman.bam                     300.0                 98.07   \n",
       "2128_6_nonhuman.bam                       NaN                   NaN   \n",
       "4060_6_nonhuman.bam                     300.0                 93.11   \n",
       "4060_7_nonhuman.bam                     300.0                 86.54   \n",
       "4060_8_nonhuman.bam                     300.0                 88.40   \n",
       "\n",
       "                       subtrack_filename  subtrack_files_bytes ebi_run_acc  \n",
       "irods_filename                                                              \n",
       "2128_7_nonhuman.bam  2128_7_nonhuman.srf                   NaN         NaN  \n",
       "2128_8_nonhuman.bam  2128_8_nonhuman.srf                   NaN         NaN  \n",
       "2128_2_nonhuman.bam  2128_2_nonhuman.srf                   NaN         NaN  \n",
       "2128_3_nonhuman.bam  2128_3_nonhuman.srf                   NaN         NaN  \n",
       "5696_2_nonhuman.bam  5696_2_nonhuman.srf                   NaN         NaN  \n",
       "2128_6_nonhuman.bam  2128_6_nonhuman.srf                   NaN         NaN  \n",
       "4060_6_nonhuman.bam  4060_6_nonhuman.srf                   NaN         NaN  \n",
       "4060_7_nonhuman.bam  4060_7_nonhuman.srf                   NaN         NaN  \n",
       "4060_8_nonhuman.bam  4060_8_nonhuman.srf                   NaN         NaN  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_assay_data.loc[df_assay_data['ebi_run_acc'].isnull() & (df_assay_data['study'] == '1004-PF-BF-OUEDRAOGO')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>irods_path</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25552_1#7.cram</th>\n",
       "      <td>dummy/25552_1#7.cram</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>SPT24658</td>\n",
       "      <td>25552_1#7</td>\n",
       "      <td>41337656</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/25552/25552_1#7.cram</td>\n",
       "      <td>5101STDY7247317</td>\n",
       "      <td>5833</td>\n",
       "      <td>5101</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>25552</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2018-04-05 12:42:30</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HX7</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>96.29</td>\n",
       "      <td>25552_1#7.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25552_1#10.cram</th>\n",
       "      <td>dummy/25552_1#10.cram</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>SPT24662</td>\n",
       "      <td>25552_1#10</td>\n",
       "      <td>38311728</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/25552/25552_1#10.cram</td>\n",
       "      <td>5101STDY7247321</td>\n",
       "      <td>5833</td>\n",
       "      <td>5101</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>25552</td>\n",
       "      <td>1</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2018-04-05 12:42:30</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HX7</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>95.05</td>\n",
       "      <td>25552_1#10.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25552_1#21.cram</th>\n",
       "      <td>dummy/25552_1#21.cram</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>SPT24686</td>\n",
       "      <td>25552_1#21</td>\n",
       "      <td>33226936</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/25552/25552_1#21.cram</td>\n",
       "      <td>5101STDY7247345</td>\n",
       "      <td>5833</td>\n",
       "      <td>5101</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>25552</td>\n",
       "      <td>1</td>\n",
       "      <td>21.0</td>\n",
       "      <td>2018-04-05 12:42:30</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HX7</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>97.55</td>\n",
       "      <td>25552_1#21.cram</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  path             study    sample  \\\n",
       "irods_filename                                                       \n",
       "25552_1#7.cram    dummy/25552_1#7.cram  1155-PF-ID-PRICE  SPT24658   \n",
       "25552_1#10.cram  dummy/25552_1#10.cram  1155-PF-ID-PRICE  SPT24662   \n",
       "25552_1#21.cram  dummy/25552_1#21.cram  1155-PF-ID-PRICE  SPT24686   \n",
       "\n",
       "                       lane     reads  paired                  irods_path  \\\n",
       "irods_filename                                                              \n",
       "25552_1#7.cram    25552_1#7  41337656     1.0   /seq/25552/25552_1#7.cram   \n",
       "25552_1#10.cram  25552_1#10  38311728     1.0  /seq/25552/25552_1#10.cram   \n",
       "25552_1#21.cram  25552_1#21  33226936     1.0  /seq/25552/25552_1#21.cram   \n",
       "\n",
       "                sanger_sample_id  taxon_id study_lims        study_name  \\\n",
       "irods_filename                                                            \n",
       "25552_1#7.cram   5101STDY7247317      5833       5101  1155-PF-ID-PRICE   \n",
       "25552_1#10.cram  5101STDY7247321      5833       5101  1155-PF-ID-PRICE   \n",
       "25552_1#21.cram  5101STDY7247345      5833       5101  1155-PF-ID-PRICE   \n",
       "\n",
       "                 id_run  position  tag_index          qc_complete  manual_qc  \\\n",
       "irods_filename                                                                 \n",
       "25552_1#7.cram    25552         1        7.0  2018-04-05 12:42:30        1.0   \n",
       "25552_1#10.cram   25552         1       10.0  2018-04-05 12:42:30        1.0   \n",
       "25552_1#21.cram   25552         1       21.0  2018-04-05 12:42:30        1.0   \n",
       "\n",
       "                 description instrument_name instrument_model  \\\n",
       "irods_filename                                                  \n",
       "25552_1#7.cram   qc complete             HX7           HiSeqX   \n",
       "25552_1#10.cram  qc complete             HX7           HiSeqX   \n",
       "25552_1#21.cram  qc complete             HX7           HiSeqX   \n",
       "\n",
       "                 forward_read_length  requested_insert_size_from  \\\n",
       "irods_filename                                                     \n",
       "25552_1#7.cram                   150                       450.0   \n",
       "25552_1#10.cram                  150                       450.0   \n",
       "25552_1#21.cram                  150                       450.0   \n",
       "\n",
       "                 requested_insert_size_to  human_percent_mapped  \\\n",
       "irods_filename                                                    \n",
       "25552_1#7.cram                      450.0                 96.29   \n",
       "25552_1#10.cram                     450.0                 95.05   \n",
       "25552_1#21.cram                     450.0                 97.55   \n",
       "\n",
       "                subtrack_filename  subtrack_files_bytes ebi_run_acc  \n",
       "irods_filename                                                       \n",
       "25552_1#7.cram     25552_1#7.cram                   NaN        None  \n",
       "25552_1#10.cram   25552_1#10.cram                   NaN        None  \n",
       "25552_1#21.cram   25552_1#21.cram                   NaN        None  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_assay_data.loc[df_assay_data['ebi_run_acc'].isnull() & (df_assay_data['id_run'] == 25552)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>irods_path</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_filename</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>245_7_nonhuman.bam</th>\n",
       "      <td>dummy/245_7_nonhuman.bam</td>\n",
       "      <td>1032-PF-BRHN-SMITHEE</td>\n",
       "      <td>PG0002-C</td>\n",
       "      <td>245_7_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/245/245_7_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>546</td>\n",
       "      <td>Plasmodium falciparum Illumina sequencing R&amp;D 1</td>\n",
       "      <td>245</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-03-21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1G</td>\n",
       "      <td>37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>245_7_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245_8_nonhuman.bam</th>\n",
       "      <td>dummy/245_8_nonhuman.bam</td>\n",
       "      <td>1032-PF-BRHN-SMITHEE</td>\n",
       "      <td>PG0002-C</td>\n",
       "      <td>245_8_nonhuman</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/seq/245/245_8_nonhuman.bam</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5833</td>\n",
       "      <td>546</td>\n",
       "      <td>Plasmodium falciparum Illumina sequencing R&amp;D 1</td>\n",
       "      <td>245</td>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2008-03-21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1G</td>\n",
       "      <td>37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>245_8_nonhuman.srf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        path                 study    sample  \\\n",
       "irods_filename                                                                 \n",
       "245_7_nonhuman.bam  dummy/245_7_nonhuman.bam  1032-PF-BRHN-SMITHEE  PG0002-C   \n",
       "245_8_nonhuman.bam  dummy/245_8_nonhuman.bam  1032-PF-BRHN-SMITHEE  PG0002-C   \n",
       "\n",
       "                              lane  reads  paired  \\\n",
       "irods_filename                                      \n",
       "245_7_nonhuman.bam  245_7_nonhuman     -1     NaN   \n",
       "245_8_nonhuman.bam  245_8_nonhuman     -1     NaN   \n",
       "\n",
       "                                     irods_path sanger_sample_id  taxon_id  \\\n",
       "irods_filename                                                               \n",
       "245_7_nonhuman.bam  /seq/245/245_7_nonhuman.bam              NaN      5833   \n",
       "245_8_nonhuman.bam  /seq/245/245_8_nonhuman.bam              NaN      5833   \n",
       "\n",
       "                   study_lims  \\\n",
       "irods_filename                  \n",
       "245_7_nonhuman.bam        546   \n",
       "245_8_nonhuman.bam        546   \n",
       "\n",
       "                                                         study_name  id_run  \\\n",
       "irods_filename                                                                \n",
       "245_7_nonhuman.bam  Plasmodium falciparum Illumina sequencing R&D 1     245   \n",
       "245_8_nonhuman.bam  Plasmodium falciparum Illumina sequencing R&D 1     245   \n",
       "\n",
       "                    position  tag_index qc_complete  manual_qc description  \\\n",
       "irods_filename                                                               \n",
       "245_7_nonhuman.bam         7        NaN  2008-03-21        NaN         NaN   \n",
       "245_8_nonhuman.bam         8        NaN  2008-03-21        NaN         NaN   \n",
       "\n",
       "                   instrument_name instrument_model  forward_read_length  \\\n",
       "irods_filename                                                             \n",
       "245_7_nonhuman.bam             NaN               1G                   37   \n",
       "245_8_nonhuman.bam             NaN               1G                   37   \n",
       "\n",
       "                    requested_insert_size_from  requested_insert_size_to  \\\n",
       "irods_filename                                                             \n",
       "245_7_nonhuman.bam                         NaN                       NaN   \n",
       "245_8_nonhuman.bam                         NaN                       NaN   \n",
       "\n",
       "                    human_percent_mapped   subtrack_filename  \\\n",
       "irods_filename                                                 \n",
       "245_7_nonhuman.bam                   NaN  245_7_nonhuman.srf   \n",
       "245_8_nonhuman.bam                   NaN  245_8_nonhuman.srf   \n",
       "\n",
       "                    subtrack_files_bytes ebi_run_acc  \n",
       "irods_filename                                        \n",
       "245_7_nonhuman.bam                   NaN         NaN  \n",
       "245_8_nonhuman.bam                   NaN         NaN  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_assay_data.loc[df_assay_data['ebi_run_acc'].isnull() & (df_assay_data['id_run'] == 245)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interim conclusions\n",
    "I had previously been thinking that subtrack only contained a subset of the files that could be accessed using queries based on mlwh. However, having spot checked a few of the above, it seems that the cases where we have no accessions from subtrack are due to one of the two following reasons:\n",
    "\n",
    "1. We are looking for a .srf file for older samples, when subtrack actually has a .bam file\n",
    "1. The data are in subtrack, but run accession and bytes are empty as files were not submitted to ENA (as expected for study 1155-PF-ID-PRICE)\n",
    "\n",
    "I will now modify the query to use the first part of the file_name to match (i.e. the basename which ignores the file extension), to check if this is universally true, i.e. that ALL files can be found in subtrack."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def set_irods_name(row):\n",
    "    # First calculate prefix using nonhuman dependent on id_run and species (taxon)\n",
    "    # Then calculate tag string (empty string if no tag)\n",
    "    # Then calculate file extension\n",
    "    # Then concat these three strings\n",
    "    if (\n",
    "        (row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 6750)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    elif (\n",
    "        ~(row['taxon_id'] in [7165, 7173, 30066, 62324]) & # Anopheles gambiae, arabiensis, merus and funestus respectively\n",
    "        (row['id_run'] <= 7100)\n",
    "    ):\n",
    "        prefix = \"%d_%d_nonhuman\" % (row['id_run'], row['position'])\n",
    "    else:\n",
    "        prefix = \"%d_%d\" % (row['id_run'], row['position'])\n",
    "        \n",
    "    if np.isnan(row['tag_index']):\n",
    "        tag_string = ''\n",
    "    else:\n",
    "        tag_string = '#%d' % row['tag_index']\n",
    "    \n",
    "    if row['id_run'] <= 10300:\n",
    "        file_extension = '.bam'\n",
    "    else:\n",
    "        file_extension = '.cram'\n",
    "    \n",
    "    irods_filename = prefix + tag_string + file_extension\n",
    "    return(irods_filename)\n",
    "\n",
    "def set_sample_id(row):\n",
    "    if row['sample'] is None:\n",
    "        return('DS_%s' % row['name'])\n",
    "    else:\n",
    "        return('DS_%s' % row['sample'])\n",
    "\n",
    "def is_valid_alfresco_code(s):\n",
    "    try: \n",
    "        int(s[0:4])\n",
    "        return True\n",
    "    except ValueError:\n",
    "        return False\n",
    "\n",
    "def create_build_manifest(\n",
    "#     taxon_ids=[5833, 5855, 5858, 5821, 7165, 7173, 30066], # Pf, Pv, Pm, Pb, Ag, A. arabiensis, A. merus\n",
    "    taxon_ids=[5833, 36329, 5847, 57267, 137071], # Pf, 3D7, V1, Dd2, HB3\n",
    "    # 1089 is excluded as this is R&D samples. 1204 and 1176 are excluded as these are CP1 samples. 1175 is excluded as some Pf R&D samples were incorrectly tagged with this study. 1157 is a Pv study - there are two suspected Pf samples in this study, but we need further investigation, and possible study change both in ROMA and in Sequencescape/mlwh/ENA/iRODS before we can include\n",
    "    studies_to_exclude = ['1089-R&amp;D-GB-ALCOCK', '1204-PF-GM-CP1', '1176-PF-KE-CP1', '1175-VO-KH-STLAURENT', '1157-PV-MULTI-PRICE'],\n",
    "    miseq_runs_to_allow = [13809, 13810], # Two Miseq runs within 24 samples on each from study 1095-PF-TZ-ISHENGOMA that were included in Pf 6.0\n",
    "    input_dir = input_dir,\n",
    "    mlwh_missing_exceptions_fn = mlwh_missing_exceptions_fn,\n",
    "    mlwh_study_exceptions_fn = mlwh_study_exceptions_fn,\n",
    "    mlwh_sample_exceptions_fn = mlwh_sample_exceptions_fn,\n",
    "    mlwh_taxon_exceptions_fn = mlwh_taxon_exceptions_fn,\n",
    "    sequencescape_alfresco_study_mappings_fn = sequencescape_alfresco_study_mappings_fn,\n",
    "    output_columns = [\n",
    "        'path',\n",
    "        'study',\n",
    "        'sample',\n",
    "        'lane',\n",
    "        'reads',\n",
    "        'paired',\n",
    "        'irods_path',\n",
    "        'sanger_sample_id',\n",
    "        'taxon_id',\n",
    "        'study_lims',\n",
    "        'study_name',\n",
    "        'id_run',\n",
    "        'position',\n",
    "        'tag_index',\n",
    "        'qc_complete',\n",
    "        'manual_qc',\n",
    "        'description',\n",
    "        'instrument_name',\n",
    "        'instrument_model',\n",
    "        'forward_read_length',\n",
    "        'requested_insert_size_from',\n",
    "        'requested_insert_size_to',\n",
    "        'human_percent_mapped',\n",
    "        'subtrack_files_sub_id',\n",
    "        'subtrack_files_file_name',\n",
    "        'subtrack_basename',\n",
    "        'subtrack_files_timestamp',\n",
    "        'subtrack_files_bytes',\n",
    "        'ebi_run_acc'\n",
    "    ]\n",
    "):\n",
    "    \"\"\"Create a DataFrame to be used as a build manifest.\n",
    "    \n",
    "    A build manifest here is a list of all the \"lanelets\" that need to be\n",
    "    included in a build. The output will typically be written to a\n",
    "    tab-delmited file, either to use as input to vr-pipe, or perhaps as\n",
    "    input to a vr-track DB.\n",
    "\n",
    "    Args:\n",
    "        taxon_ids (list of int): The taxons of the build (P. falciparum is '5833',\n",
    "            P. vivax is '5855').\n",
    "        sequencescape_alfresco_study_mappings_fn (str): filename of\n",
    "            mappings from sequencscape to alfresco study mappings\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: The build manifest.\n",
    "\n",
    "    \"\"\"\n",
    "    \n",
    "    # Read in taxon exceptions file\n",
    "    df_mlwh_taxon_exceptions = pd.read_csv(mlwh_taxon_exceptions_fn, sep='\\t', dtype={'tag_index': str, 'taxon_id': int}, index_col='irods_filename')\n",
    "    df_mlwh_taxon_exceptions['tag_index'].fillna('', inplace=True)\n",
    "    # Identify which samples in exceptions file match the taxon of current build, and which don't\n",
    "    df_mlwh_exceptions_this_taxon = df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)]\n",
    "    df_mlwh_exceptions_other_taxa = df_mlwh_taxon_exceptions.loc[~(df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids))]\n",
    "\n",
    "    # Read in sample exceptions file\n",
    "    df_mlwh_sample_exceptions = pd.read_csv(mlwh_sample_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in study exceptions file\n",
    "    df_mlwh_study_exceptions = pd.read_csv(mlwh_study_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "\n",
    "    # Read in missing exceptions file\n",
    "    df_mlwh_missing_exceptions = pd.read_csv(mlwh_missing_exceptions_fn, sep='\\t', index_col='irods_filename')\n",
    "    df_mlwh_missing_exceptions['derivative_sample_id'] = 'DS_' + df_mlwh_missing_exceptions['sample_id']\n",
    "    df_mlwh_missing_exceptions = df_mlwh_missing_exceptions.loc[\n",
    "        df_mlwh_missing_exceptions['taxon_id'].isin(taxon_ids)\n",
    "    ]\n",
    "\n",
    "    # Read in SequenceScape-Alfresco study mappings\n",
    "    df_sequencescape_alfresco_study_mappings = pd.read_csv(sequencescape_alfresco_study_mappings_fn, sep='\\t', index_col='seqscape_study_id')\n",
    "    sequencescape_alfresco_study_mappings_dict = df_sequencescape_alfresco_study_mappings.loc[\n",
    "        :,\n",
    "#         df_sequencescape_alfresco_study_mappings['build_flag']==1,\n",
    "        'alfresco_study_code'\n",
    "    ].to_dict()\n",
    "    \n",
    "    # Read in data from mlwh matching this taxon, plus samples from exceptions file matching this taxon\n",
    "    conn = pymysql.connect(\n",
    "        host='mlwh-db',\n",
    "        user='mlwh_malaria',\n",
    "        password='Solaris&2015',\n",
    "        db='mlwarehouse',\n",
    "        port=3435\n",
    "    )\n",
    "    \n",
    "    sql_query = 'SELECT \\\n",
    "        study.name as study_name, \\\n",
    "        study.id_study_lims as study_lims, \\\n",
    "        sample.supplier_name as sample, \\\n",
    "        sample.name, \\\n",
    "        sample.sanger_sample_id, \\\n",
    "        sample.taxon_id, \\\n",
    "        iseq_product_metrics.id_run, \\\n",
    "        iseq_product_metrics.position, \\\n",
    "        iseq_product_metrics.tag_index, \\\n",
    "        iseq_product_metrics.num_reads, \\\n",
    "        iseq_product_metrics.human_percent_mapped, \\\n",
    "        iseq_run_lane_metrics.instrument_name, \\\n",
    "        iseq_run_lane_metrics.instrument_model, \\\n",
    "        iseq_run_lane_metrics.paired_read, \\\n",
    "        iseq_run_lane_metrics.qc_complete, \\\n",
    "        iseq_flowcell.manual_qc, \\\n",
    "        iseq_flowcell.requested_insert_size_from, \\\n",
    "        iseq_flowcell.requested_insert_size_to, \\\n",
    "        iseq_flowcell.forward_read_length, \\\n",
    "        iseq_run_status_dict.description \\\n",
    "    FROM \\\n",
    "        study, \\\n",
    "        iseq_flowcell, \\\n",
    "        sample, \\\n",
    "        iseq_product_metrics, \\\n",
    "        iseq_run_status, \\\n",
    "        iseq_run_lane_metrics, \\\n",
    "        iseq_run_status_dict \\\n",
    "    WHERE \\\n",
    "        study.id_study_tmp = iseq_flowcell.id_study_tmp and \\\n",
    "        iseq_flowcell.id_sample_tmp = sample.id_sample_tmp and \\\n",
    "        iseq_flowcell.manual_qc = 1 and \\\n",
    "        iseq_product_metrics.id_iseq_flowcell_tmp = iseq_flowcell.id_iseq_flowcell_tmp and \\\n",
    "        iseq_run_status.id_run = iseq_product_metrics.id_run and \\\n",
    "        iseq_product_metrics.id_run = iseq_run_lane_metrics.id_run and \\\n",
    "        iseq_product_metrics.position = iseq_run_lane_metrics.position and \\\n",
    "        iseq_run_status.iscurrent = 1 and \\\n",
    "        ( ( iseq_run_lane_metrics.instrument_model != \"MiSeq\" ) or (iseq_product_metrics.id_run in (%s)) ) and \\\n",
    "        iseq_run_status.id_run_status_dict = iseq_run_status_dict.id_run_status_dict and \\\n",
    "        study.faculty_sponsor = \"Dominic Kwiatkowski\" and \\\n",
    "        ( ( sample.taxon_id in (%s) ) or ' % (\n",
    "            ', '.join([str(x) for x in miseq_runs_to_allow]),\n",
    "            ', '.join([str(x) for x in taxon_ids])\n",
    "        )\n",
    "    sql_query = sql_query + ' or '.join(\n",
    "        df_mlwh_taxon_exceptions.loc[df_mlwh_taxon_exceptions['taxon_id'].isin(taxon_ids)].apply(\n",
    "            lambda x: '(iseq_product_metrics.id_run=\"%s\" and iseq_product_metrics.position=\"%s\" and iseq_product_metrics.tag_index=\"%s\")' % (x['id_run'], x['position'], x['tag_index']),\n",
    "            1\n",
    "        )\n",
    "    )\n",
    "    sql_query = sql_query + ')'\n",
    "#         (iseq_flowcell.manual_qc = 1 or iseq_flowcell.manual_qc is null) and \\\n",
    "    \n",
    "    df_return = pd.read_sql(sql_query, conn)\n",
    "\n",
    "    # Replace missing taxon_id with -1 (can't have missing int values in pandas Series)\n",
    "    df_return['taxon_id'] = df_return['taxon_id'].fillna(-1).astype('int32')\n",
    "\n",
    "    # Determine file name in iRods\n",
    "    df_return['irods_filename'] = df_return.apply(set_irods_name, 1)\n",
    "    df_return.set_index('irods_filename', inplace=True)\n",
    "\n",
    "    # Determine alfresco study and change any incorrect studies\n",
    "    df_return['alfresco_study_code'] = df_return['study_lims'].apply(\n",
    "        lambda x: sequencescape_alfresco_study_mappings_dict[int(x)] if int(x) in sequencescape_alfresco_study_mappings_dict else ''\n",
    "    )\n",
    "    empty_alfresco_study_code = (df_return['alfresco_study_code'] == '')\n",
    "    df_return.loc[empty_alfresco_study_code, 'alfresco_study_code'] = df_return.loc[empty_alfresco_study_code, 'study_name']\n",
    "    \n",
    "    study_exception_indexes = df_mlwh_study_exceptions.index[\n",
    "        df_mlwh_study_exceptions.index.isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[study_exception_indexes, 'alfresco_study_code'] = df_mlwh_study_exceptions.loc[study_exception_indexes, 'alfresco_study_code']\n",
    "\n",
    "    # Change any incorrect taxon IDs\n",
    "    taxon_exception_indexes = df_mlwh_exceptions_this_taxon.index[\n",
    "        df_mlwh_exceptions_this_taxon.index.to_series().isin(df_return.index)\n",
    "    ]\n",
    "    df_return.loc[taxon_exception_indexes, 'taxon_id'] = df_mlwh_exceptions_this_taxon.loc[taxon_exception_indexes, 'taxon_id']\n",
    "\n",
    "    # Determine sample ID and change any incorrect sample IDs\n",
    "    df_return['derivative_sample_id'] = df_return.apply(set_sample_id, 1)\n",
    "    sample_exception_indexes = df_mlwh_sample_exceptions.index[\n",
    "        df_mlwh_sample_exceptions.index.isin(df_return.index)\n",
    "    ]    \n",
    "    df_return.loc[sample_exception_indexes, 'derivative_sample_id'] = 'DS_' + df_mlwh_sample_exceptions.loc[sample_exception_indexes, 'sample_id']\n",
    "    df_return.drop(['sample', 'name'], axis=1, inplace=True)\n",
    "    \n",
    "    # Remove any samples that are not in this taxon\n",
    "    df_return = df_return.loc[~df_return.index.isin(df_mlwh_exceptions_other_taxa.index)]\n",
    "    \n",
    "    # Merge in missing exceptions\n",
    "    df_return = df_return.append(df_mlwh_missing_exceptions.drop('study_group', 1))\n",
    "    \n",
    "    # Remove any samples that don't have an alfresco study\n",
    "    invalid_codes = ~df_return['alfresco_study_code'].apply(is_valid_alfresco_code)\n",
    "    if(np.count_nonzero(invalid_codes) > 0):\n",
    "        print('Removing %d files with invalid alfresco_study_code:' % np.count_nonzero(invalid_codes))\n",
    "        print(df_return['alfresco_study_code'].loc[invalid_codes].value_counts())\n",
    "        df_return = df_return.loc[~invalid_codes]\n",
    "\n",
    "    # Determine basename (i.e. filename without extension) in subtrack\n",
    "    # We use the basename because for many older samples, the file has a .srf extension in subtrack\n",
    "    # but this is not done consistently, e.g. some very old samples such as 245_7_nonhuman have a .bam extension\n",
    "    df_return['subtrack_basename'] = df_return.index.to_series().apply(lambda x: x.split('.')[0])\n",
    "    \n",
    "    # Read in data from substrack, most importantly to get run accessions\n",
    "    sql_query = \"\\\n",
    "    SELECT \\\n",
    "        files.sub_id as subtrack_files_sub_id, \\\n",
    "        files.file_name as subtrack_files_file_name, \\\n",
    "        SUBSTRING_INDEX(files.file_name, '.', 1) as subtrack_basename, \\\n",
    "        files.bytes as subtrack_files_bytes, \\\n",
    "        files.timestamp as subtrack_files_timestamp, \\\n",
    "        files.public_date as subtrack_files_public_date, \\\n",
    "        submission.id as subtrack_submission_id, \\\n",
    "        submission.created as subtrack_submission_created, \\\n",
    "        submission.release_date as subtrack_submission_release_date, \\\n",
    "        submission.timestamp as subtrack_submission_timestamp, \\\n",
    "        submission.ext_db as subtrack_submission_ext_db, \\\n",
    "        submission.ebi_sample_acc, \\\n",
    "        submission.ebi_exp_acc, \\\n",
    "        submission.ebi_study_acc, \\\n",
    "        submission.ebi_sub_acc, \\\n",
    "        submission.ebi_run_acc \\\n",
    "    FROM \\\n",
    "        submission, \\\n",
    "        files \\\n",
    "    WHERE \\\n",
    "        files.sub_id = submission.id AND \\\n",
    "        SUBSTRING_INDEX(files.file_name, '.', 1) in (%s)\\\n",
    "    \" % (\"'\" + \"', '\".join(df_return['subtrack_basename']) + \"'\")\n",
    "\n",
    "    conn= pymysql.connect(\n",
    "        host='shap',\n",
    "        user='ega_dataset',\n",
    "        password='ega_dataset',\n",
    "        db='subtrack',\n",
    "        port=3303\n",
    "    )\n",
    "\n",
    "    df_run_accessions = pd.read_sql(sql_query, conn).set_index('subtrack_basename')\n",
    "    \n",
    "    # Merge in subtrack data\n",
    "    df_return = df_return.join(df_run_accessions, on='subtrack_basename', how='left')\n",
    "\n",
    "    # Remove unwanted studies\n",
    "    df_return = df_return.loc[~df_return['alfresco_study_code'].isin(studies_to_exclude)]\n",
    "\n",
    "    # Create other columns\n",
    "    df_return.rename(columns={'alfresco_study_code': 'study', 'paired_read': 'paired'}, inplace=True)\n",
    "    df_return['lane'] = df_return.index.to_series().apply(lambda x: x.split('.')[0])\n",
    "    df_return['path'] = df_return.apply(lambda x: \"%s/%s\" % (input_dir, x.name), 1)\n",
    "    df_return['reads'] = df_return['num_reads'].fillna(-1).astype(int)\n",
    "    df_return['irods_path'] = df_return.apply(lambda x: \"/seq/%d/%s\" % (x['id_run'], x.name), 1)\n",
    "    df_return['sample'] = df_return['derivative_sample_id'].str.replace('DS_', '')\n",
    "\n",
    "    # Remove lanes with zero reads\n",
    "    df_return = df_return.loc[df_return['reads'] != 0]\n",
    "    \n",
    "    # Remove control samples (see emails from Sonia 03/05/2018 12:20 and from Richard 09/10/2018 13:26)\n",
    "#     df_return = df_return.loc[df_return['sample'].str.slice(0, 1) != 'C']\n",
    "    df_return = df_return.loc[(df_return['sample'].str.slice(0, 1) != 'C') & (df_return['sample'] != 'control')]\n",
    "    \n",
    "    # Restrict to certain columns\n",
    "    if output_columns is not None:\n",
    "        df_return = (\n",
    "            df_return[output_columns]\n",
    "            .sort_values(['study', 'sample', 'id_run', 'position', 'tag_index'])\n",
    "        )\n",
    "    \n",
    "    print(df_return.shape)\n",
    "\n",
    "    return(df_return) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Removing 63 files with invalid alfresco_study_code:\n",
      "AT-rich amplification                                 53\n",
      "Test sequencing of high level human contamination.     6\n",
      "Plasmodium falciparum genome variation 1               4\n",
      "Name: alfresco_study_code, dtype: int64\n",
      "(18130, 29)\n"
     ]
    }
   ],
   "source": [
    "df_new_assay_data = create_build_manifest()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check all files found in subtrack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "id_run  study           \n",
       "10901   1155-PF-ID-PRICE     6\n",
       "11019   1155-PF-ID-PRICE     2\n",
       "12299   1155-PF-ID-PRICE    14\n",
       "14323   1155-PF-ID-PRICE     1\n",
       "14324   1155-PF-ID-PRICE     3\n",
       "16166   1155-PF-ID-PRICE    36\n",
       "16189   1155-PF-ID-PRICE    11\n",
       "16229   1155-PF-ID-PRICE    19\n",
       "25237   1155-PF-ID-PRICE    37\n",
       "25552   1155-PF-ID-PRICE     3\n",
       "dtype: int64"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new_assay_data.loc[df_new_assay_data['ebi_run_acc'].isnull(), ['id_run', 'study']].groupby(['id_run', 'study']).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>path</th>\n",
       "      <th>study</th>\n",
       "      <th>sample</th>\n",
       "      <th>lane</th>\n",
       "      <th>reads</th>\n",
       "      <th>paired</th>\n",
       "      <th>irods_path</th>\n",
       "      <th>sanger_sample_id</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>study_lims</th>\n",
       "      <th>study_name</th>\n",
       "      <th>id_run</th>\n",
       "      <th>position</th>\n",
       "      <th>tag_index</th>\n",
       "      <th>qc_complete</th>\n",
       "      <th>manual_qc</th>\n",
       "      <th>description</th>\n",
       "      <th>instrument_name</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>forward_read_length</th>\n",
       "      <th>requested_insert_size_from</th>\n",
       "      <th>requested_insert_size_to</th>\n",
       "      <th>human_percent_mapped</th>\n",
       "      <th>subtrack_files_sub_id</th>\n",
       "      <th>subtrack_files_file_name</th>\n",
       "      <th>subtrack_basename</th>\n",
       "      <th>subtrack_files_timestamp</th>\n",
       "      <th>subtrack_files_bytes</th>\n",
       "      <th>ebi_run_acc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>irods_filename</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25552_1#7.cram</th>\n",
       "      <td>dummy/25552_1#7.cram</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>SPT24658</td>\n",
       "      <td>25552_1#7</td>\n",
       "      <td>41337656</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/25552/25552_1#7.cram</td>\n",
       "      <td>5101STDY7247317</td>\n",
       "      <td>5833</td>\n",
       "      <td>5101</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>25552</td>\n",
       "      <td>1</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2018-04-05 12:42:30</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HX7</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>96.29</td>\n",
       "      <td>1963391</td>\n",
       "      <td>25552_1#7.cram</td>\n",
       "      <td>25552_1#7</td>\n",
       "      <td>2018-04-06 01:20:52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25552_1#10.cram</th>\n",
       "      <td>dummy/25552_1#10.cram</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>SPT24662</td>\n",
       "      <td>25552_1#10</td>\n",
       "      <td>38311728</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/25552/25552_1#10.cram</td>\n",
       "      <td>5101STDY7247321</td>\n",
       "      <td>5833</td>\n",
       "      <td>5101</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>25552</td>\n",
       "      <td>1</td>\n",
       "      <td>10.0</td>\n",
       "      <td>2018-04-05 12:42:30</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HX7</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>95.05</td>\n",
       "      <td>1963394</td>\n",
       "      <td>25552_1#10.cram</td>\n",
       "      <td>25552_1#10</td>\n",
       "      <td>2018-04-06 01:20:52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25552_1#21.cram</th>\n",
       "      <td>dummy/25552_1#21.cram</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>SPT24686</td>\n",
       "      <td>25552_1#21</td>\n",
       "      <td>33226936</td>\n",
       "      <td>1.0</td>\n",
       "      <td>/seq/25552/25552_1#21.cram</td>\n",
       "      <td>5101STDY7247345</td>\n",
       "      <td>5833</td>\n",
       "      <td>5101</td>\n",
       "      <td>1155-PF-ID-PRICE</td>\n",
       "      <td>25552</td>\n",
       "      <td>1</td>\n",
       "      <td>21.0</td>\n",
       "      <td>2018-04-05 12:42:30</td>\n",
       "      <td>1.0</td>\n",
       "      <td>qc complete</td>\n",
       "      <td>HX7</td>\n",
       "      <td>HiSeqX</td>\n",
       "      <td>150</td>\n",
       "      <td>450.0</td>\n",
       "      <td>450.0</td>\n",
       "      <td>97.55</td>\n",
       "      <td>1963405</td>\n",
       "      <td>25552_1#21.cram</td>\n",
       "      <td>25552_1#21</td>\n",
       "      <td>2018-04-06 01:20:52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  path             study    sample  \\\n",
       "irods_filename                                                       \n",
       "25552_1#7.cram    dummy/25552_1#7.cram  1155-PF-ID-PRICE  SPT24658   \n",
       "25552_1#10.cram  dummy/25552_1#10.cram  1155-PF-ID-PRICE  SPT24662   \n",
       "25552_1#21.cram  dummy/25552_1#21.cram  1155-PF-ID-PRICE  SPT24686   \n",
       "\n",
       "                       lane     reads  paired                  irods_path  \\\n",
       "irods_filename                                                              \n",
       "25552_1#7.cram    25552_1#7  41337656     1.0   /seq/25552/25552_1#7.cram   \n",
       "25552_1#10.cram  25552_1#10  38311728     1.0  /seq/25552/25552_1#10.cram   \n",
       "25552_1#21.cram  25552_1#21  33226936     1.0  /seq/25552/25552_1#21.cram   \n",
       "\n",
       "                sanger_sample_id  taxon_id study_lims        study_name  \\\n",
       "irods_filename                                                            \n",
       "25552_1#7.cram   5101STDY7247317      5833       5101  1155-PF-ID-PRICE   \n",
       "25552_1#10.cram  5101STDY7247321      5833       5101  1155-PF-ID-PRICE   \n",
       "25552_1#21.cram  5101STDY7247345      5833       5101  1155-PF-ID-PRICE   \n",
       "\n",
       "                 id_run  position  tag_index          qc_complete  manual_qc  \\\n",
       "irods_filename                                                                 \n",
       "25552_1#7.cram    25552         1        7.0  2018-04-05 12:42:30        1.0   \n",
       "25552_1#10.cram   25552         1       10.0  2018-04-05 12:42:30        1.0   \n",
       "25552_1#21.cram   25552         1       21.0  2018-04-05 12:42:30        1.0   \n",
       "\n",
       "                 description instrument_name instrument_model  \\\n",
       "irods_filename                                                  \n",
       "25552_1#7.cram   qc complete             HX7           HiSeqX   \n",
       "25552_1#10.cram  qc complete             HX7           HiSeqX   \n",
       "25552_1#21.cram  qc complete             HX7           HiSeqX   \n",
       "\n",
       "                 forward_read_length  requested_insert_size_from  \\\n",
       "irods_filename                                                     \n",
       "25552_1#7.cram                   150                       450.0   \n",
       "25552_1#10.cram                  150                       450.0   \n",
       "25552_1#21.cram                  150                       450.0   \n",
       "\n",
       "                 requested_insert_size_to  human_percent_mapped  \\\n",
       "irods_filename                                                    \n",
       "25552_1#7.cram                      450.0                 96.29   \n",
       "25552_1#10.cram                     450.0                 95.05   \n",
       "25552_1#21.cram                     450.0                 97.55   \n",
       "\n",
       "                 subtrack_files_sub_id subtrack_files_file_name  \\\n",
       "irods_filename                                                    \n",
       "25552_1#7.cram                 1963391           25552_1#7.cram   \n",
       "25552_1#10.cram                1963394          25552_1#10.cram   \n",
       "25552_1#21.cram                1963405          25552_1#21.cram   \n",
       "\n",
       "                subtrack_basename subtrack_files_timestamp  \\\n",
       "irods_filename                                               \n",
       "25552_1#7.cram          25552_1#7      2018-04-06 01:20:52   \n",
       "25552_1#10.cram        25552_1#10      2018-04-06 01:20:52   \n",
       "25552_1#21.cram        25552_1#21      2018-04-06 01:20:52   \n",
       "\n",
       "                 subtrack_files_bytes ebi_run_acc  \n",
       "irods_filename                                     \n",
       "25552_1#7.cram                    NaN        None  \n",
       "25552_1#10.cram                   NaN        None  \n",
       "25552_1#21.cram                   NaN        None  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new_assay_data.loc[df_new_assay_data['ebi_run_acc'].isnull() & (df_new_assay_data['id_run'] == 25552)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], dtype: int64)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new_assay_data.loc[df_new_assay_data['subtrack_files_sub_id'].isnull(), ['id_run', 'study']].groupby(['id_run', 'study']).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], dtype: int64)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new_assay_data.loc[df_new_assay_data['subtrack_files_file_name'].isnull(), ['id_run', 'study']].groupby(['id_run', 'study']).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], dtype: int64)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new_assay_data.loc[df_new_assay_data['subtrack_files_timestamp'].isnull(), ['id_run', 'study']].groupby(['id_run', 'study']).size()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write out file\n",
    "This might be useful for future comparisons of my approach to creating build manifests vs FITS. Rob has said he might find this file useful."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_new_assay_data.to_csv('Pf_irods_manifest_20190117.txt.gz', sep='\\t', compression='gzip')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Conclusions\n",
    "I had previously been assuming that there are files that can be identified through mlwh queries that can not be found in subtrack. It turns out that this assumption is not true, and that I had previously been assuming it was true because in some cases I was assuming the file in subtrack would have a .srf extension when in fact the file had a .bam extension. After modifying my subtrack query to match using only the file basename (i.e. ignoring the file extension), all files found using my mlwh query could also be found in subtrack (at least this is true for the 18,310 Pf files currently found by my query)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
