# File tracking system minimal viable product scope

## Introduction
This document describes the scope of the minimal viable product (MVP) for a new file tracking system (preliminarily designated FITS). The problem that FITS is intended to solve is described in the next section. The scope of the MVP is described in the following section. Following this, some key use cases for FITS are described, including an indication of whether these use cases will be catered for by the MVP. This document concludes with pointers to other relevant documents, relationship to the separate sample information management system (SIMS) project, and future plans.

## Problem
For our work, we rely on data files generated by machines (e.g. DNA sequencing, genotyping), supplied by third parties, or generated from such files through computation. We need to be able to know where all these files are, what they contain, how they relate to samples, and to each other.

Previously, the information described above has either been sourced from the Solaris database, from Sanger sequencing systems (Multi-LIMS warehouse (mlwh), iRODs baton) or from a combination of Solaris and Sanger sequencing systems. This approach is considered non-viable going forwards because:
* The Solaris database is being decommissioned and is no longer been populated with new data
* There are known errors in file-sample mappings in the Multi-LIMS warehouse
* We will need to store data that are not in Sanger systems, e.g. files of samples sequenced at locations other than the Sanger which we want to include in builds and releases

## Scope
The MVP _will_
* Store the location of files in various storage systems
* Store metadata of these files (e.g. file type, file size, MD5 sum)
* Store metadata about the data contained these files (e.g. insert size of paired end sequencing data)
* Store the relations of files to each other (e.g. a BAM and a CRAM file containing the same read data, which files(s) another file is based on)
* Store sample identifiers and their relation to the files
* Store metadata about the sample IDs (e.g. Oxford/ROMA codes)
* Be updated via automated processes (e.g. a cronjob reading Sanger sequencing updates)
* Be updated manually (e.g. to correct false legacy data)
* Be queried to get information about samples
* Be queried to get information about files
* Store mappings from files to samples for all parasite and vector MalariaGEN samples that have been whole-genome sequenced using Illumina technology at the Sanger

The MVP will _not necessarily_ store mappings from files to samples for 
* amplicon sequencing data
* other genotyping technologies such as Sequenom/Agena
* human samples
* samples not sequenced at the Sanger

## Use cases
(Tracking of use cases)[https://github.com/malariagen/fits/labels/use%20case].

The section describes a set of use cases that must be satisfied for the MVP to be considered complete, together with a brief description of their status in the current version of the file tracking system at the time of writing. The appendix gives details of queries related to some of these use cases that can currently be performed in FITS at the time of writing.

In the following, a build manifest is considered to be a file with one line per file, and to contain at the minimum:
* iRODs path
* Sample ID (Oxford code or ROMA ID)
* Alfresco study code
* NCBI taxon ID
* Manual QC status
* Date manual QC complete
* ENA run accession
* ENA sample accession

Note that in the future, it is expected that information on Sample IDs, Alfresco study codes and taxon IDs will come from SIMS, but until SIMS is operational, this information will have to come from FITS, otherwise it will not be possible to use FITS for creating build manifests. The definitive source for file-to-sample, sample-to-study and sample-to-taxon mappings should be considered to be Solaris for samples contained in Solaris, or mlwh/iRODS for samples that are not. Note that some older files are included in Solaris but not in mlwh.

For an example of a recently created build manifest, see https://github.com/wtsi-team112/Pv4/blob/master/notebooks/rp7/20180525_Pv4_manifest.ipynb

### Create a build manifest containing all samples from a species
See (issue)[https://github.com/malariagen/fits/issues/25].

### Create a build manifest given a set of sequencescape IDs, oxford code/ROMA IDs or Alfresco study codes
See (issue)[https://github.com/malariagen/fits/issues/26].

### Determine which samples from a given study have been sequenced
See (issue)[https://github.com/malariagen/fits/issues/28].

### Determine how many samples have been sequenced, broken down by species
See (issue)[https://github.com/malariagen/fits/issues/29].

### Populate FITS with mappings from Solaris
Done.

### Manually alter FITS mappings
See (issue)[https://github.com/malariagen/fits/issues/32].

### Update the file tracking database with the latest from Multi-LIMS warehouse/baton (iRODs)
See (issue)[https://github.com/malariagen/fits/issues/31].

## Other documents
* (Database description)[https://github.com/wtsi-team112/fits/blob/master/documentation/database_design_v1.md]
* (Command line utility description)[https://github.com/wtsi-team112/fits/blob/master/documentation/fits_command_line_tool.md]
* (Process description)[https://github.com/wtsi-team112/fits/blob/master/documentation/processes.md]

## Relationship to SIMS
For the moment, FITS and SIMS are developed separately. FITS will eventually pull information from SIMS via API, but there is no perceived need for information to flow in the other direction. FITS can match samples and files to SIMS UUIDs via high-level (e.g. ROMA/Oxford) or low-level (e.g. Sequenscape) IDs.

## Future plans
The following is a list of things that are considered to be out of scope for the MVP.
(Tracked as "enhancement")[https://github.com/malariagen/fits/labels/enhancement]

### Sequencing not done at WSI
FITS can store any file, given a unique path and a storage “engine”. Currently, the only engine supported is Sanger Sequencing iRODs, but more can be added without changing the database schema. Processes to import and update such data will have to be developed.

The first use case here is likely to be storing ENA run accessions.

(Issue)[https://github.com/malariagen/fits/issues/33]

### Expanding the scope to include amplicon sequencing data
FITS is agnostic to file types. Storage engines such as “Team 112 iRODs” or “Sanger NFS” can be added. Processes to import and update such data will have to be developed.
(Issue)[https://github.com/malariagen/fits/issues/34]

### Expanding the scope to include human data
FITS is agnostic to species, however, storing of data about human samples might influence the storage location (US cloud?).
(Issue)[https://github.com/malariagen/fits/issues/35]

### Expanding the scope to include pipeline outputs, e.g. release files
FITS is agnostic to file types.  Processes to import and update such data will have to be developed.
(Issue)[https://github.com/malariagen/fits/issues/36]

### Command line query tool
A command line tool exists. It can be used to update FITS from the  Multi-LIMS warehouse database. A query engine to ease generation of manifest files without compromising generic queries is under development.
(Issue)[https://github.com/malariagen/fits/issues/37]

### Web front end
A web frontend is not planned at this time, however, one could be developed rather easily.
(Issue)[https://github.com/malariagen/fits/issues/38]

### Moving the system to a public cloud
The FITS database currently resides at Sanger. For better interoperability with Oxford and/or cloud locations, a move to a cloud-based database system is planned once the MVP has stabilized. A database-as-a-service, rather than a generic server running a MySQL client, would be preferred, for ease of maintenance, backups, availability etc.
Storage of human sample data might complicate finding an adequate cloud location.
(Issue)[https://github.com/malariagen/fits/issues/39]